{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "What was your biggest challenge in this project? (regarding writing code and not only)    \n",
    "What did you learn while working on this project? (regarding writing code and not only)     \n",
    "If you had more time on the project what other question(s) would you like to answer? (at least one question is required)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "pip install GEOparse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "22-Nov-2023 12:23:14 INFO GEOparse - Parsing ./GSE10245_family.soft.gz: \n",
      "22-Nov-2023 12:23:14 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "22-Nov-2023 12:23:14 DEBUG GEOparse - SERIES: GSE10245\n",
      "22-Nov-2023 12:23:14 DEBUG GEOparse - PLATFORM: GPL570\n",
      "/Users/ymanasa/anaconda3/lib/python3.11/site-packages/GEOparse/GEOparse.py:401: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258551\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258552\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258553\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258554\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258555\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258556\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258557\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258558\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258559\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258560\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258561\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258562\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258563\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258564\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258565\n",
      "22-Nov-2023 12:23:16 DEBUG GEOparse - SAMPLE: GSM258566\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258567\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258568\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258569\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258570\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258571\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258572\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258573\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258574\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258575\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258576\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258577\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258578\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258579\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258580\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258581\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258582\n",
      "22-Nov-2023 12:23:17 DEBUG GEOparse - SAMPLE: GSM258583\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258584\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258585\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258586\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258587\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258588\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258589\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258590\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258591\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258592\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258593\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258594\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258595\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258596\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258597\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258598\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258599\n",
      "22-Nov-2023 12:23:18 DEBUG GEOparse - SAMPLE: GSM258600\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258601\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258602\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258603\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258604\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258605\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258606\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258607\n",
      "22-Nov-2023 12:23:19 DEBUG GEOparse - SAMPLE: GSM258608\n"
     ]
    }
   ],
   "source": [
    "#Load the data\n",
    "import GEOparse\n",
    "gse = GEOparse.get_GEO(filepath=\"./GSE10245_family.soft.gz\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "pip install matplotlib\n",
    "pip install seaborn\n",
    "pip install sklearn.cluster"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Perform clustering - you can use the sklearn.cluster module for this task.\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns; sns.set()\n",
    "import numpy as np\n",
    "from sklearn.cluster import KMeans\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Q1. Load the data set and do data processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GSM example:\n",
      "Name:  GSM258551\n",
      "Metadata:\n",
      " - title : NSCLC_AC_10\n",
      " - geo_accession : GSM258551\n",
      " - status : Public on Oct 01 2009\n",
      " - submission_date : Jan 23 2008\n",
      " - last_update_date : Aug 28 2018\n",
      " - type : RNA\n",
      " - channel_count : 1\n",
      " - source_name_ch1 : human non-small cell lung cancer tumor tissue\n",
      " - organism_ch1 : Homo sapiens\n",
      " - taxid_ch1 : 9606\n",
      " - characteristics_ch1 : disease state: adenocarcinoma\n",
      " - treatment_protocol_ch1 : tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.\n",
      " - molecule_ch1 : total RNA\n",
      " - extract_protocol_ch1 : total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.\n",
      " - label_ch1 : biotin\n",
      " - label_protocol_ch1 : Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).\n",
      " - hyb_protocol : Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.\n",
      " - scan_protocol : Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.\n",
      " - description : Gene expression data from human non-small cell lung cancer tumor tissue 10.\n",
      " - data_processing : Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].\n",
      " - platform_id : GPL570\n",
      " - contact_name : Ruprecht,,Kuner\n",
      " - contact_laboratory : Unit Cancer Genome Research\n",
      " - contact_department : Molecular Genetics\n",
      " - contact_institute : German Cancer Research Center and National Center of Tumor Diseases\n",
      " - contact_address : Im Neuenheimer Feld 460\n",
      " - contact_city : Heidelberg\n",
      " - contact_zip/postal_code : 69120\n",
      " - contact_country : Germany\n",
      " - supplementary_file : ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258551/suppl/GSM258551.CEL.gz\n",
      " - relation : Reanalyzed by: GSE60486, Reanalyzed by: GSE119087\n",
      " - series_id : GSE10245\n",
      " - data_row_count : 54675\n",
      "Table data:\n",
      "      ID_REF     VALUE\n",
      "0  1007_s_at  9.129905\n",
      "1    1053_at  8.034022\n",
      "2     117_at  3.564520\n",
      "3     121_at  4.746490\n",
      "4  1255_g_at  2.320698\n"
     ]
    }
   ],
   "source": [
    "#basic information of the data set\n",
    "print()\n",
    "print(\"GSM example:\")\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "    print(\"Name: \", gsm_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gsm.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print (\"Table data:\",)\n",
    "    print (gsm.table.head())\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GPL example:\n",
      "Name:  GPL570\n",
      "Metadata:\n",
      " - title : [HG-U133_Plus_2] Affymetrix Human Genome U133 Plus 2.0 Array\n",
      " - geo_accession : GPL570\n",
      " - status : Public on Nov 07 2003\n",
      " - submission_date : Nov 07 2003\n",
      " - last_update_date : Dec 14 2020\n",
      " - technology : in situ oligonucleotide\n",
      " - distribution : commercial\n",
      " - organism : Homo sapiens\n",
      " - taxid : 9606\n",
      " - manufacturer : Affymetrix\n",
      " - manufacture_protocol : see manufacturer's web site, , Complete coverage of the Human Genome U133 Set plus 6,500 additional genes for analysis of over 47,000 transcripts, All probe sets represented on the GeneChip Human Genome U133 Set are identically replicated on the GeneChip Human Genome U133 Plus 2.0 Array. The sequences from which these probe sets were derived were selected from GenBank®, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 133, April 20, 2001) and then refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the University of California, Santa Cruz Golden-Path human genome database (April 2001 release)., In addition, there are 9,921 new probe sets representing approximately 6,500 new genes. These gene sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from the UniGene database (Build 159, January 25, 2003) and refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the NCBI human genome assembly (Build 31)., \n",
      " - description : Affymetrix submissions are typically submitted to GEO using the GEOarchive method described at http://www.ncbi.nlm.nih.gov/projects/geo/info/geo_affy.html, , June 03, 2009: annotation table updated with netaffx build 28, June 06, 2012: annotation table updated with netaffx build 32, June 23, 2016: annotation table updated with netaffx build 35\n",
      " - web_link : http://www.affymetrix.com/support/technical/byproduct.affx?product=hg-u133-plus, http://www.affymetrix.com/analysis/index.affx\n",
      " - contact_name : ,,Affymetrix, Inc.\n",
      " - contact_email : geo@ncbi.nlm.nih.gov, support@affymetrix.com\n",
      " - contact_phone : 888-362-2447\n",
      " - contact_institute : Affymetrix, Inc.\n",
      " - contact_address : \n",
      " - contact_city : Santa Clara\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 95051\n",
      " - contact_country : USA\n",
      " - contact_web_link : http://www.affymetrix.com/index.affx\n",
      " - relation : Alternative to: GPL4454 (Alternative CDF), Alternative to: GPL4866 (Alternative CDF), Alternative to: GPL5760 (Alternative CDF), Alternative to: GPL6671 (Alternative CDF), Alternative to: GPL6791 (Alternative CDF), Alternative to: GPL6879 (Alternative CDF), Alternative to: GPL7567 (Alternative CDF), Alternative to: GPL8019 (Alternative CDF), Alternative to: GPL8542 (Alternative CDF), Alternative to: GPL8715 (Alternative CDF), Alternative to: GPL8712 (Alternative CDF), Alternative to: GPL9102 (Probe Level Version), Alternative to: GPL9099 (Alternative CDF), Alternative to: GPL9101 (Alternative CDF), Alternative to: GPL9324 (Alternative CDF), Alternative to: GPL9486 (Alternative CDF), Alternative to: GPL9987 (Alternative CDF), Alternative to: GPL10175 (Alternative CDF), Alternative to: GPL10335 (Alternative CDF), Alternative to: GPL10371 (Alternative CDF), Alternative to: GPL10526 (Alternative CDF), Alternative to: GPL10881 (Alternative CDF), Alternative to: GPL10925 (Alternative CDF), Alternative to: GPL11084 (Alternative CDF), Alternative to: GPL11433 (Alternative CDF), Alternative to: GPL11670 (Alternative CDF), Alternative to: GPL13232 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 13.0.0]), Alternative to: GPL13303 (Alternative CDF.), Alternative to: GPL13916 ([Brainarray ENSG Version 14.1]), Alternative to: GPL14837 (Alternative CDF), Alternative to: GPL14877 ([Alternative CDF]), Alternative to: GPL15394 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 14.0]), Alternative to: GPL6732 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 7]), Alternative to: GPL15445 (Alternative CDF), Alternative to: GPL16066 (Alternative CDF), Alternative to: GPL16100 (Alternative CDF [GATExplorer_Ensembl v57]), Alternative to: GPL16268 (Alternative CDF [Brainarray ENSG Version 16]), Alternative to: GPL16273 (Alternative CDF [Brainarray ENSG Version 15]), Alternative to: GPL16311 (Alternative CDF [HGU133Plus2_Hs_ENTREZG Brainarray version 14.0.0]), Alternative to: GPL16356 (Alternative CDF [Brainarray HGU133Plus2_Hs_ENTREZG 15.0.0]), Alternative to: GPL16372 (Alternative CDF [Brainarray HGU133Plus2_Hs_UG v.13]), Alternative to: GPL17392 (Alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray 17.1]), Alternative to: GPL17394 (Alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray 13.0]), Alternative to: GPL17810 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 16]), Alternative to: GPL17811 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 17]), Alternative to: GPL17996 (Alternative CDF:HGU133Plus2_Hs_ENTREZG.cdf version 15.1.0), Alternative to: GPL18121 (custom CDF), Alternative to: GPL18756 (Alternative CDF [CDF: HGU133Plus2_Hs_ENTREZG_14.1.0]), Alternative to: GPL18850 (Alternative CDF), Alternative to: GPL18478 (Alternative CDF [Hs_ENTREZG_V13]), Alternative to: GPL19109 (Alternative CDF [Brainarray HGU133Plus2_Hs_ENTREZG_v18]), Alternative to: GPL19883 (Alternative CDF [Broad GenePattern]), Alternative to: GPL19918 (Gene symbol version, 10K), Alternative to: GPL20182 (Gene Symbol Version), Alternative to: GPL22321 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 20]), Alternative to: GPL22945 (alternative), Alternative to: GPL23270 ((Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 21]), Alternative to: GPL23432 (alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray version 18.0.0]), Alternative to: GPL25740 (alternative), Alternative to: GPL28718 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 23]), Alternative to: GPL29499 (Alternative CDF)\n",
      " - data_row_count : 54675\n",
      "Table data:\n",
      "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
      "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
      "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
      "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
      "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
      "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
      "\n",
      "       Sequence Type                  Sequence Source  \\\n",
      "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "1  Exemplar sequence                          GenBank   \n",
      "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "3  Exemplar sequence                          GenBank   \n",
      "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "\n",
      "                                  Target Description Representative Public ID  \\\n",
      "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
      "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
      "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
      "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
      "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
      "\n",
      "                                          Gene Title       Gene Symbol  \\\n",
      "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
      "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
      "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
      "3                                       paired box 8              PAX8   \n",
      "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
      "\n",
      "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
      "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
      "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
      "2               3310                                          NM_002155   \n",
      "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
      "4               2978                         NM_000409 /// XM_006715073   \n",
      "\n",
      "                    Gene Ontology Biological Process  \\\n",
      "0  0001558 // regulation of cell growth // inferr...   \n",
      "1  0000278 // mitotic cell cycle // traceable aut...   \n",
      "2  0000902 // cell morphogenesis // inferred from...   \n",
      "3  0001655 // urogenital system development // in...   \n",
      "4  0007165 // signal transduction // non-traceabl...   \n",
      "\n",
      "                    Gene Ontology Cellular Component  \\\n",
      "0  0005576 // extracellular region // inferred fr...   \n",
      "1  0005634 // nucleus // inferred from electronic...   \n",
      "2  0005737 // cytoplasm // inferred from direct a...   \n",
      "3  0005634 // nucleus // inferred from direct ass...   \n",
      "4  0001750 // photoreceptor outer segment // infe...   \n",
      "\n",
      "                    Gene Ontology Molecular Function  \n",
      "0  0000166 // nucleotide binding // inferred from...  \n",
      "1  0000166 // nucleotide binding // inferred from...  \n",
      "2  0000166 // nucleotide binding // inferred from...  \n",
      "3  0000979 // RNA polymerase II core promoter seq...  \n",
      "4  0005509 // calcium ion binding // inferred fro...  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>GB_ACC</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <th>Annotation Date</th>\n",
       "      <th>Sequence Type</th>\n",
       "      <th>Sequence Source</th>\n",
       "      <th>Target Description</th>\n",
       "      <th>Representative Public ID</th>\n",
       "      <th>Gene Title</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <th>RefSeq Transcript ID</th>\n",
       "      <th>Gene Ontology Biological Process</th>\n",
       "      <th>Gene Ontology Cellular Component</th>\n",
       "      <th>Gene Ontology Molecular Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>U48705</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...</td>\n",
       "      <td>U48705</td>\n",
       "      <td>discoidin domain receptor tyrosine kinase 1 //...</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "      <td>NM_001202521 /// NM_001202522 /// NM_001202523...</td>\n",
       "      <td>0001558 // regulation of cell growth // inferr...</td>\n",
       "      <td>0005576 // extracellular region // inferred fr...</td>\n",
       "      <td>0000166 // nucleotide binding // inferred from...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>M87338</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...</td>\n",
       "      <td>M87338</td>\n",
       "      <td>replication factor C (activator 1) 2, 40kDa</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "      <td>NM_001278791 /// NM_001278792 /// NM_001278793...</td>\n",
       "      <td>0000278 // mitotic cell cycle // traceable aut...</td>\n",
       "      <td>0005634 // nucleus // inferred from electronic...</td>\n",
       "      <td>0000166 // nucleotide binding // inferred from...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>X51757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...</td>\n",
       "      <td>X51757</td>\n",
       "      <td>heat shock 70kDa protein 6 (HSP70B')</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "      <td>NM_002155</td>\n",
       "      <td>0000902 // cell morphogenesis // inferred from...</td>\n",
       "      <td>0005737 // cytoplasm // inferred from direct a...</td>\n",
       "      <td>0000166 // nucleotide binding // inferred from...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>X69699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...</td>\n",
       "      <td>X69699</td>\n",
       "      <td>paired box 8</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "      <td>NM_003466 /// NM_013951 /// NM_013952 /// NM_0...</td>\n",
       "      <td>0001655 // urogenital system development // in...</td>\n",
       "      <td>0005634 // nucleus // inferred from direct ass...</td>\n",
       "      <td>0000979 // RNA polymerase II core promoter seq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>L36861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...</td>\n",
       "      <td>L36861</td>\n",
       "      <td>guanylate cyclase activator 1A (retina)</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "      <td>NM_000409 /// XM_006715073</td>\n",
       "      <td>0007165 // signal transduction // non-traceabl...</td>\n",
       "      <td>0001750 // photoreceptor outer segment // infe...</td>\n",
       "      <td>0005509 // calcium ion binding // inferred fro...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
       "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
       "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
       "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
       "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
       "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
       "\n",
       "       Sequence Type                  Sequence Source  \\\n",
       "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
       "1  Exemplar sequence                          GenBank   \n",
       "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
       "3  Exemplar sequence                          GenBank   \n",
       "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
       "\n",
       "                                  Target Description Representative Public ID  \\\n",
       "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
       "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
       "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
       "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
       "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
       "\n",
       "                                          Gene Title       Gene Symbol  \\\n",
       "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
       "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
       "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
       "3                                       paired box 8              PAX8   \n",
       "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
       "\n",
       "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
       "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
       "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
       "2               3310                                          NM_002155   \n",
       "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
       "4               2978                         NM_000409 /// XM_006715073   \n",
       "\n",
       "                    Gene Ontology Biological Process  \\\n",
       "0  0001558 // regulation of cell growth // inferr...   \n",
       "1  0000278 // mitotic cell cycle // traceable aut...   \n",
       "2  0000902 // cell morphogenesis // inferred from...   \n",
       "3  0001655 // urogenital system development // in...   \n",
       "4  0007165 // signal transduction // non-traceabl...   \n",
       "\n",
       "                    Gene Ontology Cellular Component  \\\n",
       "0  0005576 // extracellular region // inferred fr...   \n",
       "1  0005634 // nucleus // inferred from electronic...   \n",
       "2  0005737 // cytoplasm // inferred from direct a...   \n",
       "3  0005634 // nucleus // inferred from direct ass...   \n",
       "4  0001750 // photoreceptor outer segment // infe...   \n",
       "\n",
       "                    Gene Ontology Molecular Function  \n",
       "0  0000166 // nucleotide binding // inferred from...  \n",
       "1  0000166 // nucleotide binding // inferred from...  \n",
       "2  0000166 // nucleotide binding // inferred from...  \n",
       "3  0000979 // RNA polymerase II core promoter seq...  \n",
       "4  0005509 // calcium ion binding // inferred fro...  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#gene information from Gene Expression Omnibus Platform (GPL)\n",
    "print()    \n",
    "print(\"GPL example:\")    \n",
    "for gpl_name, gpl in gse.gpls.items():    \n",
    "    print(\"Name: \", gpl_name)    \n",
    "    print(\"Metadata:\",)    \n",
    "    for key, value in gpl.metadata.items():    \n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))    \n",
    "    print(\"Table data:\",)    \n",
    "    print(gpl.table.head())    \n",
    "    break   \n",
    "gpl.table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'GB_ACC', 'SPOT_ID', 'Species Scientific Name', 'Annotation Date',\n",
       "       'Sequence Type', 'Sequence Source', 'Target Description',\n",
       "       'Representative Public ID', 'Gene Title', 'Gene Symbol',\n",
       "       'ENTREZ_GENE_ID', 'RefSeq Transcript ID',\n",
       "       'Gene Ontology Biological Process', 'Gene Ontology Cellular Component',\n",
       "       'Gene Ontology Molecular Function'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gpl.table[\"Gene Symbol\"].isnull().any()\n",
    "gpl.table.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54630</th>\n",
       "      <td>AFFX-HUMGAPDH/M33197_M_at</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>2597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54631</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_3_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54632</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_5_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54633</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MA_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54634</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MB_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44134 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                ID       Gene Symbol     ENTREZ_GENE_ID\n",
       "0                        1007_s_at  DDR1 /// MIR4640  780 /// 100616237\n",
       "1                          1053_at              RFC2               5982\n",
       "2                           117_at             HSPA6               3310\n",
       "3                           121_at              PAX8               7849\n",
       "4                        1255_g_at            GUCA1A               2978\n",
       "...                            ...               ...                ...\n",
       "54630    AFFX-HUMGAPDH/M33197_M_at             GAPDH               2597\n",
       "54631   AFFX-HUMISGF3A/M97935_3_at             STAT1               6772\n",
       "54632   AFFX-HUMISGF3A/M97935_5_at             STAT1               6772\n",
       "54633  AFFX-HUMISGF3A/M97935_MA_at             STAT1               6772\n",
       "54634  AFFX-HUMISGF3A/M97935_MB_at             STAT1               6772\n",
       "\n",
       "[44134 rows x 3 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Obatian gene information table: extract Identifier and Gene symbol with gene id\n",
    "metadata = pd.DataFrame(gpl.table, columns=['ID', 'Gene Symbol', 'ENTREZ_GENE_ID'])\n",
    "metadata.dropna(inplace=True)\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>GSM258558</th>\n",
       "      <th>GSM258559</th>\n",
       "      <th>GSM258560</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID_REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1007_s_at</th>\n",
       "      <td>9.129905</td>\n",
       "      <td>9.843349</td>\n",
       "      <td>9.730661</td>\n",
       "      <td>9.032165</td>\n",
       "      <td>10.281793</td>\n",
       "      <td>9.154551</td>\n",
       "      <td>8.063788</td>\n",
       "      <td>8.881805</td>\n",
       "      <td>9.039005</td>\n",
       "      <td>10.643545</td>\n",
       "      <td>...</td>\n",
       "      <td>9.628212</td>\n",
       "      <td>10.130757</td>\n",
       "      <td>9.276419</td>\n",
       "      <td>9.296957</td>\n",
       "      <td>10.346701</td>\n",
       "      <td>9.709068</td>\n",
       "      <td>7.672985</td>\n",
       "      <td>9.199080</td>\n",
       "      <td>7.635765</td>\n",
       "      <td>8.760082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1053_at</th>\n",
       "      <td>8.034022</td>\n",
       "      <td>7.973332</td>\n",
       "      <td>8.834045</td>\n",
       "      <td>7.723965</td>\n",
       "      <td>9.040800</td>\n",
       "      <td>8.792375</td>\n",
       "      <td>8.093354</td>\n",
       "      <td>7.575620</td>\n",
       "      <td>8.438594</td>\n",
       "      <td>9.142260</td>\n",
       "      <td>...</td>\n",
       "      <td>9.215567</td>\n",
       "      <td>8.273698</td>\n",
       "      <td>8.790710</td>\n",
       "      <td>8.861625</td>\n",
       "      <td>8.512193</td>\n",
       "      <td>7.908581</td>\n",
       "      <td>9.299454</td>\n",
       "      <td>8.220959</td>\n",
       "      <td>7.432135</td>\n",
       "      <td>7.596431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117_at</th>\n",
       "      <td>3.564520</td>\n",
       "      <td>4.994852</td>\n",
       "      <td>5.066018</td>\n",
       "      <td>4.958580</td>\n",
       "      <td>4.951835</td>\n",
       "      <td>6.428274</td>\n",
       "      <td>7.364676</td>\n",
       "      <td>4.659326</td>\n",
       "      <td>6.643723</td>\n",
       "      <td>5.368356</td>\n",
       "      <td>...</td>\n",
       "      <td>5.183919</td>\n",
       "      <td>4.801579</td>\n",
       "      <td>6.161777</td>\n",
       "      <td>5.443126</td>\n",
       "      <td>4.609810</td>\n",
       "      <td>4.339004</td>\n",
       "      <td>6.384404</td>\n",
       "      <td>6.006534</td>\n",
       "      <td>5.484470</td>\n",
       "      <td>5.372583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121_at</th>\n",
       "      <td>4.746490</td>\n",
       "      <td>5.197306</td>\n",
       "      <td>5.234618</td>\n",
       "      <td>6.078180</td>\n",
       "      <td>5.205632</td>\n",
       "      <td>5.009848</td>\n",
       "      <td>5.533000</td>\n",
       "      <td>5.362637</td>\n",
       "      <td>5.210988</td>\n",
       "      <td>5.195820</td>\n",
       "      <td>...</td>\n",
       "      <td>5.202508</td>\n",
       "      <td>4.807654</td>\n",
       "      <td>5.326347</td>\n",
       "      <td>5.205004</td>\n",
       "      <td>5.201123</td>\n",
       "      <td>6.028717</td>\n",
       "      <td>5.220915</td>\n",
       "      <td>5.186452</td>\n",
       "      <td>5.220778</td>\n",
       "      <td>5.097535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1255_g_at</th>\n",
       "      <td>2.320698</td>\n",
       "      <td>2.248520</td>\n",
       "      <td>2.259504</td>\n",
       "      <td>2.262787</td>\n",
       "      <td>2.207531</td>\n",
       "      <td>2.322618</td>\n",
       "      <td>2.656929</td>\n",
       "      <td>2.255955</td>\n",
       "      <td>2.266399</td>\n",
       "      <td>2.246563</td>\n",
       "      <td>...</td>\n",
       "      <td>2.248392</td>\n",
       "      <td>2.232021</td>\n",
       "      <td>2.246738</td>\n",
       "      <td>2.245739</td>\n",
       "      <td>2.265454</td>\n",
       "      <td>2.301397</td>\n",
       "      <td>2.322604</td>\n",
       "      <td>2.236494</td>\n",
       "      <td>2.243698</td>\n",
       "      <td>2.420731</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "name       GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  GSM258556  \\\n",
       "ID_REF                                                                        \n",
       "1007_s_at   9.129905   9.843349   9.730661   9.032165  10.281793   9.154551   \n",
       "1053_at     8.034022   7.973332   8.834045   7.723965   9.040800   8.792375   \n",
       "117_at      3.564520   4.994852   5.066018   4.958580   4.951835   6.428274   \n",
       "121_at      4.746490   5.197306   5.234618   6.078180   5.205632   5.009848   \n",
       "1255_g_at   2.320698   2.248520   2.259504   2.262787   2.207531   2.322618   \n",
       "\n",
       "name       GSM258557  GSM258558  GSM258559  GSM258560  ...  GSM258599  \\\n",
       "ID_REF                                                 ...              \n",
       "1007_s_at   8.063788   8.881805   9.039005  10.643545  ...   9.628212   \n",
       "1053_at     8.093354   7.575620   8.438594   9.142260  ...   9.215567   \n",
       "117_at      7.364676   4.659326   6.643723   5.368356  ...   5.183919   \n",
       "121_at      5.533000   5.362637   5.210988   5.195820  ...   5.202508   \n",
       "1255_g_at   2.656929   2.255955   2.266399   2.246563  ...   2.248392   \n",
       "\n",
       "name       GSM258600  GSM258601  GSM258602  GSM258603  GSM258604  GSM258605  \\\n",
       "ID_REF                                                                        \n",
       "1007_s_at  10.130757   9.276419   9.296957  10.346701   9.709068   7.672985   \n",
       "1053_at     8.273698   8.790710   8.861625   8.512193   7.908581   9.299454   \n",
       "117_at      4.801579   6.161777   5.443126   4.609810   4.339004   6.384404   \n",
       "121_at      4.807654   5.326347   5.205004   5.201123   6.028717   5.220915   \n",
       "1255_g_at   2.232021   2.246738   2.245739   2.265454   2.301397   2.322604   \n",
       "\n",
       "name       GSM258606  GSM258607  GSM258608  \n",
       "ID_REF                                      \n",
       "1007_s_at   9.199080   7.635765   8.760082  \n",
       "1053_at     8.220959   7.432135   7.596431  \n",
       "117_at      6.006534   5.484470   5.372583  \n",
       "121_at      5.186452   5.220778   5.097535  \n",
       "1255_g_at   2.236494   2.243698   2.420731  \n",
       "\n",
       "[5 rows x 58 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#extract gene expression matrix from gse, which includes gene expression of 58 samples\n",
    "expr = pd.DataFrame(gse.pivot_samples('VALUE'))\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "      <td>9.129905</td>\n",
       "      <td>9.843349</td>\n",
       "      <td>9.730661</td>\n",
       "      <td>9.032165</td>\n",
       "      <td>10.281793</td>\n",
       "      <td>9.154551</td>\n",
       "      <td>8.063788</td>\n",
       "      <td>...</td>\n",
       "      <td>9.628212</td>\n",
       "      <td>10.130757</td>\n",
       "      <td>9.276419</td>\n",
       "      <td>9.296957</td>\n",
       "      <td>10.346701</td>\n",
       "      <td>9.709068</td>\n",
       "      <td>7.672985</td>\n",
       "      <td>9.199080</td>\n",
       "      <td>7.635765</td>\n",
       "      <td>8.760082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "      <td>8.034022</td>\n",
       "      <td>7.973332</td>\n",
       "      <td>8.834045</td>\n",
       "      <td>7.723965</td>\n",
       "      <td>9.040800</td>\n",
       "      <td>8.792375</td>\n",
       "      <td>8.093354</td>\n",
       "      <td>...</td>\n",
       "      <td>9.215567</td>\n",
       "      <td>8.273698</td>\n",
       "      <td>8.790710</td>\n",
       "      <td>8.861625</td>\n",
       "      <td>8.512193</td>\n",
       "      <td>7.908581</td>\n",
       "      <td>9.299454</td>\n",
       "      <td>8.220959</td>\n",
       "      <td>7.432135</td>\n",
       "      <td>7.596431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "      <td>3.564520</td>\n",
       "      <td>4.994852</td>\n",
       "      <td>5.066018</td>\n",
       "      <td>4.958580</td>\n",
       "      <td>4.951835</td>\n",
       "      <td>6.428274</td>\n",
       "      <td>7.364676</td>\n",
       "      <td>...</td>\n",
       "      <td>5.183919</td>\n",
       "      <td>4.801579</td>\n",
       "      <td>6.161777</td>\n",
       "      <td>5.443126</td>\n",
       "      <td>4.609810</td>\n",
       "      <td>4.339004</td>\n",
       "      <td>6.384404</td>\n",
       "      <td>6.006534</td>\n",
       "      <td>5.484470</td>\n",
       "      <td>5.372583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "      <td>4.746490</td>\n",
       "      <td>5.197306</td>\n",
       "      <td>5.234618</td>\n",
       "      <td>6.078180</td>\n",
       "      <td>5.205632</td>\n",
       "      <td>5.009848</td>\n",
       "      <td>5.533000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.202508</td>\n",
       "      <td>4.807654</td>\n",
       "      <td>5.326347</td>\n",
       "      <td>5.205004</td>\n",
       "      <td>5.201123</td>\n",
       "      <td>6.028717</td>\n",
       "      <td>5.220915</td>\n",
       "      <td>5.186452</td>\n",
       "      <td>5.220778</td>\n",
       "      <td>5.097535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "      <td>2.320698</td>\n",
       "      <td>2.248520</td>\n",
       "      <td>2.259504</td>\n",
       "      <td>2.262787</td>\n",
       "      <td>2.207531</td>\n",
       "      <td>2.322618</td>\n",
       "      <td>2.656929</td>\n",
       "      <td>...</td>\n",
       "      <td>2.248392</td>\n",
       "      <td>2.232021</td>\n",
       "      <td>2.246738</td>\n",
       "      <td>2.245739</td>\n",
       "      <td>2.265454</td>\n",
       "      <td>2.301397</td>\n",
       "      <td>2.322604</td>\n",
       "      <td>2.236494</td>\n",
       "      <td>2.243698</td>\n",
       "      <td>2.420731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44129</th>\n",
       "      <td>AFFX-HUMGAPDH/M33197_M_at</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>2597</td>\n",
       "      <td>13.302241</td>\n",
       "      <td>13.668732</td>\n",
       "      <td>13.779665</td>\n",
       "      <td>13.940228</td>\n",
       "      <td>14.011925</td>\n",
       "      <td>13.897492</td>\n",
       "      <td>14.083509</td>\n",
       "      <td>...</td>\n",
       "      <td>14.306833</td>\n",
       "      <td>13.721243</td>\n",
       "      <td>13.917724</td>\n",
       "      <td>13.730772</td>\n",
       "      <td>14.246234</td>\n",
       "      <td>13.940670</td>\n",
       "      <td>14.620784</td>\n",
       "      <td>13.814232</td>\n",
       "      <td>13.891675</td>\n",
       "      <td>14.212837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44130</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_3_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "      <td>10.865934</td>\n",
       "      <td>10.968102</td>\n",
       "      <td>11.918820</td>\n",
       "      <td>10.131834</td>\n",
       "      <td>10.146164</td>\n",
       "      <td>10.548552</td>\n",
       "      <td>11.078523</td>\n",
       "      <td>...</td>\n",
       "      <td>10.998265</td>\n",
       "      <td>12.282170</td>\n",
       "      <td>10.970773</td>\n",
       "      <td>11.045425</td>\n",
       "      <td>11.204857</td>\n",
       "      <td>10.952704</td>\n",
       "      <td>9.970544</td>\n",
       "      <td>11.571419</td>\n",
       "      <td>9.657875</td>\n",
       "      <td>10.392547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44131</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_5_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "      <td>5.863362</td>\n",
       "      <td>5.518090</td>\n",
       "      <td>7.942593</td>\n",
       "      <td>5.190737</td>\n",
       "      <td>5.338598</td>\n",
       "      <td>6.796106</td>\n",
       "      <td>7.495530</td>\n",
       "      <td>...</td>\n",
       "      <td>6.752478</td>\n",
       "      <td>8.297562</td>\n",
       "      <td>5.688170</td>\n",
       "      <td>7.387534</td>\n",
       "      <td>7.615553</td>\n",
       "      <td>5.866731</td>\n",
       "      <td>6.216406</td>\n",
       "      <td>7.658399</td>\n",
       "      <td>5.812330</td>\n",
       "      <td>5.069783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44132</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MA_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "      <td>8.106523</td>\n",
       "      <td>7.939480</td>\n",
       "      <td>9.870743</td>\n",
       "      <td>7.147090</td>\n",
       "      <td>7.704300</td>\n",
       "      <td>8.048070</td>\n",
       "      <td>9.600885</td>\n",
       "      <td>...</td>\n",
       "      <td>8.721723</td>\n",
       "      <td>10.428576</td>\n",
       "      <td>7.935277</td>\n",
       "      <td>9.461015</td>\n",
       "      <td>9.341084</td>\n",
       "      <td>7.936459</td>\n",
       "      <td>8.167706</td>\n",
       "      <td>9.306615</td>\n",
       "      <td>7.369962</td>\n",
       "      <td>6.955356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44133</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MB_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "      <td>8.894156</td>\n",
       "      <td>8.942695</td>\n",
       "      <td>10.625638</td>\n",
       "      <td>7.941765</td>\n",
       "      <td>8.609167</td>\n",
       "      <td>8.589324</td>\n",
       "      <td>9.853876</td>\n",
       "      <td>...</td>\n",
       "      <td>9.437128</td>\n",
       "      <td>10.930077</td>\n",
       "      <td>9.154707</td>\n",
       "      <td>9.687778</td>\n",
       "      <td>9.955542</td>\n",
       "      <td>8.833349</td>\n",
       "      <td>8.491898</td>\n",
       "      <td>10.047244</td>\n",
       "      <td>8.014284</td>\n",
       "      <td>8.257933</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44134 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                ID       Gene Symbol     ENTREZ_GENE_ID  \\\n",
       "0                        1007_s_at  DDR1 /// MIR4640  780 /// 100616237   \n",
       "1                          1053_at              RFC2               5982   \n",
       "2                           117_at             HSPA6               3310   \n",
       "3                           121_at              PAX8               7849   \n",
       "4                        1255_g_at            GUCA1A               2978   \n",
       "...                            ...               ...                ...   \n",
       "44129    AFFX-HUMGAPDH/M33197_M_at             GAPDH               2597   \n",
       "44130   AFFX-HUMISGF3A/M97935_3_at             STAT1               6772   \n",
       "44131   AFFX-HUMISGF3A/M97935_5_at             STAT1               6772   \n",
       "44132  AFFX-HUMISGF3A/M97935_MA_at             STAT1               6772   \n",
       "44133  AFFX-HUMISGF3A/M97935_MB_at             STAT1               6772   \n",
       "\n",
       "       GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  GSM258556  \\\n",
       "0       9.129905   9.843349   9.730661   9.032165  10.281793   9.154551   \n",
       "1       8.034022   7.973332   8.834045   7.723965   9.040800   8.792375   \n",
       "2       3.564520   4.994852   5.066018   4.958580   4.951835   6.428274   \n",
       "3       4.746490   5.197306   5.234618   6.078180   5.205632   5.009848   \n",
       "4       2.320698   2.248520   2.259504   2.262787   2.207531   2.322618   \n",
       "...          ...        ...        ...        ...        ...        ...   \n",
       "44129  13.302241  13.668732  13.779665  13.940228  14.011925  13.897492   \n",
       "44130  10.865934  10.968102  11.918820  10.131834  10.146164  10.548552   \n",
       "44131   5.863362   5.518090   7.942593   5.190737   5.338598   6.796106   \n",
       "44132   8.106523   7.939480   9.870743   7.147090   7.704300   8.048070   \n",
       "44133   8.894156   8.942695  10.625638   7.941765   8.609167   8.589324   \n",
       "\n",
       "       GSM258557  ...  GSM258599  GSM258600  GSM258601  GSM258602  GSM258603  \\\n",
       "0       8.063788  ...   9.628212  10.130757   9.276419   9.296957  10.346701   \n",
       "1       8.093354  ...   9.215567   8.273698   8.790710   8.861625   8.512193   \n",
       "2       7.364676  ...   5.183919   4.801579   6.161777   5.443126   4.609810   \n",
       "3       5.533000  ...   5.202508   4.807654   5.326347   5.205004   5.201123   \n",
       "4       2.656929  ...   2.248392   2.232021   2.246738   2.245739   2.265454   \n",
       "...          ...  ...        ...        ...        ...        ...        ...   \n",
       "44129  14.083509  ...  14.306833  13.721243  13.917724  13.730772  14.246234   \n",
       "44130  11.078523  ...  10.998265  12.282170  10.970773  11.045425  11.204857   \n",
       "44131   7.495530  ...   6.752478   8.297562   5.688170   7.387534   7.615553   \n",
       "44132   9.600885  ...   8.721723  10.428576   7.935277   9.461015   9.341084   \n",
       "44133   9.853876  ...   9.437128  10.930077   9.154707   9.687778   9.955542   \n",
       "\n",
       "       GSM258604  GSM258605  GSM258606  GSM258607  GSM258608  \n",
       "0       9.709068   7.672985   9.199080   7.635765   8.760082  \n",
       "1       7.908581   9.299454   8.220959   7.432135   7.596431  \n",
       "2       4.339004   6.384404   6.006534   5.484470   5.372583  \n",
       "3       6.028717   5.220915   5.186452   5.220778   5.097535  \n",
       "4       2.301397   2.322604   2.236494   2.243698   2.420731  \n",
       "...          ...        ...        ...        ...        ...  \n",
       "44129  13.940670  14.620784  13.814232  13.891675  14.212837  \n",
       "44130  10.952704   9.970544  11.571419   9.657875  10.392547  \n",
       "44131   5.866731   6.216406   7.658399   5.812330   5.069783  \n",
       "44132   7.936459   8.167706   9.306615   7.369962   6.955356  \n",
       "44133   8.833349   8.491898  10.047244   8.014284   8.257933  \n",
       "\n",
       "[44134 rows x 61 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#merge expression matrix with gene information table\n",
    "df = pd.merge(metadata,expr, left_on= \"ID\", right_on=\"ID_REF\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "can only concatenate str (not \"float\") to str",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[35], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m#each gene may have mutiple ID, only keep the ID with the highest gene expression. \u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39m#Using mean gene expression value to sort\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m df[\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m] \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39mmean(axis\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m      5\u001b[0m df \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39msort_values(by\u001b[39m=\u001b[39m[\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m],ascending\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m)\n\u001b[1;32m      6\u001b[0m \u001b[39m#drop duplicated gene symbol, keep the first row of one gene symbol\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/frame.py:11338\u001b[0m, in \u001b[0;36mDataFrame.mean\u001b[0;34m(self, axis, skipna, numeric_only, **kwargs)\u001b[0m\n\u001b[1;32m  11330\u001b[0m \u001b[39m@doc\u001b[39m(make_doc(\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m, ndim\u001b[39m=\u001b[39m\u001b[39m2\u001b[39m))\n\u001b[1;32m  11331\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mmean\u001b[39m(\n\u001b[1;32m  11332\u001b[0m     \u001b[39mself\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m  11336\u001b[0m     \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs,\n\u001b[1;32m  11337\u001b[0m ):\n\u001b[0;32m> 11338\u001b[0m     result \u001b[39m=\u001b[39m \u001b[39msuper\u001b[39m()\u001b[39m.\u001b[39mmean(axis, skipna, numeric_only, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[1;32m  11339\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(result, Series):\n\u001b[1;32m  11340\u001b[0m         result \u001b[39m=\u001b[39m result\u001b[39m.\u001b[39m__finalize__(\u001b[39mself\u001b[39m, method\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/generic.py:11978\u001b[0m, in \u001b[0;36mNDFrame.mean\u001b[0;34m(self, axis, skipna, numeric_only, **kwargs)\u001b[0m\n\u001b[1;32m  11971\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mmean\u001b[39m(\n\u001b[1;32m  11972\u001b[0m     \u001b[39mself\u001b[39m,\n\u001b[1;32m  11973\u001b[0m     axis: Axis \u001b[39m|\u001b[39m \u001b[39mNone\u001b[39;00m \u001b[39m=\u001b[39m \u001b[39m0\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m  11976\u001b[0m     \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs,\n\u001b[1;32m  11977\u001b[0m ) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m Series \u001b[39m|\u001b[39m \u001b[39mfloat\u001b[39m:\n\u001b[0;32m> 11978\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_stat_function(\n\u001b[1;32m  11979\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m, nanops\u001b[39m.\u001b[39mnanmean, axis, skipna, numeric_only, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs\n\u001b[1;32m  11980\u001b[0m     )\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/generic.py:11935\u001b[0m, in \u001b[0;36mNDFrame._stat_function\u001b[0;34m(self, name, func, axis, skipna, numeric_only, **kwargs)\u001b[0m\n\u001b[1;32m  11931\u001b[0m nv\u001b[39m.\u001b[39mvalidate_func(name, (), kwargs)\n\u001b[1;32m  11933\u001b[0m validate_bool_kwarg(skipna, \u001b[39m\"\u001b[39m\u001b[39mskipna\u001b[39m\u001b[39m\"\u001b[39m, none_allowed\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m)\n\u001b[0;32m> 11935\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_reduce(\n\u001b[1;32m  11936\u001b[0m     func, name\u001b[39m=\u001b[39mname, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, numeric_only\u001b[39m=\u001b[39mnumeric_only\n\u001b[1;32m  11937\u001b[0m )\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/frame.py:11207\u001b[0m, in \u001b[0;36mDataFrame._reduce\u001b[0;34m(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)\u001b[0m\n\u001b[1;32m  11203\u001b[0m     df \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39mT\n\u001b[1;32m  11205\u001b[0m \u001b[39m# After possibly _get_data and transposing, we are now in the\u001b[39;00m\n\u001b[1;32m  11206\u001b[0m \u001b[39m#  simple case where we can use BlockManager.reduce\u001b[39;00m\n\u001b[0;32m> 11207\u001b[0m res \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39m_mgr\u001b[39m.\u001b[39mreduce(blk_func)\n\u001b[1;32m  11208\u001b[0m out \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39m_constructor_from_mgr(res, axes\u001b[39m=\u001b[39mres\u001b[39m.\u001b[39maxes)\u001b[39m.\u001b[39miloc[\u001b[39m0\u001b[39m]\n\u001b[1;32m  11209\u001b[0m \u001b[39mif\u001b[39;00m out_dtype \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m \u001b[39mand\u001b[39;00m out\u001b[39m.\u001b[39mdtype \u001b[39m!=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mboolean\u001b[39m\u001b[39m\"\u001b[39m:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/internals/managers.py:1459\u001b[0m, in \u001b[0;36mBlockManager.reduce\u001b[0;34m(self, func)\u001b[0m\n\u001b[1;32m   1457\u001b[0m res_blocks: \u001b[39mlist\u001b[39m[Block] \u001b[39m=\u001b[39m []\n\u001b[1;32m   1458\u001b[0m \u001b[39mfor\u001b[39;00m blk \u001b[39min\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mblocks:\n\u001b[0;32m-> 1459\u001b[0m     nbs \u001b[39m=\u001b[39m blk\u001b[39m.\u001b[39mreduce(func)\n\u001b[1;32m   1460\u001b[0m     res_blocks\u001b[39m.\u001b[39mextend(nbs)\n\u001b[1;32m   1462\u001b[0m index \u001b[39m=\u001b[39m Index([\u001b[39mNone\u001b[39;00m])  \u001b[39m# placeholder\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/internals/blocks.py:377\u001b[0m, in \u001b[0;36mBlock.reduce\u001b[0;34m(self, func)\u001b[0m\n\u001b[1;32m    371\u001b[0m \u001b[39m@final\u001b[39m\n\u001b[1;32m    372\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mreduce\u001b[39m(\u001b[39mself\u001b[39m, func) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m \u001b[39mlist\u001b[39m[Block]:\n\u001b[1;32m    373\u001b[0m     \u001b[39m# We will apply the function and reshape the result into a single-row\u001b[39;00m\n\u001b[1;32m    374\u001b[0m     \u001b[39m#  Block with the same mgr_locs; squeezing will be done at a higher level\u001b[39;00m\n\u001b[1;32m    375\u001b[0m     \u001b[39massert\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mndim \u001b[39m==\u001b[39m \u001b[39m2\u001b[39m\n\u001b[0;32m--> 377\u001b[0m     result \u001b[39m=\u001b[39m func(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mvalues)\n\u001b[1;32m    379\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mvalues\u001b[39m.\u001b[39mndim \u001b[39m==\u001b[39m \u001b[39m1\u001b[39m:\n\u001b[1;32m    380\u001b[0m         res_values \u001b[39m=\u001b[39m result\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/frame.py:11139\u001b[0m, in \u001b[0;36mDataFrame._reduce.<locals>.blk_func\u001b[0;34m(values, axis)\u001b[0m\n\u001b[1;32m  11137\u001b[0m         \u001b[39mreturn\u001b[39;00m np\u001b[39m.\u001b[39marray([result])\n\u001b[1;32m  11138\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m> 11139\u001b[0m     \u001b[39mreturn\u001b[39;00m op(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwds)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/nanops.py:147\u001b[0m, in \u001b[0;36mbottleneck_switch.__call__.<locals>.f\u001b[0;34m(values, axis, skipna, **kwds)\u001b[0m\n\u001b[1;32m    145\u001b[0m         result \u001b[39m=\u001b[39m alt(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwds)\n\u001b[1;32m    146\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m--> 147\u001b[0m     result \u001b[39m=\u001b[39m alt(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwds)\n\u001b[1;32m    149\u001b[0m \u001b[39mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/nanops.py:404\u001b[0m, in \u001b[0;36m_datetimelike_compat.<locals>.new_func\u001b[0;34m(values, axis, skipna, mask, **kwargs)\u001b[0m\n\u001b[1;32m    401\u001b[0m \u001b[39mif\u001b[39;00m datetimelike \u001b[39mand\u001b[39;00m mask \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m    402\u001b[0m     mask \u001b[39m=\u001b[39m isna(values)\n\u001b[0;32m--> 404\u001b[0m result \u001b[39m=\u001b[39m func(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, mask\u001b[39m=\u001b[39mmask, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[1;32m    406\u001b[0m \u001b[39mif\u001b[39;00m datetimelike:\n\u001b[1;32m    407\u001b[0m     result \u001b[39m=\u001b[39m _wrap_results(result, orig_values\u001b[39m.\u001b[39mdtype, fill_value\u001b[39m=\u001b[39miNaT)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/nanops.py:719\u001b[0m, in \u001b[0;36mnanmean\u001b[0;34m(values, axis, skipna, mask)\u001b[0m\n\u001b[1;32m    716\u001b[0m     dtype_count \u001b[39m=\u001b[39m dtype\n\u001b[1;32m    718\u001b[0m count \u001b[39m=\u001b[39m _get_counts(values\u001b[39m.\u001b[39mshape, mask, axis, dtype\u001b[39m=\u001b[39mdtype_count)\n\u001b[0;32m--> 719\u001b[0m the_sum \u001b[39m=\u001b[39m values\u001b[39m.\u001b[39msum(axis, dtype\u001b[39m=\u001b[39mdtype_sum)\n\u001b[1;32m    720\u001b[0m the_sum \u001b[39m=\u001b[39m _ensure_numeric(the_sum)\n\u001b[1;32m    722\u001b[0m \u001b[39mif\u001b[39;00m axis \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m \u001b[39mand\u001b[39;00m \u001b[39mgetattr\u001b[39m(the_sum, \u001b[39m\"\u001b[39m\u001b[39mndim\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/numpy/core/_methods.py:49\u001b[0m, in \u001b[0;36m_sum\u001b[0;34m(a, axis, dtype, out, keepdims, initial, where)\u001b[0m\n\u001b[1;32m     47\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_sum\u001b[39m(a, axis\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, dtype\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, out\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, keepdims\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m,\n\u001b[1;32m     48\u001b[0m          initial\u001b[39m=\u001b[39m_NoValue, where\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m):\n\u001b[0;32m---> 49\u001b[0m     \u001b[39mreturn\u001b[39;00m umr_sum(a, axis, dtype, out, keepdims, initial, where)\n",
      "\u001b[0;31mTypeError\u001b[0m: can only concatenate str (not \"float\") to str"
     ]
    }
   ],
   "source": [
    "#each gene may have mutiple ID, only keep the ID with the highest gene expression. \n",
    "#Using mean gene expression value to sort\n",
    "\n",
    "df[\"mean\"] = df.mean(axis=1)\n",
    "df = df.sort_values(by=[\"mean\"],ascending=False)\n",
    "#drop duplicated gene symbol, keep the first row of one gene symbol\n",
    "df = df.drop_duplicates(\"Gene Symbol\")\n",
    "#only keep gene symbol and gene expression value of 58 samples\n",
    "df = df.drop(\"mean\",axis=1)\n",
    "df = df.drop(\"ID\",axis=1)\n",
    "df = df.drop(\"ENTREZ_GENE_ID\",axis=1)\n",
    "#set Gene Symbol name as index\n",
    "df = df.set_index([\"Gene Symbol\"])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>44124</th>\n",
       "      <th>44125</th>\n",
       "      <th>44126</th>\n",
       "      <th>44127</th>\n",
       "      <th>44128</th>\n",
       "      <th>44129</th>\n",
       "      <th>44130</th>\n",
       "      <th>44131</th>\n",
       "      <th>44132</th>\n",
       "      <th>44133</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>1053_at</td>\n",
       "      <td>117_at</td>\n",
       "      <td>121_at</td>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>1294_at</td>\n",
       "      <td>1316_at</td>\n",
       "      <td>1320_at</td>\n",
       "      <td>1405_i_at</td>\n",
       "      <td>1431_at</td>\n",
       "      <td>...</td>\n",
       "      <td>AFFX-HSAC07/X00351_3_at</td>\n",
       "      <td>AFFX-HSAC07/X00351_5_at</td>\n",
       "      <td>AFFX-HSAC07/X00351_M_at</td>\n",
       "      <td>AFFX-HUMGAPDH/M33197_3_at</td>\n",
       "      <td>AFFX-HUMGAPDH/M33197_5_at</td>\n",
       "      <td>AFFX-HUMGAPDH/M33197_M_at</td>\n",
       "      <td>AFFX-HUMISGF3A/M97935_3_at</td>\n",
       "      <td>AFFX-HUMISGF3A/M97935_5_at</td>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MA_at</td>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MB_at</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Symbol</th>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>MIR5193 /// UBA7</td>\n",
       "      <td>THRA</td>\n",
       "      <td>PTPN21</td>\n",
       "      <td>CCL5</td>\n",
       "      <td>CYP2E1</td>\n",
       "      <td>...</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>ACTB</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>STAT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <td>780 /// 100616237</td>\n",
       "      <td>5982</td>\n",
       "      <td>3310</td>\n",
       "      <td>7849</td>\n",
       "      <td>2978</td>\n",
       "      <td>7318 /// 100847079</td>\n",
       "      <td>7067</td>\n",
       "      <td>11099</td>\n",
       "      <td>6352</td>\n",
       "      <td>1571</td>\n",
       "      <td>...</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>60</td>\n",
       "      <td>2597</td>\n",
       "      <td>2597</td>\n",
       "      <td>2597</td>\n",
       "      <td>6772</td>\n",
       "      <td>6772</td>\n",
       "      <td>6772</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258551</th>\n",
       "      <td>9.129905</td>\n",
       "      <td>8.034022</td>\n",
       "      <td>3.56452</td>\n",
       "      <td>4.74649</td>\n",
       "      <td>2.320698</td>\n",
       "      <td>5.519153</td>\n",
       "      <td>3.339182</td>\n",
       "      <td>2.775395</td>\n",
       "      <td>8.303437</td>\n",
       "      <td>2.981327</td>\n",
       "      <td>...</td>\n",
       "      <td>14.540047</td>\n",
       "      <td>12.873186</td>\n",
       "      <td>13.361601</td>\n",
       "      <td>13.605885</td>\n",
       "      <td>13.139939</td>\n",
       "      <td>13.302241</td>\n",
       "      <td>10.865934</td>\n",
       "      <td>5.863362</td>\n",
       "      <td>8.106523</td>\n",
       "      <td>8.894156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258552</th>\n",
       "      <td>9.843349</td>\n",
       "      <td>7.973332</td>\n",
       "      <td>4.994852</td>\n",
       "      <td>5.197306</td>\n",
       "      <td>2.24852</td>\n",
       "      <td>5.081258</td>\n",
       "      <td>2.934516</td>\n",
       "      <td>2.617097</td>\n",
       "      <td>9.145519</td>\n",
       "      <td>4.619668</td>\n",
       "      <td>...</td>\n",
       "      <td>14.678135</td>\n",
       "      <td>12.877637</td>\n",
       "      <td>13.317487</td>\n",
       "      <td>13.947742</td>\n",
       "      <td>13.621443</td>\n",
       "      <td>13.668732</td>\n",
       "      <td>10.968102</td>\n",
       "      <td>5.51809</td>\n",
       "      <td>7.93948</td>\n",
       "      <td>8.942695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258604</th>\n",
       "      <td>9.709068</td>\n",
       "      <td>7.908581</td>\n",
       "      <td>4.339004</td>\n",
       "      <td>6.028717</td>\n",
       "      <td>2.301397</td>\n",
       "      <td>4.971613</td>\n",
       "      <td>2.949082</td>\n",
       "      <td>2.628383</td>\n",
       "      <td>5.70847</td>\n",
       "      <td>2.827657</td>\n",
       "      <td>...</td>\n",
       "      <td>14.531086</td>\n",
       "      <td>12.224497</td>\n",
       "      <td>12.79762</td>\n",
       "      <td>14.138936</td>\n",
       "      <td>13.888978</td>\n",
       "      <td>13.94067</td>\n",
       "      <td>10.952704</td>\n",
       "      <td>5.866731</td>\n",
       "      <td>7.936459</td>\n",
       "      <td>8.833349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258605</th>\n",
       "      <td>7.672985</td>\n",
       "      <td>9.299454</td>\n",
       "      <td>6.384404</td>\n",
       "      <td>5.220915</td>\n",
       "      <td>2.322604</td>\n",
       "      <td>3.611807</td>\n",
       "      <td>2.999993</td>\n",
       "      <td>3.26752</td>\n",
       "      <td>7.803824</td>\n",
       "      <td>2.750096</td>\n",
       "      <td>...</td>\n",
       "      <td>14.419252</td>\n",
       "      <td>13.419198</td>\n",
       "      <td>13.633222</td>\n",
       "      <td>14.574913</td>\n",
       "      <td>14.445077</td>\n",
       "      <td>14.620784</td>\n",
       "      <td>9.970544</td>\n",
       "      <td>6.216406</td>\n",
       "      <td>8.167706</td>\n",
       "      <td>8.491898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258606</th>\n",
       "      <td>9.19908</td>\n",
       "      <td>8.220959</td>\n",
       "      <td>6.006534</td>\n",
       "      <td>5.186452</td>\n",
       "      <td>2.236494</td>\n",
       "      <td>5.169612</td>\n",
       "      <td>2.936686</td>\n",
       "      <td>2.613397</td>\n",
       "      <td>8.62551</td>\n",
       "      <td>2.724732</td>\n",
       "      <td>...</td>\n",
       "      <td>14.271686</td>\n",
       "      <td>12.839618</td>\n",
       "      <td>13.107627</td>\n",
       "      <td>13.96402</td>\n",
       "      <td>13.807048</td>\n",
       "      <td>13.814232</td>\n",
       "      <td>11.571419</td>\n",
       "      <td>7.658399</td>\n",
       "      <td>9.306615</td>\n",
       "      <td>10.047244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258607</th>\n",
       "      <td>7.635765</td>\n",
       "      <td>7.432135</td>\n",
       "      <td>5.48447</td>\n",
       "      <td>5.220778</td>\n",
       "      <td>2.243698</td>\n",
       "      <td>3.782184</td>\n",
       "      <td>2.98487</td>\n",
       "      <td>2.622019</td>\n",
       "      <td>7.554318</td>\n",
       "      <td>2.745644</td>\n",
       "      <td>...</td>\n",
       "      <td>14.362165</td>\n",
       "      <td>12.911146</td>\n",
       "      <td>13.209227</td>\n",
       "      <td>14.008279</td>\n",
       "      <td>13.789795</td>\n",
       "      <td>13.891675</td>\n",
       "      <td>9.657875</td>\n",
       "      <td>5.81233</td>\n",
       "      <td>7.369962</td>\n",
       "      <td>8.014284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258608</th>\n",
       "      <td>8.760082</td>\n",
       "      <td>7.596431</td>\n",
       "      <td>5.372583</td>\n",
       "      <td>5.097535</td>\n",
       "      <td>2.420731</td>\n",
       "      <td>4.841838</td>\n",
       "      <td>3.192557</td>\n",
       "      <td>2.625</td>\n",
       "      <td>8.341726</td>\n",
       "      <td>2.751262</td>\n",
       "      <td>...</td>\n",
       "      <td>14.39871</td>\n",
       "      <td>12.541851</td>\n",
       "      <td>13.017797</td>\n",
       "      <td>14.313185</td>\n",
       "      <td>14.100626</td>\n",
       "      <td>14.212837</td>\n",
       "      <td>10.392547</td>\n",
       "      <td>5.069783</td>\n",
       "      <td>6.955356</td>\n",
       "      <td>8.257933</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>61 rows × 44134 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            0         1         2         3          4      \\\n",
       "ID                      1007_s_at   1053_at    117_at    121_at  1255_g_at   \n",
       "Gene Symbol      DDR1 /// MIR4640      RFC2     HSPA6      PAX8     GUCA1A   \n",
       "ENTREZ_GENE_ID  780 /// 100616237      5982      3310      7849       2978   \n",
       "GSM258551                9.129905  8.034022   3.56452   4.74649   2.320698   \n",
       "GSM258552                9.843349  7.973332  4.994852  5.197306    2.24852   \n",
       "...                           ...       ...       ...       ...        ...   \n",
       "GSM258604                9.709068  7.908581  4.339004  6.028717   2.301397   \n",
       "GSM258605                7.672985  9.299454  6.384404  5.220915   2.322604   \n",
       "GSM258606                 9.19908  8.220959  6.006534  5.186452   2.236494   \n",
       "GSM258607                7.635765  7.432135   5.48447  5.220778   2.243698   \n",
       "GSM258608                8.760082  7.596431  5.372583  5.097535   2.420731   \n",
       "\n",
       "                             5         6         7          8         9      \\\n",
       "ID                         1294_at   1316_at   1320_at  1405_i_at   1431_at   \n",
       "Gene Symbol       MIR5193 /// UBA7      THRA    PTPN21       CCL5    CYP2E1   \n",
       "ENTREZ_GENE_ID  7318 /// 100847079      7067     11099       6352      1571   \n",
       "GSM258551                 5.519153  3.339182  2.775395   8.303437  2.981327   \n",
       "GSM258552                 5.081258  2.934516  2.617097   9.145519  4.619668   \n",
       "...                            ...       ...       ...        ...       ...   \n",
       "GSM258604                 4.971613  2.949082  2.628383    5.70847  2.827657   \n",
       "GSM258605                 3.611807  2.999993   3.26752   7.803824  2.750096   \n",
       "GSM258606                 5.169612  2.936686  2.613397    8.62551  2.724732   \n",
       "GSM258607                 3.782184   2.98487  2.622019   7.554318  2.745644   \n",
       "GSM258608                 4.841838  3.192557     2.625   8.341726  2.751262   \n",
       "\n",
       "                ...                    44124                    44125  \\\n",
       "ID              ...  AFFX-HSAC07/X00351_3_at  AFFX-HSAC07/X00351_5_at   \n",
       "Gene Symbol     ...                     ACTB                     ACTB   \n",
       "ENTREZ_GENE_ID  ...                       60                       60   \n",
       "GSM258551       ...                14.540047                12.873186   \n",
       "GSM258552       ...                14.678135                12.877637   \n",
       "...             ...                      ...                      ...   \n",
       "GSM258604       ...                14.531086                12.224497   \n",
       "GSM258605       ...                14.419252                13.419198   \n",
       "GSM258606       ...                14.271686                12.839618   \n",
       "GSM258607       ...                14.362165                12.911146   \n",
       "GSM258608       ...                 14.39871                12.541851   \n",
       "\n",
       "                                  44126                      44127  \\\n",
       "ID              AFFX-HSAC07/X00351_M_at  AFFX-HUMGAPDH/M33197_3_at   \n",
       "Gene Symbol                        ACTB                      GAPDH   \n",
       "ENTREZ_GENE_ID                       60                       2597   \n",
       "GSM258551                     13.361601                  13.605885   \n",
       "GSM258552                     13.317487                  13.947742   \n",
       "...                                 ...                        ...   \n",
       "GSM258604                      12.79762                  14.138936   \n",
       "GSM258605                     13.633222                  14.574913   \n",
       "GSM258606                     13.107627                   13.96402   \n",
       "GSM258607                     13.209227                  14.008279   \n",
       "GSM258608                     13.017797                  14.313185   \n",
       "\n",
       "                                    44128                      44129  \\\n",
       "ID              AFFX-HUMGAPDH/M33197_5_at  AFFX-HUMGAPDH/M33197_M_at   \n",
       "Gene Symbol                         GAPDH                      GAPDH   \n",
       "ENTREZ_GENE_ID                       2597                       2597   \n",
       "GSM258551                       13.139939                  13.302241   \n",
       "GSM258552                       13.621443                  13.668732   \n",
       "...                                   ...                        ...   \n",
       "GSM258604                       13.888978                   13.94067   \n",
       "GSM258605                       14.445077                  14.620784   \n",
       "GSM258606                       13.807048                  13.814232   \n",
       "GSM258607                       13.789795                  13.891675   \n",
       "GSM258608                       14.100626                  14.212837   \n",
       "\n",
       "                                     44130                       44131  \\\n",
       "ID              AFFX-HUMISGF3A/M97935_3_at  AFFX-HUMISGF3A/M97935_5_at   \n",
       "Gene Symbol                          STAT1                       STAT1   \n",
       "ENTREZ_GENE_ID                        6772                        6772   \n",
       "GSM258551                        10.865934                    5.863362   \n",
       "GSM258552                        10.968102                     5.51809   \n",
       "...                                    ...                         ...   \n",
       "GSM258604                        10.952704                    5.866731   \n",
       "GSM258605                         9.970544                    6.216406   \n",
       "GSM258606                        11.571419                    7.658399   \n",
       "GSM258607                         9.657875                     5.81233   \n",
       "GSM258608                        10.392547                    5.069783   \n",
       "\n",
       "                                      44132                        44133  \n",
       "ID              AFFX-HUMISGF3A/M97935_MA_at  AFFX-HUMISGF3A/M97935_MB_at  \n",
       "Gene Symbol                           STAT1                        STAT1  \n",
       "ENTREZ_GENE_ID                         6772                         6772  \n",
       "GSM258551                          8.106523                     8.894156  \n",
       "GSM258552                           7.93948                     8.942695  \n",
       "...                                     ...                          ...  \n",
       "GSM258604                          7.936459                     8.833349  \n",
       "GSM258605                          8.167706                     8.491898  \n",
       "GSM258606                          9.306615                    10.047244  \n",
       "GSM258607                          7.369962                     8.014284  \n",
       "GSM258608                          6.955356                     8.257933  \n",
       "\n",
       "[61 rows x 44134 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alldata=df.T\n",
    "alldata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After above data processing, we get the data set of 58 samples gene expression corresponding to each gene symbol."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Q2. Perform clustering using k = 2 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>#sk-container-id-1 {color: black;}#sk-container-id-1 pre{padding: 0;}#sk-container-id-1 div.sk-toggleable {background-color: white;}#sk-container-id-1 label.sk-toggleable__label {cursor: pointer;display: block;width: 100%;margin-bottom: 0;padding: 0.3em;box-sizing: border-box;text-align: center;}#sk-container-id-1 label.sk-toggleable__label-arrow:before {content: \"▸\";float: left;margin-right: 0.25em;color: #696969;}#sk-container-id-1 label.sk-toggleable__label-arrow:hover:before {color: black;}#sk-container-id-1 div.sk-estimator:hover label.sk-toggleable__label-arrow:before {color: black;}#sk-container-id-1 div.sk-toggleable__content {max-height: 0;max-width: 0;overflow: hidden;text-align: left;background-color: #f0f8ff;}#sk-container-id-1 div.sk-toggleable__content pre {margin: 0.2em;color: black;border-radius: 0.25em;background-color: #f0f8ff;}#sk-container-id-1 input.sk-toggleable__control:checked~div.sk-toggleable__content {max-height: 200px;max-width: 100%;overflow: auto;}#sk-container-id-1 input.sk-toggleable__control:checked~label.sk-toggleable__label-arrow:before {content: \"▾\";}#sk-container-id-1 div.sk-estimator input.sk-toggleable__control:checked~label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-1 div.sk-label input.sk-toggleable__control:checked~label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-1 input.sk-hidden--visually {border: 0;clip: rect(1px 1px 1px 1px);clip: rect(1px, 1px, 1px, 1px);height: 1px;margin: -1px;overflow: hidden;padding: 0;position: absolute;width: 1px;}#sk-container-id-1 div.sk-estimator {font-family: monospace;background-color: #f0f8ff;border: 1px dotted black;border-radius: 0.25em;box-sizing: border-box;margin-bottom: 0.5em;}#sk-container-id-1 div.sk-estimator:hover {background-color: #d4ebff;}#sk-container-id-1 div.sk-parallel-item::after {content: \"\";width: 100%;border-bottom: 1px solid gray;flex-grow: 1;}#sk-container-id-1 div.sk-label:hover label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-1 div.sk-serial::before {content: \"\";position: absolute;border-left: 1px solid gray;box-sizing: border-box;top: 0;bottom: 0;left: 50%;z-index: 0;}#sk-container-id-1 div.sk-serial {display: flex;flex-direction: column;align-items: center;background-color: white;padding-right: 0.2em;padding-left: 0.2em;position: relative;}#sk-container-id-1 div.sk-item {position: relative;z-index: 1;}#sk-container-id-1 div.sk-parallel {display: flex;align-items: stretch;justify-content: center;background-color: white;position: relative;}#sk-container-id-1 div.sk-item::before, #sk-container-id-1 div.sk-parallel-item::before {content: \"\";position: absolute;border-left: 1px solid gray;box-sizing: border-box;top: 0;bottom: 0;left: 50%;z-index: -1;}#sk-container-id-1 div.sk-parallel-item {display: flex;flex-direction: column;z-index: 1;position: relative;background-color: white;}#sk-container-id-1 div.sk-parallel-item:first-child::after {align-self: flex-end;width: 50%;}#sk-container-id-1 div.sk-parallel-item:last-child::after {align-self: flex-start;width: 50%;}#sk-container-id-1 div.sk-parallel-item:only-child::after {width: 0;}#sk-container-id-1 div.sk-dashed-wrapped {border: 1px dashed gray;margin: 0 0.4em 0.5em 0.4em;box-sizing: border-box;padding-bottom: 0.4em;background-color: white;}#sk-container-id-1 div.sk-label label {font-family: monospace;font-weight: bold;display: inline-block;line-height: 1.2em;}#sk-container-id-1 div.sk-label-container {text-align: center;}#sk-container-id-1 div.sk-container {/* jupyter's `normalize.less` sets `[hidden] { display: none; }` but bootstrap.min.css set `[hidden] { display: none !important; }` so we also need the `!important` here to be able to override the default hidden behavior on the sphinx rendered scikit-learn.org. See: https://github.com/scikit-learn/scikit-learn/issues/21755 */display: inline-block !important;position: relative;}#sk-container-id-1 div.sk-text-repr-fallback {display: none;}</style><div id=\"sk-container-id-1\" class=\"sk-top-container\"><div class=\"sk-text-repr-fallback\"><pre>KMeans(n_clusters=2, random_state=66)</pre><b>In a Jupyter environment, please rerun this cell to show the HTML representation or trust the notebook. <br />On GitHub, the HTML representation is unable to render, please try loading this page with nbviewer.org.</b></div><div class=\"sk-container\" hidden><div class=\"sk-item\"><div class=\"sk-estimator sk-toggleable\"><input class=\"sk-toggleable__control sk-hidden--visually\" id=\"sk-estimator-id-1\" type=\"checkbox\" checked><label for=\"sk-estimator-id-1\" class=\"sk-toggleable__label sk-toggleable__label-arrow\">KMeans</label><div class=\"sk-toggleable__content\"><pre>KMeans(n_clusters=2, random_state=66)</pre></div></div></div></div></div>"
      ],
      "text/plain": [
       "KMeans(n_clusters=2, random_state=66)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kmeans= KMeans(n_clusters=2,random_state=66)\n",
    "kmeans"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Q3. create a DataFrame that contains the 58 sample labels as one column, the cluster number as another column and the subtype given in the dataset (adenocarcinoma or squamous cell carcinoma) as another column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "could not convert string to float: '1007_s_at'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/var/folders/gf/79pl6dp1569b76cgzy_9yv7c0000gs/T/ipykernel_73979/1572544828.py\u001b[0m in \u001b[0;36m?\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m#use the kmeans model\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mkmeans\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0malldata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/sklearn/base.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(estimator, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1147\u001b[0m                 skip_parameter_validation=(\n\u001b[1;32m   1148\u001b[0m                     \u001b[0mprefer_skip_nested_validation\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mglobal_skip_validation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1149\u001b[0m                 )\n\u001b[1;32m   1150\u001b[0m             ):\n\u001b[0;32m-> 1151\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mfit_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mestimator\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/sklearn/cluster/_kmeans.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, X, y, sample_weight)\u001b[0m\n\u001b[1;32m   1467\u001b[0m         \u001b[0;34m-\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1468\u001b[0m         \u001b[0mself\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mobject\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1469\u001b[0m             \u001b[0mFitted\u001b[0m \u001b[0mestimator\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1470\u001b[0m         \"\"\"\n\u001b[0;32m-> 1471\u001b[0;31m         X = self._validate_data(\n\u001b[0m\u001b[1;32m   1472\u001b[0m             \u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1473\u001b[0m             \u001b[0maccept_sparse\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"csr\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1474\u001b[0m             \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfloat64\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfloat32\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/sklearn/base.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, X, y, reset, validate_separately, cast_to_ndarray, **check_params)\u001b[0m\n\u001b[1;32m    600\u001b[0m                 \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    601\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    602\u001b[0m                 \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    603\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mno_val_X\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mno_val_y\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 604\u001b[0;31m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcheck_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mX\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0minput_name\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"X\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcheck_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    605\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mno_val_X\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mno_val_y\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    606\u001b[0m             \u001b[0mout\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_check_y\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mcheck_params\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    607\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator, input_name)\u001b[0m\n\u001b[1;32m    914\u001b[0m                         )\n\u001b[1;32m    915\u001b[0m                     \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mxp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    916\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    917\u001b[0m                     \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_asarray_with_order\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mxp\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mxp\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 918\u001b[0;31m             \u001b[0;32mexcept\u001b[0m \u001b[0mComplexWarning\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mcomplex_warning\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    919\u001b[0m                 raise ValueError(\n\u001b[1;32m    920\u001b[0m                     \u001b[0;34m\"Complex data not supported\\n{}\\n\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    921\u001b[0m                 ) from complex_warning\n",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/sklearn/utils/_array_api.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(array, dtype, order, copy, xp)\u001b[0m\n\u001b[1;32m    376\u001b[0m         \u001b[0;31m# Use NumPy API to support order\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    377\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcopy\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    378\u001b[0m             \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnumpy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    379\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 380\u001b[0;31m             \u001b[0marray\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnumpy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0morder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    381\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    382\u001b[0m         \u001b[0;31m# At this point array is a NumPy ndarray. We convert it to an array\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    383\u001b[0m         \u001b[0;31m# container that is consistent with the input's namespace.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m?\u001b[0;34m(self, dtype)\u001b[0m\n\u001b[1;32m   2082\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__array__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mnpt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDTypeLike\u001b[0m \u001b[0;34m|\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2083\u001b[0m         \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2084\u001b[0;31m         \u001b[0marr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2085\u001b[0m         if (\n\u001b[1;32m   2086\u001b[0m             \u001b[0mastype_is_view\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0marr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2087\u001b[0m             \u001b[0;32mand\u001b[0m \u001b[0musing_copy_on_write\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: could not convert string to float: '1007_s_at'"
     ]
    }
   ],
   "source": [
    "#use the kmeans model\n",
    "kmeans.fit(alldata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'KMeans' object has no attribute 'labels_'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[23], line 8\u001b[0m\n\u001b[1;32m      6\u001b[0m new_df \u001b[39m=\u001b[39m new_df\u001b[39m.\u001b[39mset_index(\u001b[39m'\u001b[39m\u001b[39mSample Label\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[1;32m      7\u001b[0m \u001b[39m#2.Cluster Number\u001b[39;00m\n\u001b[0;32m----> 8\u001b[0m labels \u001b[39m=\u001b[39m kmeans\u001b[39m.\u001b[39mlabels_\n\u001b[1;32m      9\u001b[0m new_df[\u001b[39m'\u001b[39m\u001b[39mCluster Number\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m labels\n\u001b[1;32m     10\u001b[0m \u001b[39m#3.Subtype\u001b[39;00m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'KMeans' object has no attribute 'labels_'"
     ]
    }
   ],
   "source": [
    "#new dataframe, contains the 58 sample labels as one column, \n",
    "#the cluster number as another column and the subtype given in the dataset \n",
    "index_data = alldata.index\n",
    "#Sample label\n",
    "new_df = pd.DataFrame(index_data, columns=['Sample Label'])\n",
    "new_df = new_df.set_index('Sample Label')\n",
    "#2.Cluster Number\n",
    "labels = kmeans.labels_\n",
    "new_df['Cluster Number'] = labels\n",
    "#3.Subtype\n",
    "series = gse.phenotype_data[\"title\"].str.split('_').str[1]\n",
    "series_df = series.to_frame('Subtype')\n",
    "new_df = new_df.merge(series_df,left_index=True, right_index=True)\n",
    "new_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Q4.Compute and display the accuracy of this clustering.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TP: cluster number = 1 & subtype = AD    \n",
    "FN: cluster number = 0 & subtype = SCC  \n",
    "TN: cluster number = 1 & subtype = SCC  \n",
    "FP: cluster number = 0 & subtype = AD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "TP_counts = sum((new_df['Cluster Number'] == 1) & (new_df['Subtype'] == 'AC'))\n",
    "FN_counts = sum((new_df['Cluster Number'] == 0) & (new_df['Subtype'] == 'SCC'))\n",
    "TN_counts = sum((new_df['Cluster Number'] == 1) & (new_df['Subtype'] == 'Scd CC'))\n",
    "FP_counts = sum((new_df['Cluster Number'] == 0) & (new_df['Subtype'] == 'AC'))\n",
    "#also, TP+FN+TN+FP = 58"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\\$\\$\n",
    "Accuracy = \\frac{TP+NF}{TP+NF+TF+NP} \n",
    "\\$\\$"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Accuracy: 0.9310344827586207\n"
     ]
    }
   ],
   "source": [
    "Accuracy = (TP_counts+FN_counts)/(TP_counts+FN_counts+TN_counts+FP_counts)\n",
    "print(\"Accuracy:\", Accuracy)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Q5. Fit the model on half of the data (balanced: 20 AD and 9 SCC) and assign a label to the clusters (AD or SCC), then use the model to predict the cluster/label for the other half of the data.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=42)\n",
    "#x is training matrix \n",
    "# y is labels we need 20 AD and 9 SCC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Q6. Compute and display the accuracy of the model for the prediction results (testing data)."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
