{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lung Cancer Subtyping\n",
    "a mathematical model that can discriminate between two lung cancer subtypes to build the model \n",
    "we use an unsupervised k-means clustering algorithm (Euclidean distance) of 58 NSCLC tumors using k=2\n",
    "\n",
    "data from: \n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE10245\n",
    "\n",
    "[GSE10245_family.soft.gz](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE10nnn/GSE10245/soft/)\n",
    "\n",
    "40 AC 18 SC\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "22-Nov-2023 11:44:44 INFO GEOparse - Parsing ./GSE10245_family.soft.gz: \n",
      "22-Nov-2023 11:44:44 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "22-Nov-2023 11:44:44 DEBUG GEOparse - SERIES: GSE10245\n",
      "22-Nov-2023 11:44:44 DEBUG GEOparse - PLATFORM: GPL570\n",
      "/Users/ymanasa/anaconda3/lib/python3.11/site-packages/GEOparse/GEOparse.py:401: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "22-Nov-2023 11:44:45 DEBUG GEOparse - SAMPLE: GSM258551\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258552\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258553\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258554\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258555\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258556\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258557\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258558\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258559\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258560\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258561\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258562\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258563\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258564\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258565\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258566\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258567\n",
      "22-Nov-2023 11:44:46 DEBUG GEOparse - SAMPLE: GSM258568\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258569\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258570\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258571\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258572\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258573\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258574\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258575\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258576\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258577\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258578\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258579\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258580\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258581\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258582\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258583\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258584\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258585\n",
      "22-Nov-2023 11:44:47 DEBUG GEOparse - SAMPLE: GSM258586\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258587\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258588\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258589\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258590\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258591\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258592\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258593\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258594\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258595\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258596\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258597\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258598\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258599\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258600\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258601\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258602\n",
      "22-Nov-2023 11:44:48 DEBUG GEOparse - SAMPLE: GSM258603\n",
      "22-Nov-2023 11:44:49 DEBUG GEOparse - SAMPLE: GSM258604\n",
      "22-Nov-2023 11:44:49 DEBUG GEOparse - SAMPLE: GSM258605\n",
      "22-Nov-2023 11:44:49 DEBUG GEOparse - SAMPLE: GSM258606\n",
      "22-Nov-2023 11:44:49 DEBUG GEOparse - SAMPLE: GSM258607\n",
      "22-Nov-2023 11:44:49 DEBUG GEOparse - SAMPLE: GSM258608\n"
     ]
    }
   ],
   "source": [
    "# load the data using the GEOparse module\n",
    "# code adapted from: https://geoparse.readthedocs.io/en/latest/usage.html\n",
    "\n",
    "import GEOparse\n",
    "gse = GEOparse.get_GEO(filepath=\"./GSE10245_family.soft.gz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_items([('GSM258551', <SAMPLE: GSM258551>), ('GSM258552', <SAMPLE: GSM258552>), ('GSM258553', <SAMPLE: GSM258553>), ('GSM258554', <SAMPLE: GSM258554>), ('GSM258555', <SAMPLE: GSM258555>), ('GSM258556', <SAMPLE: GSM258556>), ('GSM258557', <SAMPLE: GSM258557>), ('GSM258558', <SAMPLE: GSM258558>), ('GSM258559', <SAMPLE: GSM258559>), ('GSM258560', <SAMPLE: GSM258560>), ('GSM258561', <SAMPLE: GSM258561>), ('GSM258562', <SAMPLE: GSM258562>), ('GSM258563', <SAMPLE: GSM258563>), ('GSM258564', <SAMPLE: GSM258564>), ('GSM258565', <SAMPLE: GSM258565>), ('GSM258566', <SAMPLE: GSM258566>), ('GSM258567', <SAMPLE: GSM258567>), ('GSM258568', <SAMPLE: GSM258568>), ('GSM258569', <SAMPLE: GSM258569>), ('GSM258570', <SAMPLE: GSM258570>), ('GSM258571', <SAMPLE: GSM258571>), ('GSM258572', <SAMPLE: GSM258572>), ('GSM258573', <SAMPLE: GSM258573>), ('GSM258574', <SAMPLE: GSM258574>), ('GSM258575', <SAMPLE: GSM258575>), ('GSM258576', <SAMPLE: GSM258576>), ('GSM258577', <SAMPLE: GSM258577>), ('GSM258578', <SAMPLE: GSM258578>), ('GSM258579', <SAMPLE: GSM258579>), ('GSM258580', <SAMPLE: GSM258580>), ('GSM258581', <SAMPLE: GSM258581>), ('GSM258582', <SAMPLE: GSM258582>), ('GSM258583', <SAMPLE: GSM258583>), ('GSM258584', <SAMPLE: GSM258584>), ('GSM258585', <SAMPLE: GSM258585>), ('GSM258586', <SAMPLE: GSM258586>), ('GSM258587', <SAMPLE: GSM258587>), ('GSM258588', <SAMPLE: GSM258588>), ('GSM258589', <SAMPLE: GSM258589>), ('GSM258590', <SAMPLE: GSM258590>), ('GSM258591', <SAMPLE: GSM258591>), ('GSM258592', <SAMPLE: GSM258592>), ('GSM258593', <SAMPLE: GSM258593>), ('GSM258594', <SAMPLE: GSM258594>), ('GSM258595', <SAMPLE: GSM258595>), ('GSM258596', <SAMPLE: GSM258596>), ('GSM258597', <SAMPLE: GSM258597>), ('GSM258598', <SAMPLE: GSM258598>), ('GSM258599', <SAMPLE: GSM258599>), ('GSM258600', <SAMPLE: GSM258600>), ('GSM258601', <SAMPLE: GSM258601>), ('GSM258602', <SAMPLE: GSM258602>), ('GSM258603', <SAMPLE: GSM258603>), ('GSM258604', <SAMPLE: GSM258604>), ('GSM258605', <SAMPLE: GSM258605>), ('GSM258606', <SAMPLE: GSM258606>), ('GSM258607', <SAMPLE: GSM258607>), ('GSM258608', <SAMPLE: GSM258608>)])\n",
      "dict_items([('title', ['NSCLC_AC_10']), ('geo_accession', ['GSM258551']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 10.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258551/suppl/GSM258551.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_16']), ('geo_accession', ['GSM258552']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 16.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258552/suppl/GSM258552.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_22']), ('geo_accession', ['GSM258553']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 22.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258553/suppl/GSM258553.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_24']), ('geo_accession', ['GSM258554']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 24.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258554/suppl/GSM258554.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_25']), ('geo_accession', ['GSM258555']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 25.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258555/suppl/GSM258555.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_30']), ('geo_accession', ['GSM258556']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 30.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258556/suppl/GSM258556.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_37']), ('geo_accession', ['GSM258557']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 37.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258557/suppl/GSM258557.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_40']), ('geo_accession', ['GSM258558']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 40.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258558/suppl/GSM258558.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_49']), ('geo_accession', ['GSM258559']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 49.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258559/suppl/GSM258559.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_51']), ('geo_accession', ['GSM258560']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 51.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258560/suppl/GSM258560.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_53']), ('geo_accession', ['GSM258561']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 53.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258561/suppl/GSM258561.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_55']), ('geo_accession', ['GSM258562']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 55.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258562/suppl/GSM258562.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_65']), ('geo_accession', ['GSM258563']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 65.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258563/suppl/GSM258563.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_70']), ('geo_accession', ['GSM258564']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 70.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258564/suppl/GSM258564.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_75']), ('geo_accession', ['GSM258565']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 75.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258565/suppl/GSM258565.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_77']), ('geo_accession', ['GSM258566']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 77.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258566/suppl/GSM258566.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_84']), ('geo_accession', ['GSM258567']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 84.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258567/suppl/GSM258567.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_88']), ('geo_accession', ['GSM258568']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 88.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258568/suppl/GSM258568.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_89']), ('geo_accession', ['GSM258569']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 89.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258569/suppl/GSM258569.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_95']), ('geo_accession', ['GSM258570']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 95.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258570/suppl/GSM258570.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_97']), ('geo_accession', ['GSM258571']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 97.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258571/suppl/GSM258571.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_113']), ('geo_accession', ['GSM258572']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 113.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258572/suppl/GSM258572.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_136']), ('geo_accession', ['GSM258573']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 136.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258573/suppl/GSM258573.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_137']), ('geo_accession', ['GSM258574']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 137.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258574/suppl/GSM258574.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_147']), ('geo_accession', ['GSM258575']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 147.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258575/suppl/GSM258575.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_148']), ('geo_accession', ['GSM258576']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 148.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258576/suppl/GSM258576.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_161']), ('geo_accession', ['GSM258577']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 161.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258577/suppl/GSM258577.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_165']), ('geo_accession', ['GSM258578']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 165.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258578/suppl/GSM258578.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_169']), ('geo_accession', ['GSM258579']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 169.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258579/suppl/GSM258579.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_188']), ('geo_accession', ['GSM258580']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 188.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258580/suppl/GSM258580.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_202']), ('geo_accession', ['GSM258581']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 202.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258581/suppl/GSM258581.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_210']), ('geo_accession', ['GSM258582']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 210.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258582/suppl/GSM258582.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_215']), ('geo_accession', ['GSM258583']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 215.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258583/suppl/GSM258583.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_233']), ('geo_accession', ['GSM258584']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 233.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258584/suppl/GSM258584.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_248']), ('geo_accession', ['GSM258585']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 248.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258585/suppl/GSM258585.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_271']), ('geo_accession', ['GSM258586']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 271.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258586/suppl/GSM258586.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_273']), ('geo_accession', ['GSM258587']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 273.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258587/suppl/GSM258587.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_276']), ('geo_accession', ['GSM258588']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 276.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258588/suppl/GSM258588.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_282']), ('geo_accession', ['GSM258589']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 282.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258589/suppl/GSM258589.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_301']), ('geo_accession', ['GSM258590']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 301.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258590/suppl/GSM258590.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_331']), ('geo_accession', ['GSM258591']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 331.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258591/suppl/GSM258591.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_334']), ('geo_accession', ['GSM258592']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 334.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258592/suppl/GSM258592.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_342']), ('geo_accession', ['GSM258593']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 342.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258593/suppl/GSM258593.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_347']), ('geo_accession', ['GSM258594']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 347.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258594/suppl/GSM258594.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_348']), ('geo_accession', ['GSM258595']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 348.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258595/suppl/GSM258595.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_364']), ('geo_accession', ['GSM258596']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 364.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258596/suppl/GSM258596.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_368']), ('geo_accession', ['GSM258597']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 368.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258597/suppl/GSM258597.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_378']), ('geo_accession', ['GSM258598']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 378.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258598/suppl/GSM258598.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_390']), ('geo_accession', ['GSM258599']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 390.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258599/suppl/GSM258599.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_397']), ('geo_accession', ['GSM258600']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 397.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258600/suppl/GSM258600.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_400']), ('geo_accession', ['GSM258601']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 400.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258601/suppl/GSM258601.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_402']), ('geo_accession', ['GSM258602']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 402.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258602/suppl/GSM258602.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_412']), ('geo_accession', ['GSM258603']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 412.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258603/suppl/GSM258603.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_426']), ('geo_accession', ['GSM258604']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 426.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258604/suppl/GSM258604.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_434']), ('geo_accession', ['GSM258605']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 434.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258605/suppl/GSM258605.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_445']), ('geo_accession', ['GSM258606']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 445.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258606/suppl/GSM258606.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_489']), ('geo_accession', ['GSM258607']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 489.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258607/suppl/GSM258607.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_492']), ('geo_accession', ['GSM258608']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 492.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258608/suppl/GSM258608.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n"
     ]
    }
   ],
   "source": [
    "# gse.gsms is a dict \n",
    "print(gse.gsms.items())\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "# gsm is a GEOparse `object <class 'GEOparse.GEOTypes.GSM'>`\n",
    "# gsm has metadata attribute \n",
    "    print(gsm.metadata.items())\n",
    "    \n",
    "# print( gse.gsms.items())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>GSM258558</th>\n",
       "      <th>GSM258559</th>\n",
       "      <th>GSM258560</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID_REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1007_s_at</th>\n",
       "      <td>9.129905</td>\n",
       "      <td>9.843349</td>\n",
       "      <td>9.730661</td>\n",
       "      <td>9.032165</td>\n",
       "      <td>10.281793</td>\n",
       "      <td>9.154551</td>\n",
       "      <td>8.063788</td>\n",
       "      <td>8.881805</td>\n",
       "      <td>9.039005</td>\n",
       "      <td>10.643545</td>\n",
       "      <td>...</td>\n",
       "      <td>9.628212</td>\n",
       "      <td>10.130757</td>\n",
       "      <td>9.276419</td>\n",
       "      <td>9.296957</td>\n",
       "      <td>10.346701</td>\n",
       "      <td>9.709068</td>\n",
       "      <td>7.672985</td>\n",
       "      <td>9.199080</td>\n",
       "      <td>7.635765</td>\n",
       "      <td>8.760082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1053_at</th>\n",
       "      <td>8.034022</td>\n",
       "      <td>7.973332</td>\n",
       "      <td>8.834045</td>\n",
       "      <td>7.723965</td>\n",
       "      <td>9.040800</td>\n",
       "      <td>8.792375</td>\n",
       "      <td>8.093354</td>\n",
       "      <td>7.575620</td>\n",
       "      <td>8.438594</td>\n",
       "      <td>9.142260</td>\n",
       "      <td>...</td>\n",
       "      <td>9.215567</td>\n",
       "      <td>8.273698</td>\n",
       "      <td>8.790710</td>\n",
       "      <td>8.861625</td>\n",
       "      <td>8.512193</td>\n",
       "      <td>7.908581</td>\n",
       "      <td>9.299454</td>\n",
       "      <td>8.220959</td>\n",
       "      <td>7.432135</td>\n",
       "      <td>7.596431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117_at</th>\n",
       "      <td>3.564520</td>\n",
       "      <td>4.994852</td>\n",
       "      <td>5.066018</td>\n",
       "      <td>4.958580</td>\n",
       "      <td>4.951835</td>\n",
       "      <td>6.428274</td>\n",
       "      <td>7.364676</td>\n",
       "      <td>4.659326</td>\n",
       "      <td>6.643723</td>\n",
       "      <td>5.368356</td>\n",
       "      <td>...</td>\n",
       "      <td>5.183919</td>\n",
       "      <td>4.801579</td>\n",
       "      <td>6.161777</td>\n",
       "      <td>5.443126</td>\n",
       "      <td>4.609810</td>\n",
       "      <td>4.339004</td>\n",
       "      <td>6.384404</td>\n",
       "      <td>6.006534</td>\n",
       "      <td>5.484470</td>\n",
       "      <td>5.372583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121_at</th>\n",
       "      <td>4.746490</td>\n",
       "      <td>5.197306</td>\n",
       "      <td>5.234618</td>\n",
       "      <td>6.078180</td>\n",
       "      <td>5.205632</td>\n",
       "      <td>5.009848</td>\n",
       "      <td>5.533000</td>\n",
       "      <td>5.362637</td>\n",
       "      <td>5.210988</td>\n",
       "      <td>5.195820</td>\n",
       "      <td>...</td>\n",
       "      <td>5.202508</td>\n",
       "      <td>4.807654</td>\n",
       "      <td>5.326347</td>\n",
       "      <td>5.205004</td>\n",
       "      <td>5.201123</td>\n",
       "      <td>6.028717</td>\n",
       "      <td>5.220915</td>\n",
       "      <td>5.186452</td>\n",
       "      <td>5.220778</td>\n",
       "      <td>5.097535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1255_g_at</th>\n",
       "      <td>2.320698</td>\n",
       "      <td>2.248520</td>\n",
       "      <td>2.259504</td>\n",
       "      <td>2.262787</td>\n",
       "      <td>2.207531</td>\n",
       "      <td>2.322618</td>\n",
       "      <td>2.656929</td>\n",
       "      <td>2.255955</td>\n",
       "      <td>2.266399</td>\n",
       "      <td>2.246563</td>\n",
       "      <td>...</td>\n",
       "      <td>2.248392</td>\n",
       "      <td>2.232021</td>\n",
       "      <td>2.246738</td>\n",
       "      <td>2.245739</td>\n",
       "      <td>2.265454</td>\n",
       "      <td>2.301397</td>\n",
       "      <td>2.322604</td>\n",
       "      <td>2.236494</td>\n",
       "      <td>2.243698</td>\n",
       "      <td>2.420731</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "name       GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  GSM258556  \\\n",
       "ID_REF                                                                        \n",
       "1007_s_at   9.129905   9.843349   9.730661   9.032165  10.281793   9.154551   \n",
       "1053_at     8.034022   7.973332   8.834045   7.723965   9.040800   8.792375   \n",
       "117_at      3.564520   4.994852   5.066018   4.958580   4.951835   6.428274   \n",
       "121_at      4.746490   5.197306   5.234618   6.078180   5.205632   5.009848   \n",
       "1255_g_at   2.320698   2.248520   2.259504   2.262787   2.207531   2.322618   \n",
       "\n",
       "name       GSM258557  GSM258558  GSM258559  GSM258560  ...  GSM258599  \\\n",
       "ID_REF                                                 ...              \n",
       "1007_s_at   8.063788   8.881805   9.039005  10.643545  ...   9.628212   \n",
       "1053_at     8.093354   7.575620   8.438594   9.142260  ...   9.215567   \n",
       "117_at      7.364676   4.659326   6.643723   5.368356  ...   5.183919   \n",
       "121_at      5.533000   5.362637   5.210988   5.195820  ...   5.202508   \n",
       "1255_g_at   2.656929   2.255955   2.266399   2.246563  ...   2.248392   \n",
       "\n",
       "name       GSM258600  GSM258601  GSM258602  GSM258603  GSM258604  GSM258605  \\\n",
       "ID_REF                                                                        \n",
       "1007_s_at  10.130757   9.276419   9.296957  10.346701   9.709068   7.672985   \n",
       "1053_at     8.273698   8.790710   8.861625   8.512193   7.908581   9.299454   \n",
       "117_at      4.801579   6.161777   5.443126   4.609810   4.339004   6.384404   \n",
       "121_at      4.807654   5.326347   5.205004   5.201123   6.028717   5.220915   \n",
       "1255_g_at   2.232021   2.246738   2.245739   2.265454   2.301397   2.322604   \n",
       "\n",
       "name       GSM258606  GSM258607  GSM258608  \n",
       "ID_REF                                      \n",
       "1007_s_at   9.199080   7.635765   8.760082  \n",
       "1053_at     8.220959   7.432135   7.596431  \n",
       "117_at      6.006534   5.484470   5.372583  \n",
       "121_at      5.186452   5.220778   5.097535  \n",
       "1255_g_at   2.236494   2.243698   2.420731  \n",
       "\n",
       "[5 rows x 58 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# getting expression data from gse object as a dataframe \n",
    "expr = pd.DataFrame(gse.pivot_samples('VALUE'))\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_items([('GPL570', <d: GPL570>)])\n",
      "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
      "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
      "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
      "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
      "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
      "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
      "\n",
      "       Sequence Type                  Sequence Source  \\\n",
      "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "1  Exemplar sequence                          GenBank   \n",
      "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "3  Exemplar sequence                          GenBank   \n",
      "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "\n",
      "                                  Target Description Representative Public ID  \\\n",
      "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
      "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
      "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
      "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
      "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
      "\n",
      "                                          Gene Title       Gene Symbol  \\\n",
      "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
      "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
      "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
      "3                                       paired box 8              PAX8   \n",
      "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
      "\n",
      "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
      "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
      "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
      "2               3310                                          NM_002155   \n",
      "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
      "4               2978                         NM_000409 /// XM_006715073   \n",
      "\n",
      "                    Gene Ontology Biological Process  \\\n",
      "0  0001558 // regulation of cell growth // inferr...   \n",
      "1  0000278 // mitotic cell cycle // traceable aut...   \n",
      "2  0000902 // cell morphogenesis // inferred from...   \n",
      "3  0001655 // urogenital system development // in...   \n",
      "4  0007165 // signal transduction // non-traceabl...   \n",
      "\n",
      "                    Gene Ontology Cellular Component  \\\n",
      "0  0005576 // extracellular region // inferred fr...   \n",
      "1  0005634 // nucleus // inferred from electronic...   \n",
      "2  0005737 // cytoplasm // inferred from direct a...   \n",
      "3  0005634 // nucleus // inferred from direct ass...   \n",
      "4  0001750 // photoreceptor outer segment // infe...   \n",
      "\n",
      "                    Gene Ontology Molecular Function  \n",
      "0  0000166 // nucleotide binding // inferred from...  \n",
      "1  0000166 // nucleotide binding // inferred from...  \n",
      "2  0000166 // nucleotide binding // inferred from...  \n",
      "3  0000979 // RNA polymerase II core promoter seq...  \n",
      "4  0005509 // calcium ion binding // inferred fro...  \n"
     ]
    }
   ],
   "source": [
    "# gpls attribute of gse \n",
    "print(gse.gpls.items())\n",
    "for gpl_name, gpl in gse.gpls.items():\n",
    "    print(gpl.table.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>GB_ACC</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <th>Annotation Date</th>\n",
       "      <th>Sequence Type</th>\n",
       "      <th>Sequence Source</th>\n",
       "      <th>Target Description</th>\n",
       "      <th>Representative Public ID</th>\n",
       "      <th>Gene Title</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <th>RefSeq Transcript ID</th>\n",
       "      <th>Gene Ontology Biological Process</th>\n",
       "      <th>Gene Ontology Cellular Component</th>\n",
       "      <th>Gene Ontology Molecular Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>U48705</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...</td>\n",
       "      <td>U48705</td>\n",
       "      <td>discoidin domain receptor tyrosine kinase 1 //...</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "      <td>NM_001202521 /// NM_001202522 /// NM_001202523...</td>\n",
       "      <td>0001558 // regulation of cell growth // inferr...</td>\n",
       "      <td>0005576 // extracellular region // inferred fr...</td>\n",
       "      <td>0000166 // nucleotide binding // inferred from...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>M87338</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...</td>\n",
       "      <td>M87338</td>\n",
       "      <td>replication factor C (activator 1) 2, 40kDa</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "      <td>NM_001278791 /// NM_001278792 /// NM_001278793...</td>\n",
       "      <td>0000278 // mitotic cell cycle // traceable aut...</td>\n",
       "      <td>0005634 // nucleus // inferred from electronic...</td>\n",
       "      <td>0000166 // nucleotide binding // inferred from...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>X51757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...</td>\n",
       "      <td>X51757</td>\n",
       "      <td>heat shock 70kDa protein 6 (HSP70B')</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "      <td>NM_002155</td>\n",
       "      <td>0000902 // cell morphogenesis // inferred from...</td>\n",
       "      <td>0005737 // cytoplasm // inferred from direct a...</td>\n",
       "      <td>0000166 // nucleotide binding // inferred from...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>X69699</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>GenBank</td>\n",
       "      <td>X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...</td>\n",
       "      <td>X69699</td>\n",
       "      <td>paired box 8</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "      <td>NM_003466 /// NM_013951 /// NM_013952 /// NM_0...</td>\n",
       "      <td>0001655 // urogenital system development // in...</td>\n",
       "      <td>0005634 // nucleus // inferred from direct ass...</td>\n",
       "      <td>0000979 // RNA polymerase II core promoter seq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>L36861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Oct 6, 2014</td>\n",
       "      <td>Exemplar sequence</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...</td>\n",
       "      <td>L36861</td>\n",
       "      <td>guanylate cyclase activator 1A (retina)</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "      <td>NM_000409 /// XM_006715073</td>\n",
       "      <td>0007165 // signal transduction // non-traceabl...</td>\n",
       "      <td>0001750 // photoreceptor outer segment // infe...</td>\n",
       "      <td>0005509 // calcium ion binding // inferred fro...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
       "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
       "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
       "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
       "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
       "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
       "\n",
       "       Sequence Type                  Sequence Source  \\\n",
       "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
       "1  Exemplar sequence                          GenBank   \n",
       "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
       "3  Exemplar sequence                          GenBank   \n",
       "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
       "\n",
       "                                  Target Description Representative Public ID  \\\n",
       "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
       "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
       "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
       "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
       "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
       "\n",
       "                                          Gene Title       Gene Symbol  \\\n",
       "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
       "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
       "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
       "3                                       paired box 8              PAX8   \n",
       "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
       "\n",
       "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
       "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
       "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
       "2               3310                                          NM_002155   \n",
       "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
       "4               2978                         NM_000409 /// XM_006715073   \n",
       "\n",
       "                    Gene Ontology Biological Process  \\\n",
       "0  0001558 // regulation of cell growth // inferr...   \n",
       "1  0000278 // mitotic cell cycle // traceable aut...   \n",
       "2  0000902 // cell morphogenesis // inferred from...   \n",
       "3  0001655 // urogenital system development // in...   \n",
       "4  0007165 // signal transduction // non-traceabl...   \n",
       "\n",
       "                    Gene Ontology Cellular Component  \\\n",
       "0  0005576 // extracellular region // inferred fr...   \n",
       "1  0005634 // nucleus // inferred from electronic...   \n",
       "2  0005737 // cytoplasm // inferred from direct a...   \n",
       "3  0005634 // nucleus // inferred from direct ass...   \n",
       "4  0001750 // photoreceptor outer segment // infe...   \n",
       "\n",
       "                    Gene Ontology Molecular Function  \n",
       "0  0000166 // nucleotide binding // inferred from...  \n",
       "1  0000166 // nucleotide binding // inferred from...  \n",
       "2  0000166 // nucleotide binding // inferred from...  \n",
       "3  0000979 // RNA polymerase II core promoter seq...  \n",
       "4  0005509 // calcium ion binding // inferred fro...  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gpl.table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54670</th>\n",
       "      <td>AFFX-ThrX-5_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54671</th>\n",
       "      <td>AFFX-ThrX-M_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54672</th>\n",
       "      <td>AFFX-TrpnX-3_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54673</th>\n",
       "      <td>AFFX-TrpnX-5_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54674</th>\n",
       "      <td>AFFX-TrpnX-M_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54675 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID       Gene Symbol     ENTREZ_GENE_ID\n",
       "0            1007_s_at  DDR1 /// MIR4640  780 /// 100616237\n",
       "1              1053_at              RFC2               5982\n",
       "2               117_at             HSPA6               3310\n",
       "3               121_at              PAX8               7849\n",
       "4            1255_g_at            GUCA1A               2978\n",
       "...                ...               ...                ...\n",
       "54670   AFFX-ThrX-5_at               NaN                NaN\n",
       "54671   AFFX-ThrX-M_at               NaN                NaN\n",
       "54672  AFFX-TrpnX-3_at               NaN                NaN\n",
       "54673  AFFX-TrpnX-5_at               NaN                NaN\n",
       "54674  AFFX-TrpnX-M_at               NaN                NaN\n",
       "\n",
       "[54675 rows x 3 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# assinging ID, Gene Symbol, ENTREZ_GENE_ID columns from gpl.table as a pandas dataframe to metadata\n",
    "metadata = pd.DataFrame(gpl.table, columns=['ID', 'Gene Symbol', 'ENTREZ_GENE_ID'])\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54630</th>\n",
       "      <td>AFFX-HUMGAPDH/M33197_M_at</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>2597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54631</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_3_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54632</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_5_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54633</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MA_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54634</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MB_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44134 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                ID       Gene Symbol     ENTREZ_GENE_ID\n",
       "0                        1007_s_at  DDR1 /// MIR4640  780 /// 100616237\n",
       "1                          1053_at              RFC2               5982\n",
       "2                           117_at             HSPA6               3310\n",
       "3                           121_at              PAX8               7849\n",
       "4                        1255_g_at            GUCA1A               2978\n",
       "...                            ...               ...                ...\n",
       "54630    AFFX-HUMGAPDH/M33197_M_at             GAPDH               2597\n",
       "54631   AFFX-HUMISGF3A/M97935_3_at             STAT1               6772\n",
       "54632   AFFX-HUMISGF3A/M97935_5_at             STAT1               6772\n",
       "54633  AFFX-HUMISGF3A/M97935_MA_at             STAT1               6772\n",
       "54634  AFFX-HUMISGF3A/M97935_MB_at             STAT1               6772\n",
       "\n",
       "[44134 rows x 3 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# cleanup the data by dropping NaN\n",
    "metadata.dropna(inplace=True)\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "can only concatenate str (not \"float\") to str",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[19], line 8\u001b[0m\n\u001b[1;32m      4\u001b[0m merged_meta_expr \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39mmerge(metadata,expr, left_on\u001b[39m=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mID\u001b[39m\u001b[39m\"\u001b[39m, right_on\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mID_REF\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m      6\u001b[0m \u001b[39m# arrange mean from largest to smallest so that when we drop duplicates, the first occurance is kepy \u001b[39;00m\n\u001b[1;32m      7\u001b[0m \u001b[39m# the first occurance (with highest mean) will be the best one to keep\u001b[39;00m\n\u001b[0;32m----> 8\u001b[0m merged_meta_expr[\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m] \u001b[39m=\u001b[39m merged_meta_expr\u001b[39m.\u001b[39mmean(axis\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mcolumns\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[1;32m      9\u001b[0m sorted_merged \u001b[39m=\u001b[39m merged_meta_expr\u001b[39m.\u001b[39msort_values(by\u001b[39m=\u001b[39m[\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m],ascending\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m)\n\u001b[1;32m     10\u001b[0m sorted_merged\u001b[39m.\u001b[39mhead()\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/frame.py:11338\u001b[0m, in \u001b[0;36mDataFrame.mean\u001b[0;34m(self, axis, skipna, numeric_only, **kwargs)\u001b[0m\n\u001b[1;32m  11330\u001b[0m \u001b[39m@doc\u001b[39m(make_doc(\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m, ndim\u001b[39m=\u001b[39m\u001b[39m2\u001b[39m))\n\u001b[1;32m  11331\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mmean\u001b[39m(\n\u001b[1;32m  11332\u001b[0m     \u001b[39mself\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m  11336\u001b[0m     \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs,\n\u001b[1;32m  11337\u001b[0m ):\n\u001b[0;32m> 11338\u001b[0m     result \u001b[39m=\u001b[39m \u001b[39msuper\u001b[39m()\u001b[39m.\u001b[39mmean(axis, skipna, numeric_only, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[1;32m  11339\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39misinstance\u001b[39m(result, Series):\n\u001b[1;32m  11340\u001b[0m         result \u001b[39m=\u001b[39m result\u001b[39m.\u001b[39m__finalize__(\u001b[39mself\u001b[39m, method\u001b[39m=\u001b[39m\u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/generic.py:11978\u001b[0m, in \u001b[0;36mNDFrame.mean\u001b[0;34m(self, axis, skipna, numeric_only, **kwargs)\u001b[0m\n\u001b[1;32m  11971\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mmean\u001b[39m(\n\u001b[1;32m  11972\u001b[0m     \u001b[39mself\u001b[39m,\n\u001b[1;32m  11973\u001b[0m     axis: Axis \u001b[39m|\u001b[39m \u001b[39mNone\u001b[39;00m \u001b[39m=\u001b[39m \u001b[39m0\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m  11976\u001b[0m     \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs,\n\u001b[1;32m  11977\u001b[0m ) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m Series \u001b[39m|\u001b[39m \u001b[39mfloat\u001b[39m:\n\u001b[0;32m> 11978\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_stat_function(\n\u001b[1;32m  11979\u001b[0m         \u001b[39m\"\u001b[39m\u001b[39mmean\u001b[39m\u001b[39m\"\u001b[39m, nanops\u001b[39m.\u001b[39mnanmean, axis, skipna, numeric_only, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs\n\u001b[1;32m  11980\u001b[0m     )\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/generic.py:11935\u001b[0m, in \u001b[0;36mNDFrame._stat_function\u001b[0;34m(self, name, func, axis, skipna, numeric_only, **kwargs)\u001b[0m\n\u001b[1;32m  11931\u001b[0m nv\u001b[39m.\u001b[39mvalidate_func(name, (), kwargs)\n\u001b[1;32m  11933\u001b[0m validate_bool_kwarg(skipna, \u001b[39m\"\u001b[39m\u001b[39mskipna\u001b[39m\u001b[39m\"\u001b[39m, none_allowed\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m)\n\u001b[0;32m> 11935\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_reduce(\n\u001b[1;32m  11936\u001b[0m     func, name\u001b[39m=\u001b[39mname, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, numeric_only\u001b[39m=\u001b[39mnumeric_only\n\u001b[1;32m  11937\u001b[0m )\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/frame.py:11207\u001b[0m, in \u001b[0;36mDataFrame._reduce\u001b[0;34m(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)\u001b[0m\n\u001b[1;32m  11203\u001b[0m     df \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39mT\n\u001b[1;32m  11205\u001b[0m \u001b[39m# After possibly _get_data and transposing, we are now in the\u001b[39;00m\n\u001b[1;32m  11206\u001b[0m \u001b[39m#  simple case where we can use BlockManager.reduce\u001b[39;00m\n\u001b[0;32m> 11207\u001b[0m res \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39m_mgr\u001b[39m.\u001b[39mreduce(blk_func)\n\u001b[1;32m  11208\u001b[0m out \u001b[39m=\u001b[39m df\u001b[39m.\u001b[39m_constructor_from_mgr(res, axes\u001b[39m=\u001b[39mres\u001b[39m.\u001b[39maxes)\u001b[39m.\u001b[39miloc[\u001b[39m0\u001b[39m]\n\u001b[1;32m  11209\u001b[0m \u001b[39mif\u001b[39;00m out_dtype \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m \u001b[39mand\u001b[39;00m out\u001b[39m.\u001b[39mdtype \u001b[39m!=\u001b[39m \u001b[39m\"\u001b[39m\u001b[39mboolean\u001b[39m\u001b[39m\"\u001b[39m:\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/internals/managers.py:1459\u001b[0m, in \u001b[0;36mBlockManager.reduce\u001b[0;34m(self, func)\u001b[0m\n\u001b[1;32m   1457\u001b[0m res_blocks: \u001b[39mlist\u001b[39m[Block] \u001b[39m=\u001b[39m []\n\u001b[1;32m   1458\u001b[0m \u001b[39mfor\u001b[39;00m blk \u001b[39min\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mblocks:\n\u001b[0;32m-> 1459\u001b[0m     nbs \u001b[39m=\u001b[39m blk\u001b[39m.\u001b[39mreduce(func)\n\u001b[1;32m   1460\u001b[0m     res_blocks\u001b[39m.\u001b[39mextend(nbs)\n\u001b[1;32m   1462\u001b[0m index \u001b[39m=\u001b[39m Index([\u001b[39mNone\u001b[39;00m])  \u001b[39m# placeholder\u001b[39;00m\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/internals/blocks.py:377\u001b[0m, in \u001b[0;36mBlock.reduce\u001b[0;34m(self, func)\u001b[0m\n\u001b[1;32m    371\u001b[0m \u001b[39m@final\u001b[39m\n\u001b[1;32m    372\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mreduce\u001b[39m(\u001b[39mself\u001b[39m, func) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m \u001b[39mlist\u001b[39m[Block]:\n\u001b[1;32m    373\u001b[0m     \u001b[39m# We will apply the function and reshape the result into a single-row\u001b[39;00m\n\u001b[1;32m    374\u001b[0m     \u001b[39m#  Block with the same mgr_locs; squeezing will be done at a higher level\u001b[39;00m\n\u001b[1;32m    375\u001b[0m     \u001b[39massert\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mndim \u001b[39m==\u001b[39m \u001b[39m2\u001b[39m\n\u001b[0;32m--> 377\u001b[0m     result \u001b[39m=\u001b[39m func(\u001b[39mself\u001b[39m\u001b[39m.\u001b[39mvalues)\n\u001b[1;32m    379\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mvalues\u001b[39m.\u001b[39mndim \u001b[39m==\u001b[39m \u001b[39m1\u001b[39m:\n\u001b[1;32m    380\u001b[0m         res_values \u001b[39m=\u001b[39m result\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/frame.py:11139\u001b[0m, in \u001b[0;36mDataFrame._reduce.<locals>.blk_func\u001b[0;34m(values, axis)\u001b[0m\n\u001b[1;32m  11137\u001b[0m         \u001b[39mreturn\u001b[39;00m np\u001b[39m.\u001b[39marray([result])\n\u001b[1;32m  11138\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m> 11139\u001b[0m     \u001b[39mreturn\u001b[39;00m op(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwds)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/nanops.py:147\u001b[0m, in \u001b[0;36mbottleneck_switch.__call__.<locals>.f\u001b[0;34m(values, axis, skipna, **kwds)\u001b[0m\n\u001b[1;32m    145\u001b[0m         result \u001b[39m=\u001b[39m alt(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwds)\n\u001b[1;32m    146\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[0;32m--> 147\u001b[0m     result \u001b[39m=\u001b[39m alt(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwds)\n\u001b[1;32m    149\u001b[0m \u001b[39mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/nanops.py:404\u001b[0m, in \u001b[0;36m_datetimelike_compat.<locals>.new_func\u001b[0;34m(values, axis, skipna, mask, **kwargs)\u001b[0m\n\u001b[1;32m    401\u001b[0m \u001b[39mif\u001b[39;00m datetimelike \u001b[39mand\u001b[39;00m mask \u001b[39mis\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m    402\u001b[0m     mask \u001b[39m=\u001b[39m isna(values)\n\u001b[0;32m--> 404\u001b[0m result \u001b[39m=\u001b[39m func(values, axis\u001b[39m=\u001b[39maxis, skipna\u001b[39m=\u001b[39mskipna, mask\u001b[39m=\u001b[39mmask, \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs)\n\u001b[1;32m    406\u001b[0m \u001b[39mif\u001b[39;00m datetimelike:\n\u001b[1;32m    407\u001b[0m     result \u001b[39m=\u001b[39m _wrap_results(result, orig_values\u001b[39m.\u001b[39mdtype, fill_value\u001b[39m=\u001b[39miNaT)\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/pandas/core/nanops.py:719\u001b[0m, in \u001b[0;36mnanmean\u001b[0;34m(values, axis, skipna, mask)\u001b[0m\n\u001b[1;32m    716\u001b[0m     dtype_count \u001b[39m=\u001b[39m dtype\n\u001b[1;32m    718\u001b[0m count \u001b[39m=\u001b[39m _get_counts(values\u001b[39m.\u001b[39mshape, mask, axis, dtype\u001b[39m=\u001b[39mdtype_count)\n\u001b[0;32m--> 719\u001b[0m the_sum \u001b[39m=\u001b[39m values\u001b[39m.\u001b[39msum(axis, dtype\u001b[39m=\u001b[39mdtype_sum)\n\u001b[1;32m    720\u001b[0m the_sum \u001b[39m=\u001b[39m _ensure_numeric(the_sum)\n\u001b[1;32m    722\u001b[0m \u001b[39mif\u001b[39;00m axis \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m \u001b[39mand\u001b[39;00m \u001b[39mgetattr\u001b[39m(the_sum, \u001b[39m\"\u001b[39m\u001b[39mndim\u001b[39m\u001b[39m\"\u001b[39m, \u001b[39mFalse\u001b[39;00m):\n",
      "File \u001b[0;32m~/anaconda3/lib/python3.11/site-packages/numpy/core/_methods.py:49\u001b[0m, in \u001b[0;36m_sum\u001b[0;34m(a, axis, dtype, out, keepdims, initial, where)\u001b[0m\n\u001b[1;32m     47\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39m_sum\u001b[39m(a, axis\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, dtype\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, out\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, keepdims\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m,\n\u001b[1;32m     48\u001b[0m          initial\u001b[39m=\u001b[39m_NoValue, where\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m):\n\u001b[0;32m---> 49\u001b[0m     \u001b[39mreturn\u001b[39;00m umr_sum(a, axis, dtype, out, keepdims, initial, where)\n",
      "\u001b[0;31mTypeError\u001b[0m: can only concatenate str (not \"float\") to str"
     ]
    }
   ],
   "source": [
    "# merge the gsp and gpl data to remove any duplicates \n",
    "# repeating Gene Symbol values can have different IDs: for ex, STAT1 6772 repeats but the ID stays different\n",
    "# \n",
    "merged_meta_expr = pd.merge(metadata,expr, left_on= \"ID\", right_on=\"ID_REF\")\n",
    "\n",
    "# arrange mean from largest to smallest so that when we drop duplicates, the first occurance is kepy \n",
    "# the first occurance (with highest mean) will be the best one to keep\n",
    "merged_meta_expr[\"mean\"] = merged_meta_expr.mean(axis=\"columns\")\n",
    "sorted_merged = merged_meta_expr.sort_values(by=[\"mean\"],ascending=False)\n",
    "sorted_merged.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gf/79pl6dp1569b76cgzy_9yv7c0000gs/T/ipykernel_43792/3074067731.py:5: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.\n",
      "  no_mean_id = no_dupes.drop([\"mean\", \"ID\", \"ENTREZ_GENE_ID\"] ,1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>GSM258558</th>\n",
       "      <th>GSM258559</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>COX1</td>\n",
       "      <td>15.312190</td>\n",
       "      <td>15.335128</td>\n",
       "      <td>15.147944</td>\n",
       "      <td>15.235262</td>\n",
       "      <td>15.402517</td>\n",
       "      <td>15.235435</td>\n",
       "      <td>15.266640</td>\n",
       "      <td>15.236145</td>\n",
       "      <td>15.530658</td>\n",
       "      <td>...</td>\n",
       "      <td>15.354914</td>\n",
       "      <td>15.495646</td>\n",
       "      <td>15.392720</td>\n",
       "      <td>15.480795</td>\n",
       "      <td>15.372107</td>\n",
       "      <td>15.476776</td>\n",
       "      <td>15.125095</td>\n",
       "      <td>14.820450</td>\n",
       "      <td>15.409687</td>\n",
       "      <td>15.145438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19877</th>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>15.180497</td>\n",
       "      <td>15.180229</td>\n",
       "      <td>15.262090</td>\n",
       "      <td>15.246653</td>\n",
       "      <td>15.272874</td>\n",
       "      <td>15.181830</td>\n",
       "      <td>15.162696</td>\n",
       "      <td>15.126815</td>\n",
       "      <td>15.315813</td>\n",
       "      <td>...</td>\n",
       "      <td>15.165107</td>\n",
       "      <td>15.184177</td>\n",
       "      <td>15.130715</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>15.174592</td>\n",
       "      <td>15.192399</td>\n",
       "      <td>15.167122</td>\n",
       "      <td>15.298337</td>\n",
       "      <td>15.110260</td>\n",
       "      <td>15.088462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26931</th>\n",
       "      <td>IGK /// IGKC</td>\n",
       "      <td>15.394483</td>\n",
       "      <td>15.375848</td>\n",
       "      <td>15.259431</td>\n",
       "      <td>15.250159</td>\n",
       "      <td>11.996869</td>\n",
       "      <td>15.053283</td>\n",
       "      <td>15.450697</td>\n",
       "      <td>15.391052</td>\n",
       "      <td>14.658909</td>\n",
       "      <td>...</td>\n",
       "      <td>15.177085</td>\n",
       "      <td>15.370751</td>\n",
       "      <td>14.985706</td>\n",
       "      <td>14.771899</td>\n",
       "      <td>15.048471</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.678163</td>\n",
       "      <td>15.120340</td>\n",
       "      <td>15.349818</td>\n",
       "      <td>15.372265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22263</th>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>15.164475</td>\n",
       "      <td>15.337793</td>\n",
       "      <td>14.886320</td>\n",
       "      <td>14.723359</td>\n",
       "      <td>14.734361</td>\n",
       "      <td>14.755676</td>\n",
       "      <td>15.090716</td>\n",
       "      <td>14.747466</td>\n",
       "      <td>15.357297</td>\n",
       "      <td>...</td>\n",
       "      <td>14.861333</td>\n",
       "      <td>15.184035</td>\n",
       "      <td>14.716383</td>\n",
       "      <td>14.885592</td>\n",
       "      <td>14.950120</td>\n",
       "      <td>14.881242</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.600380</td>\n",
       "      <td>15.158579</td>\n",
       "      <td>15.059676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>ATP6</td>\n",
       "      <td>14.765942</td>\n",
       "      <td>14.939455</td>\n",
       "      <td>14.530669</td>\n",
       "      <td>14.956952</td>\n",
       "      <td>14.960515</td>\n",
       "      <td>14.767122</td>\n",
       "      <td>14.648069</td>\n",
       "      <td>14.847039</td>\n",
       "      <td>14.897657</td>\n",
       "      <td>...</td>\n",
       "      <td>14.963645</td>\n",
       "      <td>15.086318</td>\n",
       "      <td>14.994576</td>\n",
       "      <td>15.079311</td>\n",
       "      <td>14.952733</td>\n",
       "      <td>15.047349</td>\n",
       "      <td>14.124653</td>\n",
       "      <td>14.317633</td>\n",
       "      <td>14.954273</td>\n",
       "      <td>14.737609</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Symbol  GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  \\\n",
       "914            COX1  15.312190  15.335128  15.147944  15.235262  15.402517   \n",
       "19877        EEF1A1  15.180497  15.180229  15.262090  15.246653  15.272874   \n",
       "26931  IGK /// IGKC  15.394483  15.375848  15.259431  15.250159  11.996869   \n",
       "22263        TMSB4X  15.164475  15.337793  14.886320  14.723359  14.734361   \n",
       "936            ATP6  14.765942  14.939455  14.530669  14.956952  14.960515   \n",
       "\n",
       "       GSM258556  GSM258557  GSM258558  GSM258559  ...  GSM258599  GSM258600  \\\n",
       "914    15.235435  15.266640  15.236145  15.530658  ...  15.354914  15.495646   \n",
       "19877  15.181830  15.162696  15.126815  15.315813  ...  15.165107  15.184177   \n",
       "26931  15.053283  15.450697  15.391052  14.658909  ...  15.177085  15.370751   \n",
       "22263  14.755676  15.090716  14.747466  15.357297  ...  14.861333  15.184035   \n",
       "936    14.767122  14.648069  14.847039  14.897657  ...  14.963645  15.086318   \n",
       "\n",
       "       GSM258601  GSM258602  GSM258603  GSM258604  GSM258605  GSM258606  \\\n",
       "914    15.392720  15.480795  15.372107  15.476776  15.125095  14.820450   \n",
       "19877  15.130715  15.191393  15.174592  15.192399  15.167122  15.298337   \n",
       "26931  14.985706  14.771899  15.048471  15.236298  14.678163  15.120340   \n",
       "22263  14.716383  14.885592  14.950120  14.881242  15.236298  14.600380   \n",
       "936    14.994576  15.079311  14.952733  15.047349  14.124653  14.317633   \n",
       "\n",
       "       GSM258607  GSM258608  \n",
       "914    15.409687  15.145438  \n",
       "19877  15.110260  15.088462  \n",
       "26931  15.349818  15.372265  \n",
       "22263  15.158579  15.059676  \n",
       "936    14.954273  14.737609  \n",
       "\n",
       "[5 rows x 59 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dropping the duplicates-- keeping highest mean expression values \n",
    "no_dupes = sorted_merged.drop_duplicates(\"Gene Symbol\")\n",
    "\n",
    "# dropping other unecessary columns \n",
    "no_mean_id = no_dupes.drop([\"mean\", \"ID\", \"ENTREZ_GENE_ID\"] ,1)\n",
    "\n",
    "no_mean_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'no_mean_id' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[5], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m# resetting index values by setting Gene Symbol column as index \u001b[39;00m\n\u001b[0;32m----> 2\u001b[0m reset_index_df\u001b[39m=\u001b[39m no_mean_id\u001b[39m.\u001b[39mset_index([\u001b[39m\"\u001b[39m\u001b[39mGene Symbol\u001b[39m\u001b[39m\"\u001b[39m])\n\u001b[1;32m      3\u001b[0m reset_index_df\n",
      "\u001b[0;31mNameError\u001b[0m: name 'no_mean_id' is not defined"
     ]
    }
   ],
   "source": [
    "# resetting index values by setting Gene Symbol column as index \n",
    "reset_index_df= no_mean_id.set_index([\"Gene Symbol\"])\n",
    "reset_index_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to plot and cluster\n",
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns; sns.set()\n",
    "import numpy as np\n",
    "from sklearn.cluster import KMeans\n",
    "from sklearn.datasets._samples_generator import make_blobs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(200, 2)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAhkAAAGgCAYAAAAHN3WrAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjguMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy81sbWrAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAuLklEQVR4nO3dfXRV1Z3/8c+9eSCXlJgEvDzIwwiliQ8UO2CBAWuri+mStjPVtVyOHVm2A2If1M5Qta2ozBSlTmVA20pblY6rdagMdhZOfRgHO3ZVWYqK1XEVGxFR0CLXCbmkQiD36fcHv8Qk3CT3nHv2Pfuc8379o9zk3uzsJHd/zt7fvU+sUCgUBAAA4LG43w0AAADhRMgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYES13w0oFArK5zkPTJLi8Rh94RB95g795hx95g795lwQ+iwejykWiw37eb6HjHy+oIMHD/vdDN9VV8fV1FSvzs4jymbzfjcnEOgzd+g35+gzd+g354LSZ83N9aqqGj5ksFwCAACMIGQAAAAjCBkAAMAIQgYAADCCkAEAAIwgZAAAACMIGQAAwAhCBgAAMIKQAQAAjCBkAAAAIwgZAADACEIGAAAwgpABAACMIGTAiFS6S9te2a9UusvvpgAAfOL7rd4RPql0l27esF3dmbxqa+L6zpI5SjYm/G4WAKDCmMmA53btS6s7k5ckdWfy2rUv7W+DAAC+IGTAc9MnNaq25vivVm1NXNMnNfrbIACAL1gugeeSjQl9Z8kc7dqX1vRJjSyVAEBEETJgRLIxQbgAgIhjuQQAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIgCSOAQcAeI8trOAYcACAEcxkgGPAAQBGEDLAMeAAACNYLgHHgAMAjCBkQBLHgAMAvMdyCQAAMIKQAQAAjCBkAAAAIwgZiBwOHgOAyqDwE5HCwWMAUDnMZMAYG2cMOHgMACqHmQwYYeuMQc/BYz3t4uAxADCHkAGl0l2eH8RVbMbAhpDBwWMAUDmEjIjzesahJ7A0n1Rn7YwBB48BQGUQMiLOyxmHgYHl6xfP1MFDRyM3Y2BiZggAgoiQEXFe1igMDCwHDx3V/BnjPWppMNhaiwIAfiBkRJyXNQoUVdpbiwIAfiBkwLMaBYoqCVoA0FcoQwZr4t4rtU+jXlRJ0AKAD4QuZLAm7j361JmoBy0A6BG6Ez850dF7JvrUxtNAAQDechwyMpmM1q1bp09+8pP62Mc+pi984Qt68cUXTbTNlZ41cUlWrImHYTD1uk97ZkY2PPKqbt6wPdB9AwAYnOPlkh/96Ef65S9/qdtuu02TJk3SPffcoyuuuEKPPvqoxo4da6KNjti0Jt53maGmOqbPnzNVs1qSgZtK97pP2YEBANHgeCbj17/+tT772c9qwYIFmjJlir71rW/p/fff10svvWSgee4kGxOaP2O87wNX38E0ky1o85O7A3vl7mWfmpptGmrWaLCPhWGmCQBs5Xgmo7GxUU8++aQuu+wyjR8/Xps2bVJtba1OO+00942oDl1piCQp2XTigNydyWv3O4c0YUx9v8erquL9/htmE8bUa/WyuWrbm1bL5EYlm0a6ep2+fZbqONKvOHX1srm9rzvYx4Z6TphF6XfNK/SZO/Sbc2HrM8chY8WKFfqHf/gHnX/++aqqqlI8Htedd96pyZMnu2pAPB5TU1P98J8YQF2vt5/w2IiaKn18xoRBv+eGhmgsGzQ11atl6smevFZDQ0LPt73Xbwnm7fau3td/8fX2oh8b7PGoiMrvmpfoM3foN+fC0meOQ8bu3bvV0NCgu+66S2PHjtXmzZv1zW9+U/fff79aW1sdNyCfL6iz84jj5wXBxNGJ3oOZaqrjuugTU3X2aUmNiEsdHYf7fW5VVVwNDQl1dnYpl8v71OLSpTqOuJqJcPu8Yvr2Wd++rq2Ja+LoRG8fD/axoZ5jipffv1tB+12zAX3mDv3mXFD6rKEhUdJsS6xQKBRKfdF33nlHn/70p3Xfffdp9uzZvY9/4QtfUFNTk+666y7HDc3l8jp40Owbu59KPcSqujqupqZ6dXQcVjZr7y+W5P7cDK/P2xjYZ0P19WAfq+TBbbacNxKk3zVb0Gfu0G/OBaXPmpvrSwoZjmYy/vd//1eZTEYzZszo9/jMmTP129/+1lkLIyKMBzPtaEu52h1ielfJUH3d87GeQs+eUFHJnw+7agBEjaPKkvHjj99Rs62trd/jr732mqZMmeJdq2CtVLpLW57a0/vvmupYybtDKn2GycCdI36fz2HbGS4AYJqjmYyPfvSjmj17tr75zW9q5cqVGjdunLZs2aJnnnlGGzduNNVGWGTXvrQyfabwPn/O1JKvxit5hkmxpQm/ZxJsOsMFACrBUciIx+Nav3697rjjDn3729/WoUOH9JGPfET33XefzjrrLENNhE0G3mV0VkuypOf1rX2YP2O84VYWX5qw4Q6pYVw+A4DBOCr8NCHshZ+lCkqxj+S8WNJUweNQfZZKd+mme59VJltQTXVMq5bO7a3JiPpMQpB+12xBn7lDvzkXlD4zUvgJSM6vxiuxTFE8PMQkFf7/f49jJgEAKoeQgZK5nQUwvUwx8OTOnrqHntqRTLZ/sGE2AwAqg5ARYEE546Gn4HFHW8pI29r2ll5/YctZFQAQBYSMgKr0YOnFksdDT+9Rdyavh57e42l7WyafGCgG28nh9w4TAIgSQkZAVXqwdLLkUWyGxWR7k00jiwaKYvUXNuwwAYCoIGQYUIlljEoPlqWe8TDYDIvp9pZa0Onk+6BuAwDKQ8jwWKWWMUwc7DTcwFrKQD7YjIVNB1EN931QtwEA3iBkeKySyxhebsf0amAdasYiKNtHqdsAAG8QMjwW1DV/rwZWm2Ys3ArqzxAAbEPI8FhQB1kvB9agzFgMxunPkPoNACiOkGFApQdZLwa5csJRGAfZUn+G1G8AwOAIGQFX7iA3MCA4HSCjPshSvwEAgyNkBFw5g5wXASHqgyz1GwAwOEJGwJUzyHkREEwOskFYhglqDQ4AVAIhI+DKGeS8CAimBtnjt2rfrkw2r5rquFYttXcZJuiFrgBgCiGjiCBcQffldpDzKiCYGGR3tKX63UV1R1tKF8yZ4unXAACYRcgYIGqFjFyFAwBMifvdANsUq1OImlS6S9te2a9Uusvoc4YyqyWpmuqYJKmmOqZZLUlPXhcAUDnMZAwQ9d0CbmZyTMz+JBsTWrV0rifLVql0l3a0pSQdDy/M3ABAZRAyBojqboGeOpTOI92Od5yY2sbqxVLO8QLSZ5XJFiRJW57aY3URKQCECSGjiKjVKfSdiaipjqumOqZMtlDyTI7Nsz+79qV7A4Z0vIg0amd5AIBfCBnoNxORyeZ18aemqWFkbckzOTbP/kyf1NgbmiSpptquEAQAYUbIwAkzEW7qFmyd/emp7aAmAwAqj5ARUqWc9dH3c2ydifBCsjHBGRsA4ANCRgiVstuj2OfMnzHepxYDAMKIczJCqJSzPsJ8HojXZ3YAANxhJiOEStntUe6OEFuPXo/aia0AYDNCRgiVstujnB0hXgzkpkJK1G89DwA2IWSEVCm7PdzuCCl3IDc522DzmR0AEDWEDDhW7kBucrbB5jM7ACBqCBlwrNyB3PRsg61ndgBA1BAy4Eo5AzmzDQAQDYQM+ILZBgAIP87JAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAY4SpkbNmyRYsWLdKMGTP0mc98Ro899pjX7QIAAAHnOGQ89NBDuuGGG3TJJZfo4Ycf1qJFi7R8+XL97ne/M9E+AAAQUI5CRqFQ0J133qnLL79cl19+uaZMmaKvfe1r+ou/+As999xzptoIAAACqNrJJ7/xxht655139LnPfa7f4xs2bPC0UQAAIPgchYw333xTknTkyBEtWbJEO3fu1MSJE/WVr3xF5513nvtGVFN/WlUV7/dfDI8+c4d+c44+c4d+cy5sfRYrFAqFUj/5oYce0vXXX6+JEyfqqquuUmtrqx5//HH9+Mc/1r/+679q3rx5jhtQKBQUi8UcPw8AANjN0UxGTU2NJGnJkiW68MILJUmnnXaadu7c6Tpk5PMFdXYecfy8sKmqiquhIaHOzi7lcnm/mxMI9Jk79Jtz9Jk79JtzQemzhoZESbMtjkLGuHHjJEkf+chH+j3+4Q9/WL/5zW+cvFQ/2ay9HVlpuVye/nCIPnOHfnOOPnOHfnMuLH3maNHn9NNPV319vV5++eV+j7/22muaPHmypw0DAADB5mgmo66uTkuXLtVdd92lsWPH6qMf/ageeeQRbdu2Tffdd5+hJgIAgCByFDIk6atf/aoSiYTWrVunAwcOaNq0afrBD36gOXPmmGgfAAAIKMchQ5K+9KUv6Utf+pLXbQEAACESjo24AADAOoQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAJ0ilu7Ttlf1Kpbusfk0Adqv2uwEA7JJKd+nmDdvVncmrtiau7yyZo2RjwrrXhHdS6S7t2pfW9EmN/FzgKUIGgH527UurO5OXJHVn8tq1L132wGPiNeENAiBMYrkEQD/TJzWqtub4W0NtTVzTJzVa+ZqVkkp36amX/6h32w/73RQjigVAwCvMZADoJ9mY0HeWzPF0+tzEa1ZC36v8n/1Xm25dNkfNo+r8bpanegJgz0xGkAIg7EfIAHCCZGPC8yBg4jVN63uVfyyTU9vetOadMc7nVnkrqAEQwUDIAIBB9L3KH1FTpZbJjX43yYggBkAEAyEDAAbRc5W/+51D+viMCRoRl7LZvN/NAgKDkAEAQ0g2JjRhTL2amurV0RHO4k/AFHaXAIBDYT1YLKzfF/zDTAYAOBDWcyXC+n3BX8xkAIADYT1XIqzfF/xFyAAAB4J8sNhQwvp9wV8slwAINa/vyxHWcyXC+n3BX4QMAKFlqs4gKOdKOA1YQfm+EByEDAChFeUbs1HICRtQkwHAF5XYLhnlOgMKOWEDZjIAVFylrrKjXGfAjc9gA0IGACOGqgeo5DKGyToDr4tKvRTlgAV7EDIAeK7YTIWk3gEvDFfZQah5oJATfiNkAPDcwJmKHW0pPfT0nn4DctCvsqNcVAqUipABwHN9Zyqqq2J6vytzwoA8f8b4QA/KYZiNAUwjZADwXLIxoa9fPFNrN72kbK6gJ17Yp5rquDLZ8AzI1DwAwyNkADDi4KGjyuUKkqRMtqCLPzVVDSNrQzUgD1bzYHNBKFBJhAzAQmEYpAYuJ8xqSQb2e3EiCAWhQKUQMgDLhGWQiupyAgWhwAcIGYBlwjRIRXELJQWhwAcIGYBlbBukwrB0U0lRncEBiiFkAJaxaZAKy9JNqfoGKumDw8MmjKkv+jmD9UUUZ3CAYggZgIVsGaT8WrrxY/akb6CqqY5LKiiTLai2Jq7Vy+aqqaleqY4jvZ9TVRXT8kvO0mmTmyrSPiCIuAsrgEH5cRfTnsF+wyOv6uYN243epbWvvoEqk80rkz2+/bY7k1fb3rTebT+sX217s/dzcrmC1m16qaT2VeKOs4CNmMkAMCg/lm78mj3pWwszcCZjzEl1uur2J3Usk+v3nGyuMGz7orbkBPRFyAAwpEov3fhV+DowUEkf1GTsfudQv4ARj8WULxRKal+YdgsBThEyAFjF1OyJm4LNnv+vroppRE2VjmVyqq2J6+sXz9TBQ0dLap/T0JRKd2lHW0qSInOAGcIrVigUCn42IJfL6+DBw342wQrV1XE1NdWro+Owstm8380JBPrMnSj2W7lLFtXVcR3LS8+98kdNO+UkxwN/qYWsqXSXbrr32d56kJrquFYtDe7yShR/18oVlD5rbq5XVdXwZZ0UfgKwiokiyWK3nndq3Oh6nTNzgqsBP9mYKOmus7v2pXsDhnS8AHXXvrTjrwfYguUSANYwVSQ5fVJj711gJWnLU29YuRRxvJ2xfjMZfh/GBpSDkAHAGqaKJJONCX3+nFO1+cndko7fFdbGAsxkY0Krls6lJgOhQcgAYA2TO0tmtST10NN7rDmufTDJxoQumDPF72YAniBkALCGyXM5ynntVMcRvfh6uyaOTqh5VJ1nbTLF6Ymp3J8GphAyAAyrkoOQyXM53Lx20A7TctreoH1/CBZCBoAhBX0QKjcgBe0wLaftDdr3h2AhZAAYko2DUE9waD6pbshDsbwISH6dQOqW0/YG7ftDsBAyAAzJtkGob3DoMViA8CIgJRsTWr1srt5u7+pXk2FrHYPT2hM/7k+D6HAdMvbs2aOLLrpIN910ky666CIv2wTAIrYNQn2DQ4++AaLv4O9VQEo2jVTL1JN7T2G0fQnJae1Jpe9Pg+hwFTIymYyuvfZaHTlyxOv2ALCQTYNQ3+DQoydAFBv8TQQkG5eQABu5Chk/+MEPVF9f73VbAKDXYMsRfWdWBtZkbHtl/wmDfynHeTtl2xISYCvHIeP555/Xpk2btGXLFn3yk5800CQAUTfccsRgMyuVGvxtW0ICbOUoZHR2dur666/XjTfeqPHjx3vXiGru09ZzN7tS7mqH4+gzd4LQb7vfOdRvRmL3O4c0Yczws6cTxtRr9bK5atubVsvkRiWbRnrSnmJ9NmFM/bBtSnUc8bwtQRKE3zXbhK3PHIWMf/zHf9RZZ52lz33uc541IB6PqanJ3NLLu+2H9fs32nXG1NEaN9r+JZ6GBq6InKLP3LG53z4+Y4J+9l9tOpbJaURNlT4+Y0LJ7xNNTfVqmXqyJO///p302bvth7Xi7u2938MPr/uU6zYE7X1sIJt/12wVlj4rOWRs2bJFL7zwgn71q1952oB8vqDOTjMFpKmOI7rh7md7p05XL5tr7dVEVVVcDQ0JdXZ2KZfLD/8E0GcuBaHfRsSlW5fN6Z0FGBGXOjoOO3oNL//+3fTZc6/8UccyOUnSsUxOz73yR50zc4Ljrx2k97GBgvC7Zpug9FlDQ6Kk2ZaSQ8Yvf/lLtbe3n1CHsXLlSm3YsEGPPPKI40b2yGbNdOSrb3b0m3J99c0O6+87kMvljfVHWNFn7tjeb82j6jTvjHGS3L1HmPj7H67P+harTjvlpH71IdNOOcma76PSbP9ds1FY+qzkkLFmzRodPXq032N/+Zd/qWuuuUaLFi3yvGFeoAIciK5K//2b2j7L+xiCrOSQMXbs2KKPjx49WqeccopnDfISFeBAdFX677/Y2RlebJ/lfQxBFvpjxW06RAhAZVXy79/kjAPvYwiqskJGW1ubV+0AgEBjxgE4UehnMirN1psmATCPGQegP0KGh2y/aRIQBQR9wB6EDA9V4qZJvIECg4t60Of9AbYhZHjI9FazqL+BAsOJ8t1ReX+AjQgZHjJd+BXlN1CgFFE+U8Kv9wdmTzAUQobHTBZ+RfkNFChFlHd4+PH+wOwJhkPICJAov4EinExcBUd1h4cf7w/lzp4wCxJ+hIyACeMbKG800cRVsPcq/f5QzuwJP/9oIGT4hIH1ON5ooitKNUZh/XsvZ/YkSj//KCNk+KDYwDphTL3fzfIFbzTRFZUao7AHabezJ1H5+UcdIcMHxQZWP0OGn1dZvNFEV1RqjAjSxUXl5x91hAwf2DSw+n2V5eUbTVinpMOslKvgoP9cbfp7t00Ya8zQHyHDBzYl+IFXWTvaUrpgzpSKtsGLNxq/wxLMCMPP1aa/d6DSCBk+sSXBT5/UqJrquDLZ40Fjy1NvaFZL0oq2OcGUdDiF5edqy987UGlxvxsAfyUbE/r8Oaf2/juTLWjXvrR/DXKpZ0paElPSIcLPFQg2ZjKgWS1JPfT0nkCvGTMlHU78XIFgI2RUiM3Fa2F5I2dKOjwG/r3wcwWCiZBRAUEoXuONHLYIwt8LgNJQk1EBxYrXoiCV7tK2V/Yrle7y5PMQDVH9ewHCiJmMEpS71BGFffID+6jUq1GuWjFQFP5egKggZAzDi0EwDDUPQwWtYn1U6tbDsGxRhHfC8PcC4DhCxjC8GgSDXPMwXNAq1kelXo1y1Ypigvz3AuADhIxhBHEQ9Hony3BBq1gflXo1ylUrAIQXIWMYQRkEe4JF80l1unPzy2XXOPQNKsMFrcH6qNSr0TBctdq8RRkA/ELIKIHtg2Df5YzqqpiyuYKk47MOjz7zphbN+zNH7S+2PDJc0LK9j0yieBUAimMLawj0Xc7I5gqqqor1fuy3L+/XzRu2O9oeOtjyyPwZ430bPG3e5sqWSwAojpARAgPv77D8krP0iZnjez/udOAb7n4RlR7we2YKNjzyquPAVAncXwMAiovcckkY186L1USMbqjTszsPuCpYHaoOxY+lAdu3uQalbgcAKi1SISPMa+cDayLKHfgGq7HwY8APwg6fKNekAMBgIhUyyh0ggzYLMnDg86L95Qz4br8+MwUAEEyRChnlDpBBngXxqv1uB/xyvz4zBQAQPJEKGeVcEdtaF1Dq7ICX7Xcz4Nvaf34J2qwYALgRqZAhub8itrEuwMnsgMn2lzJgNp9U13uGhy3955egz4oBQKkiFzLcsrEuwMnsgKn2lzJgptJdunPzy71neHz94plW9J9fmNUBEBWEDAdsqwtwOjthov2lDJh9PyeXK+jgoaOetiFobJwVAwATCBkB99cLTpUkzWpJ+hKAShkwGVT7s3FWDABMIGQE1MBlilktybJeq9QBb+DnljJgMqieyLZZMQAwgZBhiVTHEb34ersmjk6oeVTdsJ/v1bq+kyLEwT63lAGTQRUAooeQYYFUuks33fusMtmCaqpjWrV07pADcirdpT+2H1Y8LuXz5d0vw0lYoWARAOAEN0grg1c3CtvRllIme/z27JlsQTvaUkN+zZvu3a7Hnt2rfF6Kx1TWbg0nN/cycSOwYn1o8x1XAQClYybDJafLDF7VI+zal1Ymm+/9d74gvbm/U6dNbnL1ek7qJbyurSjWh5I4QwIAQoKQ4VKpSwelhJFZLUlteWqPMtm8aqqHLuKcPqlRVfGYcvmCZ9+Lk3qJcmorBoatYn3Y8/99HyNkAEAwETJcKnVbZilhJNmY0HevnKu327uGLfxMNia0/G/O0toHfqdcXqqpjpW1s6RSioWtwfqQ7a4AEA6EDJdKXTooNYwkm0aqZerJ6ug4rGyf5ZBiTpvcpFuXzfN0S6jpe2kUC1vzZ4wv2odsdwWAcCBklKHUrZsmBk0vt4RW4l4ag4WtYt8H210BIBwIGRVg+6BZia2pHMgFANFDyEDFjv22PWwBALxFyACzDAAAIwgZkMQsAwDAe5z4CQAAjCBkAD7iCHUAYcZyCeCTSmwdBgA/MZMB+GSwY9UBICwIGYBPTNzVFgBswnIJ4BO2DgMIO0KGh0zf/wPhw9ZhAGFGyPBIJYr4CDEAgCAhZHjExP0/+oYKSexEAAAECiHDI17f/2PgzMhfLzjV+E3MAADwEiHDI14X8Q2cGZFUkZuYAQDgFUKGh7ws4hs4MzKrJalZLUlqMgAAgeEoZKTTaa1du1a/+c1v9P7776ulpUXf+MY3NHv2bFPti6zBZkbKCRcUjgIAKslRyFi+fLna29u1du1aNTc3a+PGjVqyZIn+4z/+Q9OmTTPVxsjycmaEI6wBAJVW8omfb731lrZt26aVK1dq9uzZmjp1qlasWKGxY8fq4YcfNtlGeIAjrAEAlVZyyGhqatLdd9+tM888s/exWCymQqGgQ4cOGWkcvMMR1gCASit5uaShoUHnnntuv8cee+wx7d27VwsWLCivEdXcQqWqKt7vv16bMKZeq5fNVdvetFomNyrZNNLI16kk030WVvSbc/SZO/Sbc2Hrs1ihUCi4eeKOHTu0dOlSzZs3T+vXr3fdgEKhoFgs5vr5AADATq5CxhNPPKFrr71WM2fO1E9+8hPV1dW5bkAul1dnZ5fr54dFVVVcDQ0JdXZ2KZfL+92cQKDP3KHfnKPP3KHfnAtKnzU0JEqabXF8Tsb999+vW2+9VQsXLtSaNWtUW1vrqoF9ZbP2dmSl5XJ5+sOhcvosytt6+V1zjj5zh35zLix95ihkbNy4UatWrdLixYt1ww03KB4Px5oRooltvYgSt4E6ykEc5Ss5ZOzZs0erV6/WwoULdeWVV6q9vb33Y3V1dRo1apSRBgKmmLipHWAjt4G67/NqquP6/DmnalZLkr8TlKzkkPH4448rk8lo69at2rp1a7+PXXjhhbrttts8bxxgktc3tQNs5TZQ931eJpvX5id366Gn9zDrh5KVHDK+/OUv68tf/rLJtgAV5fVN7QBbuQ3UfZ/Xg1k/OMEN0hBpXh7dDtjKbaDued6OtpS2PPWGMtmC41m/d9sP67lX/qhpp5zE31oEETIAIALcBupkY0IXzJni6i7QqY4jWnH3dh3L5CiujihCBgBgWG5CStvetI5lcpJYZokq9qACQASl0l3a9sp+pdLmDkNsmdyoETVVkrhnUlQxkwEAEVOpM2KSTSP1w+s+RU2GQ2GqYyFkAEDEVPKMmHGj63XOzAmhOL2yEsJWx8JyCQBETM/WVIllDNsUq2MJMmYyACBignBGTFSPM++pY+mZyRgqAAahjwgZMCIIv/xAlNl8RkyU7ytUah1LUPqIkAHPBeWXH4Cdon5foVLqWILSR9RkwHPFfvkBoFTUjAwvKH3ETAY8x43HAHsEcekyCDUjfgtKHxEy4Lmg/PIDYRfkpUuba0ZsEYQ+ImTAiCD88gNhF5R1e4QXNRkAEFJBWbf3SiWOSg8SG/qDmQwACKkoLV0GeWmoRyrdpd3vHNLHZ0zQiDKnAGzpD2YyACAg3FyZJhsTmj9jfOAGXKeG2tVmwxX9cHpCwT2/2qmrbn9SqY4jZb2eLbv8mMkAgACw5cq0GBt2sAy2q83mfuurbyg4lsmpbW9a884Y5/r1bNnlR8gAgACwtYjTlkF8sKUhW/ttoL6hYERNlVomN5b1erYslREyACAASrkyLWdGwe1zbRrEi+1qc3NF78fMTE8o6FuTUe6da23Y5UfIAIAAGO7KtJwZhXKe68W0vMlB3ekVvZ8zM8nGhCaMqVdTU706Og5X5GuaRsgAgIAY6srUyYzCwEG9nNmIcqflKzGoO7mit2lmJgwIGQAQAqXOKBQb1MudjShnWt62Qd2WgsmwIGQAgM+8WC4odUah2KA+f8Z434oEbRvUbSmYDAtCBgD4yMvlglJmFAYb1P0qEjQxqJcb2mwomAwLQgYA+KjSywU2Xql7Oag7CW02nO8RdoQMAPCR6a2pxYT5Sr3U0EYYqQxCBgD4yOTW1CgqtcbDRBjBiQgZAOAzr7amovTlIK/DCIojZACAxWzbfREEpSwHeR1GUBwhAwAsVspyCvUC7vS9v0nffw/8HNsKZYOEkAEAlhvsytyreoHhgkpYg0yp/RfmQlnTCBkAEFBe1AsMN9CGufCRegvz4n43AADgTk+9gCTX9QLFBlonHw8yL/oPQ2MmAwACyot6geaT6lRdFVM2Vyg60Npc+OjFyZ5fv3imtv/+Xc05YxyzGAYQMgAgwMqpF0ilu3Tn5peVzRVUVRXT1y+eecJr2Vr42HcZp6Y6rs+fc6pmtSQdta/n++/O5PXszgOhWgqyBcslABBRfZdCcrmCDh46WvTzko0JzZ8xvncATqW7tO2V/UqluyrW1oH6tj2TzWvzk7t184btjtoU5qUgWzCTAQAR5WYpxJZC0L5t7+G0eNPmpaCwIGQAQES5WQqxZUdGT9t3tKW05ak3lMkWrykp5TVsWwoKE0IGAASQV2dXOK3psOnqP9mY0AVzpmhWS9J1X3AGhlmEDACwWLEw4eeShY1X/wQFexEyAMBSg4UJv5csGNRRKnaXAIClBtv9wCFSCApmMgDAUoPVP9i4ZFFMquOIXny9XRNHJ9Q8qs7v5kgK731YbEXIAABLDRUm3C5ZVGqQtWWrq+1tCjtCBgBYzMv6h0oOsgOXena0pdQwstbXGQS/a1miiJABABExcJB99Jk3tWjenxkZaPsu9dRUx7TlqT3KZO05wItalsogZABARAw8JfO3L+83ds+OZGNCq5fN1dvtXdr/3p+06devS7LjAC9qMiqH3SUAEBE9g+wnZo7vfczkPTuSTSN1/tmTdXZr0prdMAPvwwKzmMkAgAhJNia0aN6f6dmdByq2bJBsGskMQkQRMgAgYvxYNuAAr2giZABABAVp0E+lu7SjLSVJmtWSrFi7OVOjfIQMAIC1Uuku3XTvdmWyx4tVtzz1hlYtnWt80OdMDW9Q+AkAMCrVcUTbXtmvVLrL8XN37Uv3BgxJymQLxgpVB37dYke6wxlmMgAAxrzbflg33P2s6xmB6ZMaVVMd7w0aNdWxiuxO4UwNbxAyAADG/P6N9rJO2Uw2JrRq6ZyK12RwpoY3CBkAAEecFESeMXV02TMCycaELpgzxWVr3QtScaytCBkAgJI5LYgcN7peq5fN1atvdjAjEEGEDABAydzcZCzZNNKaW72jsthdAgAoWU9BpOT/EeGwHzMZABAipg+QoiASTjgOGfl8Xj/84Q+1efNmdXZ2atasWVq5cqWmTKl8UQ4A4AOVOkDK1oJIv04GxeAcL5esX79eDzzwgG655RZt2rRJsVhMV1xxhbq7u020DwBQoigfIHX8ZNBntfnJ3dr85G7ddO92V4d/wVuOQkZ3d7d++tOf6uqrr9a5556r1tZWrVu3TgcOHNDWrVtNtREAUIIo10scPxm00PvvTDZaIctWjpZL/vCHP+jw4cOaO3du72MNDQ06/fTT9fzzz+szn/mM5w0EAJQmyvUSx08GjfUGjZrqaIUsWzkKGe+++64kafz48f0eTyaT2r9/v/tGVLPJpaoq3u+/GB595g795lyQ+mzCmHpNGFPvdzMkVbbfJoyp13evnKfnX01JMens1qSSTSONf12vBel3rRSOQkZX1/H1rdra2n6PjxgxQocOHXLVgHg8pqYmO/4gbNDQEJ0rD6/QZ+7Qb87RZ+5Uqt+amurVMvXkinwt08Lyu+YoZNTVHT9Mpbu7u/f/JenYsWNKJNx1SD5fUGfnEVfPDZOqqrgaGhLq7OxSLpcf/gmgz1yi35yjz9yh35wLSp81NCRKmm1xFDJ6lklSqZQmT57c+3gqlVJra6vDJn4gm7W3Iystl8vTHw7RZ+7Qb87RZ+7Qb86Fpc8cLfq0trbqQx/6kLZv3977WGdnp3bu3KnZs2d73jgAABBcjmYyamtrddlll2nNmjVqbm7WKaecottvv13jxo3TwoULTbURAAAEkOMTP6+55hpls1ndeOONOnr0qM4++2xt2LDhhGJQAAAQbY5DRlVVla677jpdd911JtoDAPCB6XueIJq4QRoARFyl7nmC6AnHaR8AANeifM8TmEXIAICIi/I9T2AWyyUAEHFRvucJzCJkAACUbEwQLuA5lksAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGxAqFQsHPBhQKBeXzvjbBGlVVceVyeb+bESj0mTv0m3P0mTv0m3NB6LN4PKZYLDbs5/keMgAAQDixXAIAAIwgZAAAACMIGQAAwAhCBgAAMIKQAQAAjCBkAAAAIwgZAADACEIGAAAwgpABAACMIGQAAAAjCBkAAMAIQgYAADCCkAEAAIwgZFhm//79Wr58uebPn6+zzz5bS5Ys0a5du/xuVqCsWLFC3/rWt/xuhnXy+by+//3v65xzztHMmTP1d3/3d3rrrbf8blagrF+/XosXL/a7GdZLp9O6+eab9YlPfEJ//ud/rksvvVQvvPCC382yWnt7u6677jrNnTtXH/vYx7Rs2TK9/vrrfjerbIQMi3R3d2vZsmVqb2/XT37yE23cuFGjRo3S5ZdfroMHD/rdPOvlcjn98z//sx588EG/m2Kl9evX64EHHtAtt9yiTZs2KRaL6YorrlB3d7ffTQuE++67T9///vf9bkYgLF++XC+//LLWrl2rBx98UGeccYaWLFmi3bt3+900a33lK1/Rvn37dM899+jBBx9UXV2dvvjFL6qrq8vvppWFkGGRF154Qa+99pq+973v6cwzz9T06dP1ve99T0eOHNH//M//+N08q+3evVuXXnqptmzZogkTJvjdHOt0d3frpz/9qa6++mqde+65am1t1bp163TgwAFt3brV7+ZZ7cCBA1q6dKnuvPNOnXrqqX43x3pvvfWWtm3bppUrV2r27NmaOnWqVqxYobFjx+rhhx/2u3lW6ujo0MSJE7Vq1SrNmDFD06ZN01e/+lW99957gZ/JJmRYZPr06br77rs1duzYfo8XCgUdOnTIp1YFw3PPPafTTjtNDz/8sCZOnOh3c6zzhz/8QYcPH9bcuXN7H2toaNDpp5+u559/3seW2e/3v/+9TjrpJP3nf/6nZs6c6XdzrNfU1KS7775bZ555Zu9jsViM97EhNDU1ae3atZo+fbok6f/+7/+0YcMGjRs3Th/+8Id9bl15qv1uAD5w8skn69xzz+332M9+9jMdO3ZM8+fP96lVwXDppZf63QSrvfvuu5Kk8ePH93s8mUxq//79fjQpMM477zydd955fjcjMBoaGk54H3vssce0d+9eLViwwKdWBcdNN92kf//3f1dtba1+9KMfaeTIkX43qSyEjAp6++23df755w/68aefflonn3xy77//+7//W+vWrdPixYvV2tpaiSZayWm/4UQ967q1tbX9Hh8xYgRXlzBqx44duuGGG3T++ecT1kpw+eWX65JLLtEvfvELfe1rX9PGjRt1xhln+N0s1wgZFTR27Fg9+uijg368ubm59/9/8YtfaNWqVVq0aJG+/e1vV6J51nLSbyiurq5O0vHajJ7/l6Rjx44pkUj41SyE3BNPPKFrr71WM2fO1Nq1a/1uTiD0LI+sWrVKL730ku6//35997vf9blV7hEyKqimpkbTpk0b9vPWrFmje+65R4sXL9aKFSsUi8Uq0Dp7ldpvGFzPMkkqldLkyZN7H0+lUpGeJYM5999/v2699VYtXLhQa9asOWEWDR9ob2/XM888owsuuEBVVVWSpHg8rmnTpimVSvncuvJQ+GmZ22+/Xffcc4+uv/563XjjjZEPGPBGa2urPvShD2n79u29j3V2dmrnzp2aPXu2jy1DGG3cuFGrVq3S3/7t3+qOO+4gYAwjlUrpG9/4hp577rnexzKZjHbu3Bn4CyxmMiyyfft23XvvvVq8eLH+6q/+Su+9917vx0aOHKn6+nofW4cgq62t1WWXXaY1a9aoublZp5xyim6//XaNGzdOCxcu9Lt5CJE9e/Zo9erVWrhwoa688kq1t7f3fqyurk6jRo3ysXV2am1t1YIFC/RP//RPuuWWW9TQ0KAf//jH6uzs1Be/+EW/m1cWQoZFevaQ//znP9fPf/7zfh+76qqrdPXVV/vRLITENddco2w2qxtvvFFHjx7V2WefrQ0bNnCVCU89/vjjymQy2rp16wlnsFx44YW67bbbfGqZvWKxmO644w79y7/8i/7+7/9ef/rTnzR79mz927/9W+DP/YkVCoWC340AAADhQ00GAAAwgpABAACMIGQAAAAjCBkAAMAIQgYAADCCkAEAAIwgZAAAACMIGQAAwAhCBgAAMIKQAQAAjCBkAAAAI/4fUmXhMgze4aoAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# just simulating and visualizing clustering of synthetic data using make_blobs() from scikit-learn\n",
    "# just to test how the clustering might look for our data\n",
    "\n",
    "# number of samples from our dataset \n",
    "n_samples = 200\n",
    "# how many k means clusters should the model make\n",
    "k_clusters = 4\n",
    "# Smaller cluster_std: Tighter, more compact clusters.\n",
    "# Larger cluster_std: More spread out, less compact clusters.\n",
    "# The choice of cluster_std depends on the characteristics you want in your synthetic data and the specific requirements of your clustering experiment.\n",
    "cluster_std  = 0.50\n",
    "\n",
    "# synthetic data made using the info passed into make_blobs() \n",
    "X, y_true = make_blobs(n_samples = n_samples, centers = k_clusters, cluster_std = cluster_std, random_state = 0)\n",
    "\n",
    "# X[:, 0] represents the first feature of the data points (x)\n",
    "# and X[:, 1] represents the second feature (y)\n",
    "#  The s=20 parameter specifies the size of the markers in the plot\n",
    "plt.scatter(X[:, 0], X[:, 1], s = 3);\n",
    "## plt.show()\n",
    "\n",
    "X.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'reset_index_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m# columns should be feautures (genes) and rows should be data points (gene expression per sample)\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39m# the model will sepearte the samples (rows) into two clusters based on their features (expression values per gene)\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m final_df \u001b[39m=\u001b[39m reset_index_df\u001b[39m.\u001b[39mT\n\u001b[1;32m      4\u001b[0m final_df\n",
      "\u001b[0;31mNameError\u001b[0m: name 'reset_index_df' is not defined"
     ]
    }
   ],
   "source": [
    "# columns should be feautures (genes) and rows should be data points (gene expression per sample)\n",
    "# the model will sepearte the samples (rows) into two clusters based on their features (expression values per gene)\n",
    "final_df = reset_index_df.T\n",
    "final_df\n",
    "\n",
    "## number of rows and columns in our dataframe\n",
    "#final_df.shape\n",
    "\n",
    "# .values gives an array of all the values for each feature (for GSM258551 [15.312190, 15.180497, 15.394483...], for GSM258552 [15.33512815, 15.18022902, 15.37584761...], for GSM258553 [15.14794428, 15.26208991, 15.25943145...], ... )\n",
    "# final_df.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'KMeans' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[2], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m# create a K-Means model with the specified number of clusters'\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39m# n_init is the number of times the K-Means algorithm will be run with different centroid seeds\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m kmeans \u001b[39m=\u001b[39m KMeans(n_clusters \u001b[39m=\u001b[39m \u001b[39m2\u001b[39m, n_init\u001b[39m=\u001b[39m\u001b[39m10\u001b[39m, random_state \u001b[39m=\u001b[39m \u001b[39m0\u001b[39m)\n\u001b[1;32m      5\u001b[0m \u001b[39m# fit the model to our data (array for the gene expression values per feature)\u001b[39;00m\n\u001b[1;32m      6\u001b[0m kmeans\u001b[39m.\u001b[39mfit(final_df)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'KMeans' is not defined"
     ]
    }
   ],
   "source": [
    "# create a K-Means model with the specified number of clusters'\n",
    "# n_init is the number of times the K-Means algorithm will be run with different centroid seeds\n",
    "kmeans = KMeans(n_clusters = 2, n_init=10, random_state = 0)\n",
    "\n",
    "# fit the model to our data (array for the gene expression values per feature)\n",
    "kmeans.fit(final_df)\n",
    "\n",
    "# the number of clusters (2) and the number of features (genes) in our data\n",
    "kmeans.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'final_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1], line 4\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m# create dataframe with 58 smaple lables as a column, cluster number and cancer type as another column \u001b[39;00m\n\u001b[1;32m      2\u001b[0m \n\u001b[1;32m      3\u001b[0m \u001b[39m# sample label\u001b[39;00m\n\u001b[0;32m----> 4\u001b[0m index_data \u001b[39m=\u001b[39m final_df\u001b[39m.\u001b[39mindex\n\u001b[1;32m      5\u001b[0m cluster_type_df \u001b[39m=\u001b[39m pd\u001b[39m.\u001b[39mDataFrame(index_data, columns\u001b[39m=\u001b[39m[\u001b[39m'\u001b[39m\u001b[39mSample Label\u001b[39m\u001b[39m'\u001b[39m])\n\u001b[1;32m      6\u001b[0m cluster_type_df \u001b[39m=\u001b[39m cluster_type_df\u001b[39m.\u001b[39mset_index(\u001b[39m'\u001b[39m\u001b[39mSample Label\u001b[39m\u001b[39m'\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'final_df' is not defined"
     ]
    }
   ],
   "source": [
    "# create dataframe with 58 smaple lables as a column, cluster number and cancer type as another column \n",
    "\n",
    "# sample label\n",
    "index_data = final_df.index\n",
    "cluster_type_df = pd.DataFrame(index_data, columns=['Sample Label'])\n",
    "cluster_type_df = cluster_type_df.set_index('Sample Label')\n",
    "\n",
    "# cluster number\n",
    "cluster_num = kmeans.labels_\n",
    "cluster_type_df['Cluster Numbers'] = cluster_num\n",
    "\n",
    "# ac or scc\n",
    "ac_scc_series = gse.phenotype_data[\"title\"].str.split('_').str[1]\n",
    "ac_scc_df = ac_scc_series.to_frame('Subtype')\n",
    "\n",
    "cluster_type_df = cluster_type_df.merge(cluster_type_df, left_index=True, right_index=True)\n",
    "cluster_type_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Q8. maybe see and plot which genes have highest expression in AC vs SC cells using plotting we did in session 12"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
