{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lung Cancer Subtyping\n",
    "a mathematical model that can discriminate between two lung cancer subtypes to build the model \n",
    "we use an unsupervised k-means clustering algorithm (Euclidean distance) of 58 NSCLC tumors using k=2\n",
    "\n",
    "data from: \n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE10245\n",
    "\n",
    "[GSE10245_family.soft.gz](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE10nnn/GSE10245/soft/)\n",
    "\n",
    "40 AC 18 SC\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "02-Nov-2023 20:13:15 INFO GEOparse - Parsing ./GSE10245_family.soft.gz: \n",
      "02-Nov-2023 20:13:15 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "02-Nov-2023 20:13:15 DEBUG GEOparse - SERIES: GSE10245\n",
      "02-Nov-2023 20:13:15 DEBUG GEOparse - PLATFORM: GPL570\n",
      "/Users/ymanasa/anaconda3/lib/python3.11/site-packages/GEOparse/GEOparse.py:401: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258551\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258552\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258553\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258554\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258555\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258556\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258557\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258558\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258559\n",
      "02-Nov-2023 20:13:17 DEBUG GEOparse - SAMPLE: GSM258560\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258561\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258562\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258563\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258564\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258565\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258566\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258567\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258568\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258569\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258570\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258571\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258572\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258573\n",
      "02-Nov-2023 20:13:18 DEBUG GEOparse - SAMPLE: GSM258574\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258575\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258576\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258577\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258578\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258579\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258580\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258581\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258582\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258583\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258584\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258585\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258586\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258587\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258588\n",
      "02-Nov-2023 20:13:19 DEBUG GEOparse - SAMPLE: GSM258589\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258590\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258591\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258592\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258593\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258594\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258595\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258596\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258597\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258598\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258599\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258600\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258601\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258602\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258603\n",
      "02-Nov-2023 20:13:20 DEBUG GEOparse - SAMPLE: GSM258604\n",
      "02-Nov-2023 20:13:21 DEBUG GEOparse - SAMPLE: GSM258605\n",
      "02-Nov-2023 20:13:21 DEBUG GEOparse - SAMPLE: GSM258606\n",
      "02-Nov-2023 20:13:21 DEBUG GEOparse - SAMPLE: GSM258607\n",
      "02-Nov-2023 20:13:21 DEBUG GEOparse - SAMPLE: GSM258608\n"
     ]
    }
   ],
   "source": [
    "# load the data using the GEOparse module\n",
    "# code adapted from: https://geoparse.readthedocs.io/en/latest/usage.html\n",
    "\n",
    "import GEOparse\n",
    "gse = GEOparse.get_GEO(filepath=\"./GSE10245_family.soft.gz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['__class__',\n",
       " '__delattr__',\n",
       " '__dict__',\n",
       " '__dir__',\n",
       " '__doc__',\n",
       " '__eq__',\n",
       " '__format__',\n",
       " '__ge__',\n",
       " '__getattribute__',\n",
       " '__getstate__',\n",
       " '__gt__',\n",
       " '__hash__',\n",
       " '__init__',\n",
       " '__init_subclass__',\n",
       " '__le__',\n",
       " '__lt__',\n",
       " '__metaclass__',\n",
       " '__module__',\n",
       " '__ne__',\n",
       " '__new__',\n",
       " '__reduce__',\n",
       " '__reduce_ex__',\n",
       " '__repr__',\n",
       " '__setattr__',\n",
       " '__sizeof__',\n",
       " '__str__',\n",
       " '__subclasshook__',\n",
       " '__weakref__',\n",
       " '_get_metadata_as_string',\n",
       " '_get_object_as_soft',\n",
       " '_phenotype_data',\n",
       " 'database',\n",
       " 'download_SRA',\n",
       " 'download_supplementary_files',\n",
       " 'geotype',\n",
       " 'get_accession',\n",
       " 'get_metadata_attribute',\n",
       " 'get_type',\n",
       " 'gpls',\n",
       " 'gsms',\n",
       " 'merge_and_average',\n",
       " 'metadata',\n",
       " 'name',\n",
       " 'phenotype_data',\n",
       " 'pivot_and_annotate',\n",
       " 'pivot_samples',\n",
       " 'relations',\n",
       " 'show_metadata',\n",
       " 'to_soft']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(gse)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_items([('GSM258551', <SAMPLE: GSM258551>), ('GSM258552', <SAMPLE: GSM258552>), ('GSM258553', <SAMPLE: GSM258553>), ('GSM258554', <SAMPLE: GSM258554>), ('GSM258555', <SAMPLE: GSM258555>), ('GSM258556', <SAMPLE: GSM258556>), ('GSM258557', <SAMPLE: GSM258557>), ('GSM258558', <SAMPLE: GSM258558>), ('GSM258559', <SAMPLE: GSM258559>), ('GSM258560', <SAMPLE: GSM258560>), ('GSM258561', <SAMPLE: GSM258561>), ('GSM258562', <SAMPLE: GSM258562>), ('GSM258563', <SAMPLE: GSM258563>), ('GSM258564', <SAMPLE: GSM258564>), ('GSM258565', <SAMPLE: GSM258565>), ('GSM258566', <SAMPLE: GSM258566>), ('GSM258567', <SAMPLE: GSM258567>), ('GSM258568', <SAMPLE: GSM258568>), ('GSM258569', <SAMPLE: GSM258569>), ('GSM258570', <SAMPLE: GSM258570>), ('GSM258571', <SAMPLE: GSM258571>), ('GSM258572', <SAMPLE: GSM258572>), ('GSM258573', <SAMPLE: GSM258573>), ('GSM258574', <SAMPLE: GSM258574>), ('GSM258575', <SAMPLE: GSM258575>), ('GSM258576', <SAMPLE: GSM258576>), ('GSM258577', <SAMPLE: GSM258577>), ('GSM258578', <SAMPLE: GSM258578>), ('GSM258579', <SAMPLE: GSM258579>), ('GSM258580', <SAMPLE: GSM258580>), ('GSM258581', <SAMPLE: GSM258581>), ('GSM258582', <SAMPLE: GSM258582>), ('GSM258583', <SAMPLE: GSM258583>), ('GSM258584', <SAMPLE: GSM258584>), ('GSM258585', <SAMPLE: GSM258585>), ('GSM258586', <SAMPLE: GSM258586>), ('GSM258587', <SAMPLE: GSM258587>), ('GSM258588', <SAMPLE: GSM258588>), ('GSM258589', <SAMPLE: GSM258589>), ('GSM258590', <SAMPLE: GSM258590>), ('GSM258591', <SAMPLE: GSM258591>), ('GSM258592', <SAMPLE: GSM258592>), ('GSM258593', <SAMPLE: GSM258593>), ('GSM258594', <SAMPLE: GSM258594>), ('GSM258595', <SAMPLE: GSM258595>), ('GSM258596', <SAMPLE: GSM258596>), ('GSM258597', <SAMPLE: GSM258597>), ('GSM258598', <SAMPLE: GSM258598>), ('GSM258599', <SAMPLE: GSM258599>), ('GSM258600', <SAMPLE: GSM258600>), ('GSM258601', <SAMPLE: GSM258601>), ('GSM258602', <SAMPLE: GSM258602>), ('GSM258603', <SAMPLE: GSM258603>), ('GSM258604', <SAMPLE: GSM258604>), ('GSM258605', <SAMPLE: GSM258605>), ('GSM258606', <SAMPLE: GSM258606>), ('GSM258607', <SAMPLE: GSM258607>), ('GSM258608', <SAMPLE: GSM258608>)])\n",
      "dict_items([('title', ['NSCLC_AC_10']), ('geo_accession', ['GSM258551']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 10.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258551/suppl/GSM258551.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_16']), ('geo_accession', ['GSM258552']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 16.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258552/suppl/GSM258552.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_22']), ('geo_accession', ['GSM258553']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 22.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258553/suppl/GSM258553.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_24']), ('geo_accession', ['GSM258554']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 24.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258554/suppl/GSM258554.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_25']), ('geo_accession', ['GSM258555']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 25.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258555/suppl/GSM258555.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_30']), ('geo_accession', ['GSM258556']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 30.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258556/suppl/GSM258556.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_37']), ('geo_accession', ['GSM258557']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 37.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258557/suppl/GSM258557.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_40']), ('geo_accession', ['GSM258558']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 40.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258558/suppl/GSM258558.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_49']), ('geo_accession', ['GSM258559']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 49.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258559/suppl/GSM258559.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_51']), ('geo_accession', ['GSM258560']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 51.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258560/suppl/GSM258560.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_53']), ('geo_accession', ['GSM258561']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 53.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258561/suppl/GSM258561.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_55']), ('geo_accession', ['GSM258562']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 55.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258562/suppl/GSM258562.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_65']), ('geo_accession', ['GSM258563']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 65.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258563/suppl/GSM258563.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_70']), ('geo_accession', ['GSM258564']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 70.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258564/suppl/GSM258564.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_75']), ('geo_accession', ['GSM258565']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 75.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258565/suppl/GSM258565.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_77']), ('geo_accession', ['GSM258566']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 77.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258566/suppl/GSM258566.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_84']), ('geo_accession', ['GSM258567']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 84.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258567/suppl/GSM258567.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_88']), ('geo_accession', ['GSM258568']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 88.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258568/suppl/GSM258568.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_89']), ('geo_accession', ['GSM258569']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 89.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258569/suppl/GSM258569.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_95']), ('geo_accession', ['GSM258570']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 95.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258570/suppl/GSM258570.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_97']), ('geo_accession', ['GSM258571']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 97.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258571/suppl/GSM258571.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_113']), ('geo_accession', ['GSM258572']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 113.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258572/suppl/GSM258572.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_136']), ('geo_accession', ['GSM258573']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 136.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258573/suppl/GSM258573.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_137']), ('geo_accession', ['GSM258574']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 137.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258574/suppl/GSM258574.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_147']), ('geo_accession', ['GSM258575']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 147.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258575/suppl/GSM258575.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_148']), ('geo_accession', ['GSM258576']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 148.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258576/suppl/GSM258576.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_161']), ('geo_accession', ['GSM258577']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 161.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258577/suppl/GSM258577.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_165']), ('geo_accession', ['GSM258578']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 165.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258578/suppl/GSM258578.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_169']), ('geo_accession', ['GSM258579']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 169.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258579/suppl/GSM258579.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_188']), ('geo_accession', ['GSM258580']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 188.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258580/suppl/GSM258580.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_202']), ('geo_accession', ['GSM258581']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 202.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258581/suppl/GSM258581.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_210']), ('geo_accession', ['GSM258582']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 210.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258582/suppl/GSM258582.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_215']), ('geo_accession', ['GSM258583']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 215.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258583/suppl/GSM258583.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_233']), ('geo_accession', ['GSM258584']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 233.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258584/suppl/GSM258584.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_248']), ('geo_accession', ['GSM258585']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 248.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258585/suppl/GSM258585.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_271']), ('geo_accession', ['GSM258586']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 271.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258586/suppl/GSM258586.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_273']), ('geo_accession', ['GSM258587']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 273.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258587/suppl/GSM258587.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_276']), ('geo_accession', ['GSM258588']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 276.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258588/suppl/GSM258588.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_282']), ('geo_accession', ['GSM258589']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 282.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258589/suppl/GSM258589.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_301']), ('geo_accession', ['GSM258590']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 301.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258590/suppl/GSM258590.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_331']), ('geo_accession', ['GSM258591']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 331.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258591/suppl/GSM258591.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_334']), ('geo_accession', ['GSM258592']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 334.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258592/suppl/GSM258592.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_342']), ('geo_accession', ['GSM258593']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 342.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258593/suppl/GSM258593.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_347']), ('geo_accession', ['GSM258594']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 347.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258594/suppl/GSM258594.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_348']), ('geo_accession', ['GSM258595']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 348.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258595/suppl/GSM258595.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_364']), ('geo_accession', ['GSM258596']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 364.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258596/suppl/GSM258596.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_368']), ('geo_accession', ['GSM258597']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 368.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258597/suppl/GSM258597.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_378']), ('geo_accession', ['GSM258598']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 378.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258598/suppl/GSM258598.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_390']), ('geo_accession', ['GSM258599']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 390.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258599/suppl/GSM258599.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_397']), ('geo_accession', ['GSM258600']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 397.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258600/suppl/GSM258600.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_400']), ('geo_accession', ['GSM258601']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 400.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258601/suppl/GSM258601.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_402']), ('geo_accession', ['GSM258602']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 402.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258602/suppl/GSM258602.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_SCC_412']), ('geo_accession', ['GSM258603']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: squamous cell carcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 412.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258603/suppl/GSM258603.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_426']), ('geo_accession', ['GSM258604']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 426.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258604/suppl/GSM258604.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_434']), ('geo_accession', ['GSM258605']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 434.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258605/suppl/GSM258605.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_445']), ('geo_accession', ['GSM258606']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 445.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258606/suppl/GSM258606.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_489']), ('geo_accession', ['GSM258607']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 489.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258607/suppl/GSM258607.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n",
      "dict_items([('title', ['NSCLC_AC_492']), ('geo_accession', ['GSM258608']), ('status', ['Public on Oct 01 2009']), ('submission_date', ['Jan 23 2008']), ('last_update_date', ['Aug 28 2018']), ('type', ['RNA']), ('channel_count', ['1']), ('source_name_ch1', ['human non-small cell lung cancer tumor tissue']), ('organism_ch1', ['Homo sapiens']), ('taxid_ch1', ['9606']), ('characteristics_ch1', ['disease state: adenocarcinoma']), ('treatment_protocol_ch1', ['tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.']), ('molecule_ch1', ['total RNA']), ('extract_protocol_ch1', ['total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.']), ('label_ch1', ['biotin']), ('label_protocol_ch1', ['Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).']), ('hyb_protocol', ['Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.']), ('scan_protocol', ['Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.']), ('description', ['Gene expression data from human non-small cell lung cancer tumor tissue 492.']), ('data_processing', ['Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].']), ('platform_id', ['GPL570']), ('contact_name', ['Ruprecht,,Kuner']), ('contact_laboratory', ['Unit Cancer Genome Research']), ('contact_department', ['Molecular Genetics']), ('contact_institute', ['German Cancer Research Center and National Center of Tumor Diseases']), ('contact_address', ['Im Neuenheimer Feld 460']), ('contact_city', ['Heidelberg']), ('contact_zip/postal_code', ['69120']), ('contact_country', ['Germany']), ('supplementary_file', ['ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258608/suppl/GSM258608.CEL.gz']), ('relation', ['Reanalyzed by: GSE60486', 'Reanalyzed by: GSE119087']), ('series_id', ['GSE10245']), ('data_row_count', ['54675'])])\n"
     ]
    }
   ],
   "source": [
    "# gse.gsms is a dict \n",
    "print(gse.gsms.items())\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "# gsm is a GEOparse `object <class 'GEOparse.GEOTypes.GSM'>`\n",
    "# gsm has metadata attribute \n",
    "    print(gsm.metadata.items())\n",
    "    \n",
    "# print( gse.gsms.items())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>name</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>GSM258558</th>\n",
       "      <th>GSM258559</th>\n",
       "      <th>GSM258560</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID_REF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1007_s_at</th>\n",
       "      <td>9.129905</td>\n",
       "      <td>9.843349</td>\n",
       "      <td>9.730661</td>\n",
       "      <td>9.032165</td>\n",
       "      <td>10.281793</td>\n",
       "      <td>9.154551</td>\n",
       "      <td>8.063788</td>\n",
       "      <td>8.881805</td>\n",
       "      <td>9.039005</td>\n",
       "      <td>10.643545</td>\n",
       "      <td>...</td>\n",
       "      <td>9.628212</td>\n",
       "      <td>10.130757</td>\n",
       "      <td>9.276419</td>\n",
       "      <td>9.296957</td>\n",
       "      <td>10.346701</td>\n",
       "      <td>9.709068</td>\n",
       "      <td>7.672985</td>\n",
       "      <td>9.199080</td>\n",
       "      <td>7.635765</td>\n",
       "      <td>8.760082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1053_at</th>\n",
       "      <td>8.034022</td>\n",
       "      <td>7.973332</td>\n",
       "      <td>8.834045</td>\n",
       "      <td>7.723965</td>\n",
       "      <td>9.040800</td>\n",
       "      <td>8.792375</td>\n",
       "      <td>8.093354</td>\n",
       "      <td>7.575620</td>\n",
       "      <td>8.438594</td>\n",
       "      <td>9.142260</td>\n",
       "      <td>...</td>\n",
       "      <td>9.215567</td>\n",
       "      <td>8.273698</td>\n",
       "      <td>8.790710</td>\n",
       "      <td>8.861625</td>\n",
       "      <td>8.512193</td>\n",
       "      <td>7.908581</td>\n",
       "      <td>9.299454</td>\n",
       "      <td>8.220959</td>\n",
       "      <td>7.432135</td>\n",
       "      <td>7.596431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117_at</th>\n",
       "      <td>3.564520</td>\n",
       "      <td>4.994852</td>\n",
       "      <td>5.066018</td>\n",
       "      <td>4.958580</td>\n",
       "      <td>4.951835</td>\n",
       "      <td>6.428274</td>\n",
       "      <td>7.364676</td>\n",
       "      <td>4.659326</td>\n",
       "      <td>6.643723</td>\n",
       "      <td>5.368356</td>\n",
       "      <td>...</td>\n",
       "      <td>5.183919</td>\n",
       "      <td>4.801579</td>\n",
       "      <td>6.161777</td>\n",
       "      <td>5.443126</td>\n",
       "      <td>4.609810</td>\n",
       "      <td>4.339004</td>\n",
       "      <td>6.384404</td>\n",
       "      <td>6.006534</td>\n",
       "      <td>5.484470</td>\n",
       "      <td>5.372583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121_at</th>\n",
       "      <td>4.746490</td>\n",
       "      <td>5.197306</td>\n",
       "      <td>5.234618</td>\n",
       "      <td>6.078180</td>\n",
       "      <td>5.205632</td>\n",
       "      <td>5.009848</td>\n",
       "      <td>5.533000</td>\n",
       "      <td>5.362637</td>\n",
       "      <td>5.210988</td>\n",
       "      <td>5.195820</td>\n",
       "      <td>...</td>\n",
       "      <td>5.202508</td>\n",
       "      <td>4.807654</td>\n",
       "      <td>5.326347</td>\n",
       "      <td>5.205004</td>\n",
       "      <td>5.201123</td>\n",
       "      <td>6.028717</td>\n",
       "      <td>5.220915</td>\n",
       "      <td>5.186452</td>\n",
       "      <td>5.220778</td>\n",
       "      <td>5.097535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1255_g_at</th>\n",
       "      <td>2.320698</td>\n",
       "      <td>2.248520</td>\n",
       "      <td>2.259504</td>\n",
       "      <td>2.262787</td>\n",
       "      <td>2.207531</td>\n",
       "      <td>2.322618</td>\n",
       "      <td>2.656929</td>\n",
       "      <td>2.255955</td>\n",
       "      <td>2.266399</td>\n",
       "      <td>2.246563</td>\n",
       "      <td>...</td>\n",
       "      <td>2.248392</td>\n",
       "      <td>2.232021</td>\n",
       "      <td>2.246738</td>\n",
       "      <td>2.245739</td>\n",
       "      <td>2.265454</td>\n",
       "      <td>2.301397</td>\n",
       "      <td>2.322604</td>\n",
       "      <td>2.236494</td>\n",
       "      <td>2.243698</td>\n",
       "      <td>2.420731</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "name       GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  GSM258556  \\\n",
       "ID_REF                                                                        \n",
       "1007_s_at   9.129905   9.843349   9.730661   9.032165  10.281793   9.154551   \n",
       "1053_at     8.034022   7.973332   8.834045   7.723965   9.040800   8.792375   \n",
       "117_at      3.564520   4.994852   5.066018   4.958580   4.951835   6.428274   \n",
       "121_at      4.746490   5.197306   5.234618   6.078180   5.205632   5.009848   \n",
       "1255_g_at   2.320698   2.248520   2.259504   2.262787   2.207531   2.322618   \n",
       "\n",
       "name       GSM258557  GSM258558  GSM258559  GSM258560  ...  GSM258599  \\\n",
       "ID_REF                                                 ...              \n",
       "1007_s_at   8.063788   8.881805   9.039005  10.643545  ...   9.628212   \n",
       "1053_at     8.093354   7.575620   8.438594   9.142260  ...   9.215567   \n",
       "117_at      7.364676   4.659326   6.643723   5.368356  ...   5.183919   \n",
       "121_at      5.533000   5.362637   5.210988   5.195820  ...   5.202508   \n",
       "1255_g_at   2.656929   2.255955   2.266399   2.246563  ...   2.248392   \n",
       "\n",
       "name       GSM258600  GSM258601  GSM258602  GSM258603  GSM258604  GSM258605  \\\n",
       "ID_REF                                                                        \n",
       "1007_s_at  10.130757   9.276419   9.296957  10.346701   9.709068   7.672985   \n",
       "1053_at     8.273698   8.790710   8.861625   8.512193   7.908581   9.299454   \n",
       "117_at      4.801579   6.161777   5.443126   4.609810   4.339004   6.384404   \n",
       "121_at      4.807654   5.326347   5.205004   5.201123   6.028717   5.220915   \n",
       "1255_g_at   2.232021   2.246738   2.245739   2.265454   2.301397   2.322604   \n",
       "\n",
       "name       GSM258606  GSM258607  GSM258608  \n",
       "ID_REF                                      \n",
       "1007_s_at   9.199080   7.635765   8.760082  \n",
       "1053_at     8.220959   7.432135   7.596431  \n",
       "117_at      6.006534   5.484470   5.372583  \n",
       "121_at      5.186452   5.220778   5.097535  \n",
       "1255_g_at   2.236494   2.243698   2.420731  \n",
       "\n",
       "[5 rows x 58 columns]"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# getting expression data from gse object as a dataframe \n",
    "expr = pd.DataFrame(gse.pivot_samples('VALUE'))\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gpls attribute of gse \n",
    "print(gse.gpls.items())\n",
    "for gpl_name, gpl in gse.gpls.items():\n",
    "    print(gpl.table.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gpl.table.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54670</th>\n",
       "      <td>AFFX-ThrX-5_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54671</th>\n",
       "      <td>AFFX-ThrX-M_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54672</th>\n",
       "      <td>AFFX-TrpnX-3_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54673</th>\n",
       "      <td>AFFX-TrpnX-5_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54674</th>\n",
       "      <td>AFFX-TrpnX-M_at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54675 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    ID       Gene Symbol     ENTREZ_GENE_ID\n",
       "0            1007_s_at  DDR1 /// MIR4640  780 /// 100616237\n",
       "1              1053_at              RFC2               5982\n",
       "2               117_at             HSPA6               3310\n",
       "3               121_at              PAX8               7849\n",
       "4            1255_g_at            GUCA1A               2978\n",
       "...                ...               ...                ...\n",
       "54670   AFFX-ThrX-5_at               NaN                NaN\n",
       "54671   AFFX-ThrX-M_at               NaN                NaN\n",
       "54672  AFFX-TrpnX-3_at               NaN                NaN\n",
       "54673  AFFX-TrpnX-5_at               NaN                NaN\n",
       "54674  AFFX-TrpnX-M_at               NaN                NaN\n",
       "\n",
       "[54675 rows x 3 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# assinging ID, Gene Symbol, ENTREZ_GENE_ID columns from gpl.table as a pandas dataframe to metadata\n",
    "metadata = pd.DataFrame(gpl.table, columns=['ID', 'Gene Symbol', 'ENTREZ_GENE_ID'])\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1007_s_at</td>\n",
       "      <td>DDR1 /// MIR4640</td>\n",
       "      <td>780 /// 100616237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1053_at</td>\n",
       "      <td>RFC2</td>\n",
       "      <td>5982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>117_at</td>\n",
       "      <td>HSPA6</td>\n",
       "      <td>3310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>121_at</td>\n",
       "      <td>PAX8</td>\n",
       "      <td>7849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1255_g_at</td>\n",
       "      <td>GUCA1A</td>\n",
       "      <td>2978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54630</th>\n",
       "      <td>AFFX-HUMGAPDH/M33197_M_at</td>\n",
       "      <td>GAPDH</td>\n",
       "      <td>2597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54631</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_3_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54632</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_5_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54633</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MA_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54634</th>\n",
       "      <td>AFFX-HUMISGF3A/M97935_MB_at</td>\n",
       "      <td>STAT1</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44134 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                ID       Gene Symbol     ENTREZ_GENE_ID\n",
       "0                        1007_s_at  DDR1 /// MIR4640  780 /// 100616237\n",
       "1                          1053_at              RFC2               5982\n",
       "2                           117_at             HSPA6               3310\n",
       "3                           121_at              PAX8               7849\n",
       "4                        1255_g_at            GUCA1A               2978\n",
       "...                            ...               ...                ...\n",
       "54630    AFFX-HUMGAPDH/M33197_M_at             GAPDH               2597\n",
       "54631   AFFX-HUMISGF3A/M97935_3_at             STAT1               6772\n",
       "54632   AFFX-HUMISGF3A/M97935_5_at             STAT1               6772\n",
       "54633  AFFX-HUMISGF3A/M97935_MA_at             STAT1               6772\n",
       "54634  AFFX-HUMISGF3A/M97935_MB_at             STAT1               6772\n",
       "\n",
       "[44134 rows x 3 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# cleanup the data by dropping NaN\n",
    "metadata.dropna(inplace=True)\n",
    "metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gf/79pl6dp1569b76cgzy_9yv7c0000gs/T/ipykernel_43792/3238646634.py:5: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.\n",
      "  merged_meta_expr[\"mean\"] = merged_meta_expr.mean(axis=\"columns\")\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>ENTREZ_GENE_ID</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "      <th>mean</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>1553538_s_at</td>\n",
       "      <td>COX1</td>\n",
       "      <td>4512</td>\n",
       "      <td>15.312190</td>\n",
       "      <td>15.335128</td>\n",
       "      <td>15.147944</td>\n",
       "      <td>15.235262</td>\n",
       "      <td>15.402517</td>\n",
       "      <td>15.235435</td>\n",
       "      <td>15.266640</td>\n",
       "      <td>...</td>\n",
       "      <td>15.495646</td>\n",
       "      <td>15.392720</td>\n",
       "      <td>15.480795</td>\n",
       "      <td>15.372107</td>\n",
       "      <td>15.476776</td>\n",
       "      <td>15.125095</td>\n",
       "      <td>14.820450</td>\n",
       "      <td>15.409687</td>\n",
       "      <td>15.145438</td>\n",
       "      <td>15.380371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19877</th>\n",
       "      <td>213614_x_at</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1915</td>\n",
       "      <td>15.180497</td>\n",
       "      <td>15.180229</td>\n",
       "      <td>15.262090</td>\n",
       "      <td>15.246653</td>\n",
       "      <td>15.272874</td>\n",
       "      <td>15.181830</td>\n",
       "      <td>15.162696</td>\n",
       "      <td>...</td>\n",
       "      <td>15.184177</td>\n",
       "      <td>15.130715</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>15.174592</td>\n",
       "      <td>15.192399</td>\n",
       "      <td>15.167122</td>\n",
       "      <td>15.298337</td>\n",
       "      <td>15.110260</td>\n",
       "      <td>15.088462</td>\n",
       "      <td>15.199665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19749</th>\n",
       "      <td>213477_x_at</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1915</td>\n",
       "      <td>14.982105</td>\n",
       "      <td>14.974366</td>\n",
       "      <td>15.006630</td>\n",
       "      <td>15.037649</td>\n",
       "      <td>15.094668</td>\n",
       "      <td>14.986383</td>\n",
       "      <td>14.972177</td>\n",
       "      <td>...</td>\n",
       "      <td>15.024057</td>\n",
       "      <td>14.913341</td>\n",
       "      <td>15.057497</td>\n",
       "      <td>15.007305</td>\n",
       "      <td>15.019143</td>\n",
       "      <td>14.995539</td>\n",
       "      <td>14.990898</td>\n",
       "      <td>15.032457</td>\n",
       "      <td>14.929415</td>\n",
       "      <td>15.014144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26931</th>\n",
       "      <td>221671_x_at</td>\n",
       "      <td>IGK /// IGKC</td>\n",
       "      <td>3514 /// 50802</td>\n",
       "      <td>15.394483</td>\n",
       "      <td>15.375848</td>\n",
       "      <td>15.259431</td>\n",
       "      <td>15.250159</td>\n",
       "      <td>11.996869</td>\n",
       "      <td>15.053283</td>\n",
       "      <td>15.450697</td>\n",
       "      <td>...</td>\n",
       "      <td>15.370751</td>\n",
       "      <td>14.985706</td>\n",
       "      <td>14.771899</td>\n",
       "      <td>15.048471</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.678163</td>\n",
       "      <td>15.120340</td>\n",
       "      <td>15.349818</td>\n",
       "      <td>15.372265</td>\n",
       "      <td>15.004009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22263</th>\n",
       "      <td>216438_s_at</td>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>7114</td>\n",
       "      <td>15.164475</td>\n",
       "      <td>15.337793</td>\n",
       "      <td>14.886320</td>\n",
       "      <td>14.723359</td>\n",
       "      <td>14.734361</td>\n",
       "      <td>14.755676</td>\n",
       "      <td>15.090716</td>\n",
       "      <td>...</td>\n",
       "      <td>15.184035</td>\n",
       "      <td>14.716383</td>\n",
       "      <td>14.885592</td>\n",
       "      <td>14.950120</td>\n",
       "      <td>14.881242</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.600380</td>\n",
       "      <td>15.158579</td>\n",
       "      <td>15.059676</td>\n",
       "      <td>14.990839</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 62 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 ID   Gene Symbol  ENTREZ_GENE_ID  GSM258551  GSM258552  \\\n",
       "914    1553538_s_at          COX1            4512  15.312190  15.335128   \n",
       "19877   213614_x_at        EEF1A1            1915  15.180497  15.180229   \n",
       "19749   213477_x_at        EEF1A1            1915  14.982105  14.974366   \n",
       "26931   221671_x_at  IGK /// IGKC  3514 /// 50802  15.394483  15.375848   \n",
       "22263   216438_s_at        TMSB4X            7114  15.164475  15.337793   \n",
       "\n",
       "       GSM258553  GSM258554  GSM258555  GSM258556  GSM258557  ...  GSM258600  \\\n",
       "914    15.147944  15.235262  15.402517  15.235435  15.266640  ...  15.495646   \n",
       "19877  15.262090  15.246653  15.272874  15.181830  15.162696  ...  15.184177   \n",
       "19749  15.006630  15.037649  15.094668  14.986383  14.972177  ...  15.024057   \n",
       "26931  15.259431  15.250159  11.996869  15.053283  15.450697  ...  15.370751   \n",
       "22263  14.886320  14.723359  14.734361  14.755676  15.090716  ...  15.184035   \n",
       "\n",
       "       GSM258601  GSM258602  GSM258603  GSM258604  GSM258605  GSM258606  \\\n",
       "914    15.392720  15.480795  15.372107  15.476776  15.125095  14.820450   \n",
       "19877  15.130715  15.191393  15.174592  15.192399  15.167122  15.298337   \n",
       "19749  14.913341  15.057497  15.007305  15.019143  14.995539  14.990898   \n",
       "26931  14.985706  14.771899  15.048471  15.236298  14.678163  15.120340   \n",
       "22263  14.716383  14.885592  14.950120  14.881242  15.236298  14.600380   \n",
       "\n",
       "       GSM258607  GSM258608       mean  \n",
       "914    15.409687  15.145438  15.380371  \n",
       "19877  15.110260  15.088462  15.199665  \n",
       "19749  15.032457  14.929415  15.014144  \n",
       "26931  15.349818  15.372265  15.004009  \n",
       "22263  15.158579  15.059676  14.990839  \n",
       "\n",
       "[5 rows x 62 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge the gsp and gpl data to remove any duplicates \n",
    "# repeating Gene Symbol values can have different IDs: for ex, STAT1 6772 repeats but the ID stays different\n",
    "# \n",
    "merged_meta_expr = pd.merge(metadata,expr, left_on= \"ID\", right_on=\"ID_REF\")\n",
    "\n",
    "# arrange mean from largest to smallest so that when we drop duplicates, the first occurance is kepy \n",
    "# the first occurance (with highest mean) will be the best one to keep\n",
    "merged_meta_expr[\"mean\"] = merged_meta_expr.mean(axis=\"columns\")\n",
    "sorted_merged = merged_meta_expr.sort_values(by=[\"mean\"],ascending=False)\n",
    "sorted_merged.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/gf/79pl6dp1569b76cgzy_9yv7c0000gs/T/ipykernel_43792/3074067731.py:5: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.\n",
      "  no_mean_id = no_dupes.drop([\"mean\", \"ID\", \"ENTREZ_GENE_ID\"] ,1)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>GSM258558</th>\n",
       "      <th>GSM258559</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>COX1</td>\n",
       "      <td>15.312190</td>\n",
       "      <td>15.335128</td>\n",
       "      <td>15.147944</td>\n",
       "      <td>15.235262</td>\n",
       "      <td>15.402517</td>\n",
       "      <td>15.235435</td>\n",
       "      <td>15.266640</td>\n",
       "      <td>15.236145</td>\n",
       "      <td>15.530658</td>\n",
       "      <td>...</td>\n",
       "      <td>15.354914</td>\n",
       "      <td>15.495646</td>\n",
       "      <td>15.392720</td>\n",
       "      <td>15.480795</td>\n",
       "      <td>15.372107</td>\n",
       "      <td>15.476776</td>\n",
       "      <td>15.125095</td>\n",
       "      <td>14.820450</td>\n",
       "      <td>15.409687</td>\n",
       "      <td>15.145438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19877</th>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>15.180497</td>\n",
       "      <td>15.180229</td>\n",
       "      <td>15.262090</td>\n",
       "      <td>15.246653</td>\n",
       "      <td>15.272874</td>\n",
       "      <td>15.181830</td>\n",
       "      <td>15.162696</td>\n",
       "      <td>15.126815</td>\n",
       "      <td>15.315813</td>\n",
       "      <td>...</td>\n",
       "      <td>15.165107</td>\n",
       "      <td>15.184177</td>\n",
       "      <td>15.130715</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>15.174592</td>\n",
       "      <td>15.192399</td>\n",
       "      <td>15.167122</td>\n",
       "      <td>15.298337</td>\n",
       "      <td>15.110260</td>\n",
       "      <td>15.088462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26931</th>\n",
       "      <td>IGK /// IGKC</td>\n",
       "      <td>15.394483</td>\n",
       "      <td>15.375848</td>\n",
       "      <td>15.259431</td>\n",
       "      <td>15.250159</td>\n",
       "      <td>11.996869</td>\n",
       "      <td>15.053283</td>\n",
       "      <td>15.450697</td>\n",
       "      <td>15.391052</td>\n",
       "      <td>14.658909</td>\n",
       "      <td>...</td>\n",
       "      <td>15.177085</td>\n",
       "      <td>15.370751</td>\n",
       "      <td>14.985706</td>\n",
       "      <td>14.771899</td>\n",
       "      <td>15.048471</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.678163</td>\n",
       "      <td>15.120340</td>\n",
       "      <td>15.349818</td>\n",
       "      <td>15.372265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22263</th>\n",
       "      <td>TMSB4X</td>\n",
       "      <td>15.164475</td>\n",
       "      <td>15.337793</td>\n",
       "      <td>14.886320</td>\n",
       "      <td>14.723359</td>\n",
       "      <td>14.734361</td>\n",
       "      <td>14.755676</td>\n",
       "      <td>15.090716</td>\n",
       "      <td>14.747466</td>\n",
       "      <td>15.357297</td>\n",
       "      <td>...</td>\n",
       "      <td>14.861333</td>\n",
       "      <td>15.184035</td>\n",
       "      <td>14.716383</td>\n",
       "      <td>14.885592</td>\n",
       "      <td>14.950120</td>\n",
       "      <td>14.881242</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.600380</td>\n",
       "      <td>15.158579</td>\n",
       "      <td>15.059676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>ATP6</td>\n",
       "      <td>14.765942</td>\n",
       "      <td>14.939455</td>\n",
       "      <td>14.530669</td>\n",
       "      <td>14.956952</td>\n",
       "      <td>14.960515</td>\n",
       "      <td>14.767122</td>\n",
       "      <td>14.648069</td>\n",
       "      <td>14.847039</td>\n",
       "      <td>14.897657</td>\n",
       "      <td>...</td>\n",
       "      <td>14.963645</td>\n",
       "      <td>15.086318</td>\n",
       "      <td>14.994576</td>\n",
       "      <td>15.079311</td>\n",
       "      <td>14.952733</td>\n",
       "      <td>15.047349</td>\n",
       "      <td>14.124653</td>\n",
       "      <td>14.317633</td>\n",
       "      <td>14.954273</td>\n",
       "      <td>14.737609</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 59 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Symbol  GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  \\\n",
       "914            COX1  15.312190  15.335128  15.147944  15.235262  15.402517   \n",
       "19877        EEF1A1  15.180497  15.180229  15.262090  15.246653  15.272874   \n",
       "26931  IGK /// IGKC  15.394483  15.375848  15.259431  15.250159  11.996869   \n",
       "22263        TMSB4X  15.164475  15.337793  14.886320  14.723359  14.734361   \n",
       "936            ATP6  14.765942  14.939455  14.530669  14.956952  14.960515   \n",
       "\n",
       "       GSM258556  GSM258557  GSM258558  GSM258559  ...  GSM258599  GSM258600  \\\n",
       "914    15.235435  15.266640  15.236145  15.530658  ...  15.354914  15.495646   \n",
       "19877  15.181830  15.162696  15.126815  15.315813  ...  15.165107  15.184177   \n",
       "26931  15.053283  15.450697  15.391052  14.658909  ...  15.177085  15.370751   \n",
       "22263  14.755676  15.090716  14.747466  15.357297  ...  14.861333  15.184035   \n",
       "936    14.767122  14.648069  14.847039  14.897657  ...  14.963645  15.086318   \n",
       "\n",
       "       GSM258601  GSM258602  GSM258603  GSM258604  GSM258605  GSM258606  \\\n",
       "914    15.392720  15.480795  15.372107  15.476776  15.125095  14.820450   \n",
       "19877  15.130715  15.191393  15.174592  15.192399  15.167122  15.298337   \n",
       "26931  14.985706  14.771899  15.048471  15.236298  14.678163  15.120340   \n",
       "22263  14.716383  14.885592  14.950120  14.881242  15.236298  14.600380   \n",
       "936    14.994576  15.079311  14.952733  15.047349  14.124653  14.317633   \n",
       "\n",
       "       GSM258607  GSM258608  \n",
       "914    15.409687  15.145438  \n",
       "19877  15.110260  15.088462  \n",
       "26931  15.349818  15.372265  \n",
       "22263  15.158579  15.059676  \n",
       "936    14.954273  14.737609  \n",
       "\n",
       "[5 rows x 59 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dropping the duplicates-- keeping highest mean expression values \n",
    "no_dupes = sorted_merged.drop_duplicates(\"Gene Symbol\")\n",
    "\n",
    "# dropping other unecessary columns \n",
    "no_mean_id = no_dupes.drop([\"mean\", \"ID\", \"ENTREZ_GENE_ID\"] ,1)\n",
    "\n",
    "no_mean_id.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM258551</th>\n",
       "      <th>GSM258552</th>\n",
       "      <th>GSM258553</th>\n",
       "      <th>GSM258554</th>\n",
       "      <th>GSM258555</th>\n",
       "      <th>GSM258556</th>\n",
       "      <th>GSM258557</th>\n",
       "      <th>GSM258558</th>\n",
       "      <th>GSM258559</th>\n",
       "      <th>GSM258560</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM258599</th>\n",
       "      <th>GSM258600</th>\n",
       "      <th>GSM258601</th>\n",
       "      <th>GSM258602</th>\n",
       "      <th>GSM258603</th>\n",
       "      <th>GSM258604</th>\n",
       "      <th>GSM258605</th>\n",
       "      <th>GSM258606</th>\n",
       "      <th>GSM258607</th>\n",
       "      <th>GSM258608</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>COX1</th>\n",
       "      <td>15.312190</td>\n",
       "      <td>15.335128</td>\n",
       "      <td>15.147944</td>\n",
       "      <td>15.235262</td>\n",
       "      <td>15.402517</td>\n",
       "      <td>15.235435</td>\n",
       "      <td>15.266640</td>\n",
       "      <td>15.236145</td>\n",
       "      <td>15.530658</td>\n",
       "      <td>15.323297</td>\n",
       "      <td>...</td>\n",
       "      <td>15.354914</td>\n",
       "      <td>15.495646</td>\n",
       "      <td>15.392720</td>\n",
       "      <td>15.480795</td>\n",
       "      <td>15.372107</td>\n",
       "      <td>15.476776</td>\n",
       "      <td>15.125095</td>\n",
       "      <td>14.820450</td>\n",
       "      <td>15.409687</td>\n",
       "      <td>15.145438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EEF1A1</th>\n",
       "      <td>15.180497</td>\n",
       "      <td>15.180229</td>\n",
       "      <td>15.262090</td>\n",
       "      <td>15.246653</td>\n",
       "      <td>15.272874</td>\n",
       "      <td>15.181830</td>\n",
       "      <td>15.162696</td>\n",
       "      <td>15.126815</td>\n",
       "      <td>15.315813</td>\n",
       "      <td>15.406522</td>\n",
       "      <td>...</td>\n",
       "      <td>15.165107</td>\n",
       "      <td>15.184177</td>\n",
       "      <td>15.130715</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>15.174592</td>\n",
       "      <td>15.192399</td>\n",
       "      <td>15.167122</td>\n",
       "      <td>15.298337</td>\n",
       "      <td>15.110260</td>\n",
       "      <td>15.088462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGK /// IGKC</th>\n",
       "      <td>15.394483</td>\n",
       "      <td>15.375848</td>\n",
       "      <td>15.259431</td>\n",
       "      <td>15.250159</td>\n",
       "      <td>11.996869</td>\n",
       "      <td>15.053283</td>\n",
       "      <td>15.450697</td>\n",
       "      <td>15.391052</td>\n",
       "      <td>14.658909</td>\n",
       "      <td>14.706833</td>\n",
       "      <td>...</td>\n",
       "      <td>15.177085</td>\n",
       "      <td>15.370751</td>\n",
       "      <td>14.985706</td>\n",
       "      <td>14.771899</td>\n",
       "      <td>15.048471</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.678163</td>\n",
       "      <td>15.120340</td>\n",
       "      <td>15.349818</td>\n",
       "      <td>15.372265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMSB4X</th>\n",
       "      <td>15.164475</td>\n",
       "      <td>15.337793</td>\n",
       "      <td>14.886320</td>\n",
       "      <td>14.723359</td>\n",
       "      <td>14.734361</td>\n",
       "      <td>14.755676</td>\n",
       "      <td>15.090716</td>\n",
       "      <td>14.747466</td>\n",
       "      <td>15.357297</td>\n",
       "      <td>14.563453</td>\n",
       "      <td>...</td>\n",
       "      <td>14.861333</td>\n",
       "      <td>15.184035</td>\n",
       "      <td>14.716383</td>\n",
       "      <td>14.885592</td>\n",
       "      <td>14.950120</td>\n",
       "      <td>14.881242</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.600380</td>\n",
       "      <td>15.158579</td>\n",
       "      <td>15.059676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATP6</th>\n",
       "      <td>14.765942</td>\n",
       "      <td>14.939455</td>\n",
       "      <td>14.530669</td>\n",
       "      <td>14.956952</td>\n",
       "      <td>14.960515</td>\n",
       "      <td>14.767122</td>\n",
       "      <td>14.648069</td>\n",
       "      <td>14.847039</td>\n",
       "      <td>14.897657</td>\n",
       "      <td>15.093474</td>\n",
       "      <td>...</td>\n",
       "      <td>14.963645</td>\n",
       "      <td>15.086318</td>\n",
       "      <td>14.994576</td>\n",
       "      <td>15.079311</td>\n",
       "      <td>14.952733</td>\n",
       "      <td>15.047349</td>\n",
       "      <td>14.124653</td>\n",
       "      <td>14.317633</td>\n",
       "      <td>14.954273</td>\n",
       "      <td>14.737609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EHMT1-IT1</th>\n",
       "      <td>1.364011</td>\n",
       "      <td>1.354702</td>\n",
       "      <td>1.371686</td>\n",
       "      <td>1.373798</td>\n",
       "      <td>1.362018</td>\n",
       "      <td>1.357067</td>\n",
       "      <td>1.369805</td>\n",
       "      <td>1.367987</td>\n",
       "      <td>1.367468</td>\n",
       "      <td>1.365011</td>\n",
       "      <td>...</td>\n",
       "      <td>1.369809</td>\n",
       "      <td>1.360947</td>\n",
       "      <td>1.350041</td>\n",
       "      <td>1.365309</td>\n",
       "      <td>1.362940</td>\n",
       "      <td>1.348269</td>\n",
       "      <td>1.372869</td>\n",
       "      <td>1.360560</td>\n",
       "      <td>1.367447</td>\n",
       "      <td>1.359985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GALR3</th>\n",
       "      <td>1.355402</td>\n",
       "      <td>1.341295</td>\n",
       "      <td>1.351569</td>\n",
       "      <td>1.358518</td>\n",
       "      <td>1.340931</td>\n",
       "      <td>1.340861</td>\n",
       "      <td>1.360618</td>\n",
       "      <td>1.353805</td>\n",
       "      <td>1.353732</td>\n",
       "      <td>1.348822</td>\n",
       "      <td>...</td>\n",
       "      <td>1.353969</td>\n",
       "      <td>1.355172</td>\n",
       "      <td>1.331026</td>\n",
       "      <td>1.351185</td>\n",
       "      <td>1.346951</td>\n",
       "      <td>1.326824</td>\n",
       "      <td>1.364472</td>\n",
       "      <td>1.342032</td>\n",
       "      <td>1.356828</td>\n",
       "      <td>1.343060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX1-1</th>\n",
       "      <td>1.213887</td>\n",
       "      <td>1.206120</td>\n",
       "      <td>1.219612</td>\n",
       "      <td>1.222785</td>\n",
       "      <td>1.211382</td>\n",
       "      <td>1.205536</td>\n",
       "      <td>1.224959</td>\n",
       "      <td>1.217283</td>\n",
       "      <td>1.219779</td>\n",
       "      <td>1.212259</td>\n",
       "      <td>...</td>\n",
       "      <td>1.217651</td>\n",
       "      <td>1.212179</td>\n",
       "      <td>1.196986</td>\n",
       "      <td>1.215158</td>\n",
       "      <td>1.211097</td>\n",
       "      <td>1.194717</td>\n",
       "      <td>1.226010</td>\n",
       "      <td>1.206894</td>\n",
       "      <td>1.221090</td>\n",
       "      <td>1.207954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UTS2R</th>\n",
       "      <td>1.198629</td>\n",
       "      <td>1.192367</td>\n",
       "      <td>1.205589</td>\n",
       "      <td>1.208315</td>\n",
       "      <td>1.197830</td>\n",
       "      <td>1.192689</td>\n",
       "      <td>1.211072</td>\n",
       "      <td>1.202751</td>\n",
       "      <td>1.204510</td>\n",
       "      <td>1.198294</td>\n",
       "      <td>...</td>\n",
       "      <td>1.199005</td>\n",
       "      <td>1.198235</td>\n",
       "      <td>1.183507</td>\n",
       "      <td>1.202357</td>\n",
       "      <td>1.194424</td>\n",
       "      <td>1.181320</td>\n",
       "      <td>1.212394</td>\n",
       "      <td>1.193261</td>\n",
       "      <td>1.207828</td>\n",
       "      <td>1.193456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OR7E104P</th>\n",
       "      <td>1.167121</td>\n",
       "      <td>1.160904</td>\n",
       "      <td>1.374079</td>\n",
       "      <td>1.175638</td>\n",
       "      <td>1.165753</td>\n",
       "      <td>1.155603</td>\n",
       "      <td>1.176772</td>\n",
       "      <td>1.172249</td>\n",
       "      <td>1.169585</td>\n",
       "      <td>1.168115</td>\n",
       "      <td>...</td>\n",
       "      <td>1.178356</td>\n",
       "      <td>1.164804</td>\n",
       "      <td>1.154469</td>\n",
       "      <td>1.162814</td>\n",
       "      <td>1.166227</td>\n",
       "      <td>1.148943</td>\n",
       "      <td>1.177165</td>\n",
       "      <td>1.161136</td>\n",
       "      <td>1.174995</td>\n",
       "      <td>1.167486</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21989 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              GSM258551  GSM258552  GSM258553  GSM258554  GSM258555  \\\n",
       "Gene Symbol                                                           \n",
       "COX1          15.312190  15.335128  15.147944  15.235262  15.402517   \n",
       "EEF1A1        15.180497  15.180229  15.262090  15.246653  15.272874   \n",
       "IGK /// IGKC  15.394483  15.375848  15.259431  15.250159  11.996869   \n",
       "TMSB4X        15.164475  15.337793  14.886320  14.723359  14.734361   \n",
       "ATP6          14.765942  14.939455  14.530669  14.956952  14.960515   \n",
       "...                 ...        ...        ...        ...        ...   \n",
       "EHMT1-IT1      1.364011   1.354702   1.371686   1.373798   1.362018   \n",
       "GALR3          1.355402   1.341295   1.351569   1.358518   1.340931   \n",
       "NKX1-1         1.213887   1.206120   1.219612   1.222785   1.211382   \n",
       "UTS2R          1.198629   1.192367   1.205589   1.208315   1.197830   \n",
       "OR7E104P       1.167121   1.160904   1.374079   1.175638   1.165753   \n",
       "\n",
       "              GSM258556  GSM258557  GSM258558  GSM258559  GSM258560  ...  \\\n",
       "Gene Symbol                                                          ...   \n",
       "COX1          15.235435  15.266640  15.236145  15.530658  15.323297  ...   \n",
       "EEF1A1        15.181830  15.162696  15.126815  15.315813  15.406522  ...   \n",
       "IGK /// IGKC  15.053283  15.450697  15.391052  14.658909  14.706833  ...   \n",
       "TMSB4X        14.755676  15.090716  14.747466  15.357297  14.563453  ...   \n",
       "ATP6          14.767122  14.648069  14.847039  14.897657  15.093474  ...   \n",
       "...                 ...        ...        ...        ...        ...  ...   \n",
       "EHMT1-IT1      1.357067   1.369805   1.367987   1.367468   1.365011  ...   \n",
       "GALR3          1.340861   1.360618   1.353805   1.353732   1.348822  ...   \n",
       "NKX1-1         1.205536   1.224959   1.217283   1.219779   1.212259  ...   \n",
       "UTS2R          1.192689   1.211072   1.202751   1.204510   1.198294  ...   \n",
       "OR7E104P       1.155603   1.176772   1.172249   1.169585   1.168115  ...   \n",
       "\n",
       "              GSM258599  GSM258600  GSM258601  GSM258602  GSM258603  \\\n",
       "Gene Symbol                                                           \n",
       "COX1          15.354914  15.495646  15.392720  15.480795  15.372107   \n",
       "EEF1A1        15.165107  15.184177  15.130715  15.191393  15.174592   \n",
       "IGK /// IGKC  15.177085  15.370751  14.985706  14.771899  15.048471   \n",
       "TMSB4X        14.861333  15.184035  14.716383  14.885592  14.950120   \n",
       "ATP6          14.963645  15.086318  14.994576  15.079311  14.952733   \n",
       "...                 ...        ...        ...        ...        ...   \n",
       "EHMT1-IT1      1.369809   1.360947   1.350041   1.365309   1.362940   \n",
       "GALR3          1.353969   1.355172   1.331026   1.351185   1.346951   \n",
       "NKX1-1         1.217651   1.212179   1.196986   1.215158   1.211097   \n",
       "UTS2R          1.199005   1.198235   1.183507   1.202357   1.194424   \n",
       "OR7E104P       1.178356   1.164804   1.154469   1.162814   1.166227   \n",
       "\n",
       "              GSM258604  GSM258605  GSM258606  GSM258607  GSM258608  \n",
       "Gene Symbol                                                          \n",
       "COX1          15.476776  15.125095  14.820450  15.409687  15.145438  \n",
       "EEF1A1        15.192399  15.167122  15.298337  15.110260  15.088462  \n",
       "IGK /// IGKC  15.236298  14.678163  15.120340  15.349818  15.372265  \n",
       "TMSB4X        14.881242  15.236298  14.600380  15.158579  15.059676  \n",
       "ATP6          15.047349  14.124653  14.317633  14.954273  14.737609  \n",
       "...                 ...        ...        ...        ...        ...  \n",
       "EHMT1-IT1      1.348269   1.372869   1.360560   1.367447   1.359985  \n",
       "GALR3          1.326824   1.364472   1.342032   1.356828   1.343060  \n",
       "NKX1-1         1.194717   1.226010   1.206894   1.221090   1.207954  \n",
       "UTS2R          1.181320   1.212394   1.193261   1.207828   1.193456  \n",
       "OR7E104P       1.148943   1.177165   1.161136   1.174995   1.167486  \n",
       "\n",
       "[21989 rows x 58 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# resetting index values by setting Gene Symbol column as index \n",
    "reset_index_df= no_mean_id.set_index([\"Gene Symbol\"])\n",
    "reset_index_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "# to plot and cluster\n",
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns; sns.set()\n",
    "import numpy as np\n",
    "from sklearn.cluster import KMeans\n",
    "from sklearn.datasets._samples_generator import make_blobs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(200, 2)"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAhkAAAGgCAYAAAAHN3WrAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAuLklEQVR4nO3dfXRV1Z3/8c+9eSCXlJgEvDzIwwiliQ8UO2CBAWuri+mStjPVtVyOHVm2A2If1M5Qta2ozBSlTmVA20pblY6rdagMdhZOfRgHO3ZVWYqK1XEVGxFR0CLXCbmkQiD36fcHv8Qk3CT3nHv2Pfuc8379o9zk3uzsJHd/zt7fvU+sUCgUBAAA4LG43w0AAADhRMgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYES13w0oFArK5zkPTJLi8Rh94RB95g795hx95g795lwQ+iwejykWiw37eb6HjHy+oIMHD/vdDN9VV8fV1FSvzs4jymbzfjcnEOgzd+g35+gzd+g354LSZ83N9aqqGj5ksFwCAACMIGQAAAAjCBkAAMAIQgYAADCCkAEAAIwgZAAAACMIGQAAwAhCBgAAMIKQAQAAjCBkAAAAIwgZAADACEIGAAAwgpABAACMIGTAiFS6S9te2a9UusvvpgAAfOL7rd4RPql0l27esF3dmbxqa+L6zpI5SjYm/G4WAKDCmMmA53btS6s7k5ckdWfy2rUv7W+DAAC+IGTAc9MnNaq25vivVm1NXNMnNfrbIACAL1gugeeSjQl9Z8kc7dqX1vRJjSyVAEBEETJgRLIxQbgAgIhjuQQAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIgCSOAQcAeI8trOAYcACAEcxkgGPAAQBGEDLAMeAAACNYLgHHgAMAjCBkQBLHgAMAvMdyCQAAMIKQAQAAjCBkAAAAIwgZiBwOHgOAyqDwE5HCwWMAUDnMZMAYG2cMOHgMACqHmQwYYeuMQc/BYz3t4uAxADCHkAGl0l2eH8RVbMbAhpDBwWMAUDmEjIjzesahJ7A0n1Rn7YwBB48BQGUQMiLOyxmHgYHl6xfP1MFDRyM3Y2BiZggAgoiQEXFe1igMDCwHDx3V/BnjPWppMNhaiwIAfiBkRJyXNQoUVdpbiwIAfiBkwLMaBYoqCVoA0FcoQwZr4t4rtU+jXlRJ0AKAD4QuZLAm7j361JmoBy0A6BG6Ez850dF7JvrUxtNAAQDechwyMpmM1q1bp09+8pP62Mc+pi984Qt68cUXTbTNlZ41cUlWrImHYTD1uk97ZkY2PPKqbt6wPdB9AwAYnOPlkh/96Ef65S9/qdtuu02TJk3SPffcoyuuuEKPPvqoxo4da6KNjti0Jt53maGmOqbPnzNVs1qSgZtK97pP2YEBANHgeCbj17/+tT772c9qwYIFmjJlir71rW/p/fff10svvWSgee4kGxOaP2O87wNX38E0ky1o85O7A3vl7mWfmpptGmrWaLCPhWGmCQBs5Xgmo7GxUU8++aQuu+wyjR8/Xps2bVJtba1OO+00942oDl1piCQp2XTigNydyWv3O4c0YUx9v8erquL9/htmE8bUa/WyuWrbm1bL5EYlm0a6ep2+fZbqONKvOHX1srm9rzvYx4Z6TphF6XfNK/SZO/Sbc2HrM8chY8WKFfqHf/gHnX/++aqqqlI8Htedd96pyZMnu2pAPB5TU1P98J8YQF2vt5/w2IiaKn18xoRBv+eGhmgsGzQ11atl6smevFZDQ0LPt73Xbwnm7fau3td/8fX2oh8b7PGoiMrvmpfoM3foN+fC0meOQ8bu3bvV0NCgu+66S2PHjtXmzZv1zW9+U/fff79aW1sdNyCfL6iz84jj5wXBxNGJ3oOZaqrjuugTU3X2aUmNiEsdHYf7fW5VVVwNDQl1dnYpl8v71OLSpTqOuJqJcPu8Yvr2Wd++rq2Ja+LoRG8fD/axoZ5jipffv1tB+12zAX3mDv3mXFD6rKEhUdJsS6xQKBRKfdF33nlHn/70p3Xfffdp9uzZvY9/4QtfUFNTk+666y7HDc3l8jp40Owbu59KPcSqujqupqZ6dXQcVjZr7y+W5P7cDK/P2xjYZ0P19WAfq+TBbbacNxKk3zVb0Gfu0G/OBaXPmpvrSwoZjmYy/vd//1eZTEYzZszo9/jMmTP129/+1lkLIyKMBzPtaEu52h1ielfJUH3d87GeQs+eUFHJnw+7agBEjaPKkvHjj99Rs62trd/jr732mqZMmeJdq2CtVLpLW57a0/vvmupYybtDKn2GycCdI36fz2HbGS4AYJqjmYyPfvSjmj17tr75zW9q5cqVGjdunLZs2aJnnnlGGzduNNVGWGTXvrQyfabwPn/O1JKvxit5hkmxpQm/ZxJsOsMFACrBUciIx+Nav3697rjjDn3729/WoUOH9JGPfET33XefzjrrLENNhE0G3mV0VkuypOf1rX2YP2O84VYWX5qw4Q6pYVw+A4DBOCr8NCHshZ+lCkqxj+S8WNJUweNQfZZKd+mme59VJltQTXVMq5bO7a3JiPpMQpB+12xBn7lDvzkXlD4zUvgJSM6vxiuxTFE8PMQkFf7/f49jJgEAKoeQgZK5nQUwvUwx8OTOnrqHntqRTLZ/sGE2AwAqg5ARYEE546Gn4HFHW8pI29r2ll5/YctZFQAQBYSMgKr0YOnFksdDT+9Rdyavh57e42l7WyafGCgG28nh9w4TAIgSQkZAVXqwdLLkUWyGxWR7k00jiwaKYvUXNuwwAYCoIGQYUIlljEoPlqWe8TDYDIvp9pZa0Onk+6BuAwDKQ8jwWKWWMUwc7DTcwFrKQD7YjIVNB1EN931QtwEA3iBkeKySyxhebsf0amAdasYiKNtHqdsAAG8QMjwW1DV/rwZWm2Ys3ArqzxAAbEPI8FhQB1kvB9agzFgMxunPkPoNACiOkGFApQdZLwa5csJRGAfZUn+G1G8AwOAIGQFX7iA3MCA4HSCjPshSvwEAgyNkBFw5g5wXASHqgyz1GwAwOEJGwJUzyHkREEwOskFYhglqDQ4AVAIhI+DKGeS8CAimBtnjt2rfrkw2r5rquFYttXcZJuiFrgBgCiGjiCBcQffldpDzKiCYGGR3tKX63UV1R1tKF8yZ4unXAACYRcgYIGqFjFyFAwBMifvdANsUq1OImlS6S9te2a9Uusvoc4YyqyWpmuqYJKmmOqZZLUlPXhcAUDnMZAwQ9d0CbmZyTMz+JBsTWrV0rifLVql0l3a0pSQdDy/M3ABAZRAyBojqboGeOpTOI92Od5yY2sbqxVLO8QLSZ5XJFiRJW57aY3URKQCECSGjiKjVKfSdiaipjqumOqZMtlDyTI7Nsz+79qV7A4Z0vIg0amd5AIBfCBnoNxORyeZ18aemqWFkbckzOTbP/kyf1NgbmiSpptquEAQAYUbIwAkzEW7qFmyd/emp7aAmAwAqj5ARUqWc9dH3c2ydifBCsjHBGRsA4ANCRgiVstuj2OfMnzHepxYDAMKIczJCqJSzPsJ8HojXZ3YAANxhJiOEStntUe6OEFuPXo/aia0AYDNCRgiVstujnB0hXgzkpkJK1G89DwA2IWSEVCm7PdzuCCl3IDc522DzmR0AEDWEDDhW7kBucrbB5jM7ACBqCBlwrNyB3PRsg61ndgBA1BAy4Eo5AzmzDQAQDYQM+ILZBgAIP87JAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAY4SpkbNmyRYsWLdKMGTP0mc98Ro899pjX7QIAAAHnOGQ89NBDuuGGG3TJJZfo4Ycf1qJFi7R8+XL97ne/M9E+AAAQUI5CRqFQ0J133qnLL79cl19+uaZMmaKvfe1r+ou/+As999xzptoIAAACqNrJJ7/xxht655139LnPfa7f4xs2bPC0UQAAIPgchYw333xTknTkyBEtWbJEO3fu1MSJE/WVr3xF5513nvtGVFN/WlUV7/dfDI8+c4d+c44+c4d+cy5sfRYrFAqFUj/5oYce0vXXX6+JEyfqqquuUmtrqx5//HH9+Mc/1r/+679q3rx5jhtQKBQUi8UcPw8AANjN0UxGTU2NJGnJkiW68MILJUmnnXaadu7c6Tpk5PMFdXYecfy8sKmqiquhIaHOzi7lcnm/mxMI9Jk79Jtz9Jk79JtzQemzhoZESbMtjkLGuHHjJEkf+chH+j3+4Q9/WL/5zW+cvFQ/2ay9HVlpuVye/nCIPnOHfnOOPnOHfnMuLH3maNHn9NNPV319vV5++eV+j7/22muaPHmypw0DAADB5mgmo66uTkuXLtVdd92lsWPH6qMf/ageeeQRbdu2Tffdd5+hJgIAgCByFDIk6atf/aoSiYTWrVunAwcOaNq0afrBD36gOXPmmGgfAAAIKMchQ5K+9KUv6Utf+pLXbQEAACESjo24AADAOoQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGEDIAAIARhAwAJ0ilu7Ttlf1Kpbusfk0Adqv2uwEA7JJKd+nmDdvVncmrtiau7yyZo2RjwrrXhHdS6S7t2pfW9EmN/FzgKUIGgH527UurO5OXJHVn8tq1L132wGPiNeENAiBMYrkEQD/TJzWqtub4W0NtTVzTJzVa+ZqVkkp36amX/6h32w/73RQjigVAwCvMZADoJ9mY0HeWzPF0+tzEa1ZC36v8n/1Xm25dNkfNo+r8bpanegJgz0xGkAIg7EfIAHCCZGPC8yBg4jVN63uVfyyTU9vetOadMc7nVnkrqAEQwUDIAIBB9L3KH1FTpZbJjX43yYggBkAEAyEDAAbRc5W/+51D+viMCRoRl7LZvN/NAgKDkAEAQ0g2JjRhTL2amurV0RHO4k/AFHaXAIBDYT1YLKzfF/zDTAYAOBDWcyXC+n3BX8xkAIADYT1XIqzfF/xFyAAAB4J8sNhQwvp9wV8slwAINa/vyxHWcyXC+n3BX4QMAKFlqs4gKOdKOA1YQfm+EByEDAChFeUbs1HICRtQkwHAF5XYLhnlOgMKOWEDZjIAVFylrrKjXGfAjc9gA0IGACOGqgeo5DKGyToDr4tKvRTlgAV7EDIAeK7YTIWk3gEvDFfZQah5oJATfiNkAPDcwJmKHW0pPfT0nn4DctCvsqNcVAqUipABwHN9Zyqqq2J6vytzwoA8f8b4QA/KYZiNAUwjZADwXLIxoa9fPFNrN72kbK6gJ17Yp5rquDLZ8AzI1DwAwyNkADDi4KGjyuUKkqRMtqCLPzVVDSNrQzUgD1bzYHNBKFBJhAzAQmEYpAYuJ8xqSQb2e3EiCAWhQKUQMgDLhGWQiupyAgWhwAcIGYBlwjRIRXELJQWhwAcIGYBlbBukwrB0U0lRncEBiiFkAJaxaZAKy9JNqfoGKumDw8MmjKkv+jmD9UUUZ3CAYggZgIVsGaT8WrrxY/akb6CqqY5LKiiTLai2Jq7Vy+aqqaleqY4jvZ9TVRXT8kvO0mmTmyrSPiCIuAsrgEH5cRfTnsF+wyOv6uYN243epbWvvoEqk80rkz2+/bY7k1fb3rTebT+sX217s/dzcrmC1m16qaT2VeKOs4CNmMkAMCg/lm78mj3pWwszcCZjzEl1uur2J3Usk+v3nGyuMGz7orbkBPRFyAAwpEov3fhV+DowUEkf1GTsfudQv4ARj8WULxRKal+YdgsBThEyAFjF1OyJm4LNnv+vroppRE2VjmVyqq2J6+sXz9TBQ0dLap/T0JRKd2lHW0qSInOAGcIrVigUCn42IJfL6+DBw342wQrV1XE1NdWro+Owstm8380JBPrMnSj2W7lLFtXVcR3LS8+98kdNO+UkxwN/qYWsqXSXbrr32d56kJrquFYtDe7yShR/18oVlD5rbq5XVdXwZZ0UfgKwiokiyWK3nndq3Oh6nTNzgqsBP9mYKOmus7v2pXsDhnS8AHXXvrTjrwfYguUSANYwVSQ5fVJj711gJWnLU29YuRRxvJ2xfjMZfh/GBpSDkAHAGqaKJJONCX3+nFO1+cndko7fFdbGAsxkY0Krls6lJgOhQcgAYA2TO0tmtST10NN7rDmufTDJxoQumDPF72YAniBkALCGyXM5ynntVMcRvfh6uyaOTqh5VJ1nbTLF6Ymp3J8GphAyAAyrkoOQyXM53Lx20A7TctreoH1/CBZCBoAhBX0QKjcgBe0wLaftDdr3h2AhZAAYko2DUE9waD6pbshDsbwISH6dQOqW0/YG7ftDsBAyAAzJtkGob3DoMViA8CIgJRsTWr1srt5u7+pXk2FrHYPT2hM/7k+D6HAdMvbs2aOLLrpIN910ky666CIv2wTAIrYNQn2DQ4++AaLv4O9VQEo2jVTL1JN7T2G0fQnJae1Jpe9Pg+hwFTIymYyuvfZaHTlyxOv2ALCQTYNQ3+DQoydAFBv8TQQkG5eQABu5Chk/+MEPVF9f73VbAKDXYMsRfWdWBtZkbHtl/wmDfynHeTtl2xISYCvHIeP555/Xpk2btGXLFn3yk5800CQAUTfccsRgMyuVGvxtW0ICbOUoZHR2dur666/XjTfeqPHjx3vXiGru09ZzN7tS7mqH4+gzd4LQb7vfOdRvRmL3O4c0Yczws6cTxtRr9bK5atubVsvkRiWbRnrSnmJ9NmFM/bBtSnUc8bwtQRKE3zXbhK3PHIWMf/zHf9RZZ52lz33uc541IB6PqanJ3NLLu+2H9fs32nXG1NEaN9r+JZ6GBq6InKLP3LG53z4+Y4J+9l9tOpbJaURNlT4+Y0LJ7xNNTfVqmXqyJO///p302bvth7Xi7u2938MPr/uU6zYE7X1sIJt/12wVlj4rOWRs2bJFL7zwgn71q1952oB8vqDOTjMFpKmOI7rh7md7p05XL5tr7dVEVVVcDQ0JdXZ2KZfLD/8E0GcuBaHfRsSlW5fN6Z0FGBGXOjoOO3oNL//+3fTZc6/8UccyOUnSsUxOz73yR50zc4Ljrx2k97GBgvC7Zpug9FlDQ6Kk2ZaSQ8Yvf/lLtbe3n1CHsXLlSm3YsEGPPPKI40b2yGbNdOSrb3b0m3J99c0O6+87kMvljfVHWNFn7tjeb82j6jTvjHGS3L1HmPj7H67P+harTjvlpH71IdNOOcma76PSbP9ds1FY+qzkkLFmzRodPXq032N/+Zd/qWuuuUaLFi3yvGFeoAIciK5K//2b2j7L+xiCrOSQMXbs2KKPjx49WqeccopnDfISFeBAdFX677/Y2RlebJ/lfQxBFvpjxW06RAhAZVXy79/kjAPvYwiqskJGW1ubV+0AgEBjxgE4UehnMirN1psmATCPGQegP0KGh2y/aRIQBQR9wB6EDA9V4qZJvIECg4t60Of9AbYhZHjI9FazqL+BAsOJ8t1ReX+AjQgZHjJd+BXlN1CgFFE+U8Kv9wdmTzAUQobHTBZ+RfkNFChFlHd4+PH+wOwJhkPICJAov4EinExcBUd1h4cf7w/lzp4wCxJ+hIyACeMbKG800cRVsPcq/f5QzuwJP/9oIGT4hIH1ON5ooitKNUZh/XsvZ/YkSj//KCNk+KDYwDphTL3fzfIFbzTRFZUao7AHabezJ1H5+UcdIcMHxQZWP0OGn1dZvNFEV1RqjAjSxUXl5x91hAwf2DSw+n2V5eUbTVinpMOslKvgoP9cbfp7t00Ya8zQHyHDBzYl+IFXWTvaUrpgzpSKtsGLNxq/wxLMCMPP1aa/d6DSCBk+sSXBT5/UqJrquDLZ40Fjy1NvaFZL0oq2OcGUdDiF5edqy987UGlxvxsAfyUbE/r8Oaf2/juTLWjXvrR/DXKpZ0paElPSIcLPFQg2ZjKgWS1JPfT0nkCvGTMlHU78XIFgI2RUiM3Fa2F5I2dKOjwG/r3wcwWCiZBRAUEoXuONHLYIwt8LgNJQk1EBxYrXoiCV7tK2V/Yrle7y5PMQDVH9ewHCiJmMEpS71BGFffID+6jUq1GuWjFQFP5egKggZAzDi0EwDDUPQwWtYn1U6tbDsGxRhHfC8PcC4DhCxjC8GgSDXPMwXNAq1kelXo1y1Ypigvz3AuADhIxhBHEQ9Hony3BBq1gflXo1ylUrAIQXIWMYQRkEe4JF80l1unPzy2XXOPQNKsMFrcH6qNSr0TBctdq8RRkA/ELIKIHtg2Df5YzqqpiyuYKk47MOjz7zphbN+zNH7S+2PDJc0LK9j0yieBUAimMLawj0Xc7I5gqqqor1fuy3L+/XzRu2O9oeOtjyyPwZ430bPG3e5sqWSwAojpARAgPv77D8krP0iZnjez/udOAb7n4RlR7we2YKNjzyquPAVAncXwMAiovcckkY186L1USMbqjTszsPuCpYHaoOxY+lAdu3uQalbgcAKi1SISPMa+cDayLKHfgGq7HwY8APwg6fKNekAMBgIhUyyh0ggzYLMnDg86L95Qz4br8+MwUAEEyRChnlDpBBngXxqv1uB/xyvz4zBQAQPJEKGeVcEdtaF1Dq7ICX7Xcz4Nvaf34J2qwYALgRqZAhub8itrEuwMnsgMn2lzJgNp9U13uGhy3955egz4oBQKkiFzLcsrEuwMnsgKn2lzJgptJdunPzy71neHz94plW9J9fmNUBEBWEDAdsqwtwOjthov2lDJh9PyeXK+jgoaOetiFobJwVAwATCBkB99cLTpUkzWpJ+hKAShkwGVT7s3FWDABMIGQE1MBlilktybJeq9QBb+DnljJgMqieyLZZMQAwgZBhiVTHEb34ersmjk6oeVTdsJ/v1bq+kyLEwT63lAGTQRUAooeQYYFUuks33fusMtmCaqpjWrV07pADcirdpT+2H1Y8LuXz5d0vw0lYoWARAOAEN0grg1c3CtvRllIme/z27JlsQTvaUkN+zZvu3a7Hnt2rfF6Kx1TWbg0nN/cycSOwYn1o8x1XAQClYybDJafLDF7VI+zal1Ymm+/9d74gvbm/U6dNbnL1ek7qJbyurSjWh5I4QwIAQoKQ4VKpSwelhJFZLUlteWqPMtm8aqqHLuKcPqlRVfGYcvmCZ9+Lk3qJcmorBoatYn3Y8/99HyNkAEAwETJcKnVbZilhJNmY0HevnKu327uGLfxMNia0/G/O0toHfqdcXqqpjpW1s6RSioWtwfqQ7a4AEA6EDJdKXTooNYwkm0aqZerJ6ug4rGyf5ZBiTpvcpFuXzfN0S6jpe2kUC1vzZ4wv2odsdwWAcCBklKHUrZsmBk0vt4RW4l4ag4WtYt8H210BIBwIGRVg+6BZia2pHMgFANFDyEDFjv22PWwBALxFyACzDAAAIwgZkMQsAwDAe5z4CQAAjCBkAD7iCHUAYcZyCeCTSmwdBgA/MZMB+GSwY9UBICwIGYBPTNzVFgBswnIJ4BO2DgMIO0KGh0zf/wPhw9ZhAGFGyPBIJYr4CDEAgCAhZHjExP0/+oYKSexEAAAECiHDI17f/2PgzMhfLzjV+E3MAADwEiHDI14X8Q2cGZFUkZuYAQDgFUKGh7ws4hs4MzKrJalZLUlqMgAAgeEoZKTTaa1du1a/+c1v9P7776ulpUXf+MY3NHv2bFPti6zBZkbKCRcUjgIAKslRyFi+fLna29u1du1aNTc3a+PGjVqyZIn+4z/+Q9OmTTPVxsjycmaEI6wBAJVW8omfb731lrZt26aVK1dq9uzZmjp1qlasWKGxY8fq4YcfNtlGeIAjrAEAlVZyyGhqatLdd9+tM888s/exWCymQqGgQ4cOGWkcvMMR1gCASit5uaShoUHnnntuv8cee+wx7d27VwsWLCivEdXcQqWqKt7vv16bMKZeq5fNVdvetFomNyrZNNLI16kk030WVvSbc/SZO/Sbc2Hrs1ihUCi4eeKOHTu0dOlSzZs3T+vXr3fdgEKhoFgs5vr5AADATq5CxhNPPKFrr71WM2fO1E9+8hPV1dW5bkAul1dnZ5fr54dFVVVcDQ0JdXZ2KZfL+92cQKDP3KHfnKPP3KHfnAtKnzU0JEqabXF8Tsb999+vW2+9VQsXLtSaNWtUW1vrqoF9ZbP2dmSl5XJ5+sOhcvosytt6+V1zjj5zh35zLix95ihkbNy4UatWrdLixYt1ww03KB4Px5oRooltvYgSt4E6ykEc5Ss5ZOzZs0erV6/WwoULdeWVV6q9vb33Y3V1dRo1apSRBgKmmLipHWAjt4G67/NqquP6/DmnalZLkr8TlKzkkPH4448rk8lo69at2rp1a7+PXXjhhbrttts8bxxgktc3tQNs5TZQ931eJpvX5id366Gn9zDrh5KVHDK+/OUv68tf/rLJtgAV5fVN7QBbuQ3UfZ/Xg1k/OMEN0hBpXh7dDtjKbaDued6OtpS2PPWGMtmC41m/d9sP67lX/qhpp5zE31oEETIAIALcBupkY0IXzJni6i7QqY4jWnH3dh3L5CiujihCBgBgWG5CStvetI5lcpJYZokq9qACQASl0l3a9sp+pdLmDkNsmdyoETVVkrhnUlQxkwEAEVOpM2KSTSP1w+s+RU2GQ2GqYyFkAEDEVPKMmHGj63XOzAmhOL2yEsJWx8JyCQBETM/WVIllDNsUq2MJMmYyACBignBGTFSPM++pY+mZyRgqAAahjwgZMCIIv/xAlNl8RkyU7ytUah1LUPqIkAHPBeWXH4Cdon5foVLqWILSR9RkwHPFfvkBoFTUjAwvKH3ETAY8x43HAHsEcekyCDUjfgtKHxEy4Lmg/PIDYRfkpUuba0ZsEYQ+ImTAiCD88gNhF5R1e4QXNRkAEFJBWbf3SiWOSg8SG/qDmQwACKkoLV0GeWmoRyrdpd3vHNLHZ0zQiDKnAGzpD2YyACAg3FyZJhsTmj9jfOAGXKeG2tVmwxX9cHpCwT2/2qmrbn9SqY4jZb2eLbv8mMkAgACw5cq0GBt2sAy2q83mfuurbyg4lsmpbW9a884Y5/r1bNnlR8gAgACwtYjTlkF8sKUhW/ttoL6hYERNlVomN5b1erYslREyACAASrkyLWdGwe1zbRrEi+1qc3NF78fMTE8o6FuTUe6da23Y5UfIAIAAGO7KtJwZhXKe68W0vMlB3ekVvZ8zM8nGhCaMqVdTU706Og5X5GuaRsgAgIAY6srUyYzCwEG9nNmIcqflKzGoO7mit2lmJgwIGQAQAqXOKBQb1MudjShnWt62Qd2WgsmwIGQAgM+8WC4odUah2KA+f8Z434oEbRvUbSmYDAtCBgD4yMvlglJmFAYb1P0qEjQxqJcb2mwomAwLQgYA+KjSywU2Xql7Oag7CW02nO8RdoQMAPCR6a2pxYT5Sr3U0EYYqQxCBgD4yOTW1CgqtcbDRBjBiQgZAOAzr7amovTlIK/DCIojZACAxWzbfREEpSwHeR1GUBwhAwAsVspyCvUC7vS9v0nffw/8HNsKZYOEkAEAlhvsytyreoHhgkpYg0yp/RfmQlnTCBkAEFBe1AsMN9CGufCRegvz4n43AADgTk+9gCTX9QLFBlonHw8yL/oPQ2MmAwACyot6geaT6lRdFVM2Vyg60Npc+OjFyZ5fv3imtv/+Xc05YxyzGAYQMgAgwMqpF0ilu3Tn5peVzRVUVRXT1y+eecJr2Vr42HcZp6Y6rs+fc6pmtSQdta/n++/O5PXszgOhWgqyBcslABBRfZdCcrmCDh46WvTzko0JzZ8xvncATqW7tO2V/UqluyrW1oH6tj2TzWvzk7t184btjtoU5qUgWzCTAQAR5WYpxJZC0L5t7+G0eNPmpaCwIGQAQES5WQqxZUdGT9t3tKW05ak3lMkWrykp5TVsWwoKE0IGAASQV2dXOK3psOnqP9mY0AVzpmhWS9J1X3AGhlmEDACwWLEw4eeShY1X/wQFexEyAMBSg4UJv5csGNRRKnaXAIClBtv9wCFSCApmMgDAUoPVP9i4ZFFMquOIXny9XRNHJ9Q8qs7v5kgK731YbEXIAABLDRUm3C5ZVGqQtWWrq+1tCjtCBgBYzMv6h0oOsgOXena0pdQwstbXGQS/a1miiJABABExcJB99Jk3tWjenxkZaPsu9dRUx7TlqT3KZO05wItalsogZABARAw8JfO3L+83ds+OZGNCq5fN1dvtXdr/3p+06devS7LjAC9qMiqH3SUAEBE9g+wnZo7vfczkPTuSTSN1/tmTdXZr0prdMAPvwwKzmMkAgAhJNia0aN6f6dmdByq2bJBsGskMQkQRMgAgYvxYNuAAr2giZABABAVp0E+lu7SjLSVJmtWSrFi7OVOjfIQMAIC1Uuku3XTvdmWyx4tVtzz1hlYtnWt80OdMDW9Q+AkAMCrVcUTbXtmvVLrL8XN37Uv3BgxJymQLxgpVB37dYke6wxlmMgAAxrzbflg33P2s6xmB6ZMaVVMd7w0aNdWxiuxO4UwNbxAyAADG/P6N9rJO2Uw2JrRq6ZyK12RwpoY3CBkAAEecFESeMXV02TMCycaELpgzxWVr3QtScaytCBkAgJI5LYgcN7peq5fN1atvdjAjEEGEDABAydzcZCzZNNKaW72jsthdAgAoWU9BpOT/EeGwHzMZABAipg+QoiASTjgOGfl8Xj/84Q+1efNmdXZ2atasWVq5cqWmTKl8UQ4A4AOVOkDK1oJIv04GxeAcL5esX79eDzzwgG655RZt2rRJsVhMV1xxhbq7u020DwBQoigfIHX8ZNBntfnJ3dr85G7ddO92V4d/wVuOQkZ3d7d++tOf6uqrr9a5556r1tZWrVu3TgcOHNDWrVtNtREAUIIo10scPxm00PvvTDZaIctWjpZL/vCHP+jw4cOaO3du72MNDQ06/fTT9fzzz+szn/mM5w0EAJQmyvUSx08GjfUGjZrqaIUsWzkKGe+++64kafz48f0eTyaT2r9/v/tGVLPJpaoq3u+/GB595g795lyQ+mzCmHpNGFPvdzMkVbbfJoyp13evnKfnX01JMens1qSSTSONf12vBel3rRSOQkZX1/H1rdra2n6PjxgxQocOHXLVgHg8pqYmO/4gbNDQEJ0rD6/QZ+7Qb87RZ+5Uqt+amurVMvXkinwt08Lyu+YoZNTVHT9Mpbu7u/f/JenYsWNKJNx1SD5fUGfnEVfPDZOqqrgaGhLq7OxSLpcf/gmgz1yi35yjz9yh35wLSp81NCRKmm1xFDJ6lklSqZQmT57c+3gqlVJra6vDJn4gm7W3Iystl8vTHw7RZ+7Qb87RZ+7Qb86Fpc8cLfq0trbqQx/6kLZv3977WGdnp3bu3KnZs2d73jgAABBcjmYyamtrddlll2nNmjVqbm7WKaecottvv13jxo3TwoULTbURAAAEkOMTP6+55hpls1ndeOONOnr0qM4++2xt2LDhhGJQAAAQbY5DRlVVla677jpdd911JtoDAPCB6XueIJq4QRoARFyl7nmC6AnHaR8AANeifM8TmEXIAICIi/I9T2AWyyUAEHFRvucJzCJkAACUbEwQLuA5lksAAIARhAwAAGAEIQMAABhByAAAAEYQMgAAgBGEDAAAYAQhAwAAGEHIAAAARhAyAACAEYQMAABgBCEDAAAYQcgAAABGxAqFQsHPBhQKBeXzvjbBGlVVceVyeb+bESj0mTv0m3P0mTv0m3NB6LN4PKZYLDbs5/keMgAAQDixXAIAAIwgZAAAACMIGQAAwAhCBgAAMIKQAQAAjCBkAAAAIwgZAADACEIGAAAwgpABAACMIGQAAAAjCBkAAMAIQgYAADCCkAEAAIwgZFhm//79Wr58uebPn6+zzz5bS5Ys0a5du/xuVqCsWLFC3/rWt/xuhnXy+by+//3v65xzztHMmTP1d3/3d3rrrbf8blagrF+/XosXL/a7GdZLp9O6+eab9YlPfEJ//ud/rksvvVQvvPCC382yWnt7u6677jrNnTtXH/vYx7Rs2TK9/vrrfjerbIQMi3R3d2vZsmVqb2/XT37yE23cuFGjRo3S5ZdfroMHD/rdPOvlcjn98z//sx588EG/m2Kl9evX64EHHtAtt9yiTZs2KRaL6YorrlB3d7ffTQuE++67T9///vf9bkYgLF++XC+//LLWrl2rBx98UGeccYaWLFmi3bt3+900a33lK1/Rvn37dM899+jBBx9UXV2dvvjFL6qrq8vvppWFkGGRF154Qa+99pq+973v6cwzz9T06dP1ve99T0eOHNH//M//+N08q+3evVuXXnqptmzZogkTJvjdHOt0d3frpz/9qa6++mqde+65am1t1bp163TgwAFt3brV7+ZZ7cCBA1q6dKnuvPNOnXrqqX43x3pvvfWWtm3bppUrV2r27NmaOnWqVqxYobFjx+rhhx/2u3lW6ujo0MSJE7Vq1SrNmDFD06ZN01e/+lW99957gZ/JJmRYZPr06br77rs1duzYfo8XCgUdOnTIp1YFw3PPPafTTjtNDz/8sCZOnOh3c6zzhz/8QYcPH9bcuXN7H2toaNDpp5+u559/3seW2e/3v/+9TjrpJP3nf/6nZs6c6XdzrNfU1KS7775bZ555Zu9jsViM97EhNDU1ae3atZo+fbok6f/+7/+0YcMGjRs3Th/+8Id9bl15qv1uAD5w8skn69xzz+332M9+9jMdO3ZM8+fP96lVwXDppZf63QSrvfvuu5Kk8ePH93s8mUxq//79fjQpMM477zydd955fjcjMBoaGk54H3vssce0d+9eLViwwKdWBcdNN92kf//3f1dtba1+9KMfaeTIkX43qSyEjAp6++23df755w/68aefflonn3xy77//+7//W+vWrdPixYvV2tpaiSZayWm/4UQ967q1tbX9Hh8xYgRXlzBqx44duuGGG3T++ecT1kpw+eWX65JLLtEvfvELfe1rX9PGjRt1xhln+N0s1wgZFTR27Fg9+uijg368ubm59/9/8YtfaNWqVVq0aJG+/e1vV6J51nLSbyiurq5O0vHajJ7/l6Rjx44pkUj41SyE3BNPPKFrr71WM2fO1Nq1a/1uTiD0LI+sWrVKL730ku6//35997vf9blV7hEyKqimpkbTpk0b9vPWrFmje+65R4sXL9aKFSsUi8Uq0Dp7ldpvGFzPMkkqldLkyZN7H0+lUpGeJYM5999/v2699VYtXLhQa9asOWEWDR9ob2/XM888owsuuEBVVVWSpHg8rmnTpimVSvncuvJQ+GmZ22+/Xffcc4+uv/563XjjjZEPGPBGa2urPvShD2n79u29j3V2dmrnzp2aPXu2jy1DGG3cuFGrVq3S3/7t3+qOO+4gYAwjlUrpG9/4hp577rnexzKZjHbu3Bn4CyxmMiyyfft23XvvvVq8eLH+6q/+Su+9917vx0aOHKn6+nofW4cgq62t1WWXXaY1a9aoublZp5xyim6//XaNGzdOCxcu9Lt5CJE9e/Zo9erVWrhwoa688kq1t7f3fqyurk6jRo3ysXV2am1t1YIFC/RP//RPuuWWW9TQ0KAf//jH6uzs1Be/+EW/m1cWQoZFevaQ//znP9fPf/7zfh+76qqrdPXVV/vRLITENddco2w2qxtvvFFHjx7V2WefrQ0bNnCVCU89/vjjymQy2rp16wlnsFx44YW67bbbfGqZvWKxmO644w79y7/8i/7+7/9ef/rTnzR79mz927/9W+DP/YkVCoWC340AAADhQ00GAAAwgpABAACMIGQAAAAjCBkAAMAIQgYAADCCkAEAAIwgZAAAACMIGQAAwAhCBgAAMIKQAQAAjCBkAAAAI/4fUmXhMgze4aoAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# just simulating and visualizing clustering of synthetic data using make_blobs() from scikit-learn\n",
    "# just to test how the clustering might look for our data\n",
    "\n",
    "# number of samples from our dataset \n",
    "n_samples = 200\n",
    "# how many k means clusters should the model make\n",
    "k_clusters = 4\n",
    "# Smaller cluster_std: Tighter, more compact clusters.\n",
    "# Larger cluster_std: More spread out, less compact clusters.\n",
    "# The choice of cluster_std depends on the characteristics you want in your synthetic data and the specific requirements of your clustering experiment.\n",
    "cluster_std  = 0.50\n",
    "\n",
    "# synthetic data made using the info passed into make_blobs() \n",
    "X, y_true = make_blobs(n_samples = n_samples, centers = k_clusters, cluster_std = cluster_std, random_state = 0)\n",
    "\n",
    "# X[:, 0] represents the first feature of the data points (x)\n",
    "# and X[:, 1] represents the second feature (y)\n",
    "#  The s=20 parameter specifies the size of the markers in the plot\n",
    "plt.scatter(X[:, 0], X[:, 1], s = 3);\n",
    "## plt.show()\n",
    "\n",
    "X.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>COX1</th>\n",
       "      <th>EEF1A1</th>\n",
       "      <th>IGK /// IGKC</th>\n",
       "      <th>TMSB4X</th>\n",
       "      <th>ATP6</th>\n",
       "      <th>RPLP1</th>\n",
       "      <th>RPL37</th>\n",
       "      <th>RPL37A</th>\n",
       "      <th>IGLC1</th>\n",
       "      <th>RPL41</th>\n",
       "      <th>...</th>\n",
       "      <th>LOC100130078</th>\n",
       "      <th>BAGE2 /// BAGE4</th>\n",
       "      <th>VPS9D1-AS1</th>\n",
       "      <th>LOC100128343</th>\n",
       "      <th>SNORA78</th>\n",
       "      <th>EHMT1-IT1</th>\n",
       "      <th>GALR3</th>\n",
       "      <th>NKX1-1</th>\n",
       "      <th>UTS2R</th>\n",
       "      <th>OR7E104P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM258551</th>\n",
       "      <td>15.312190</td>\n",
       "      <td>15.180497</td>\n",
       "      <td>15.394483</td>\n",
       "      <td>15.164475</td>\n",
       "      <td>14.765942</td>\n",
       "      <td>14.899661</td>\n",
       "      <td>14.607402</td>\n",
       "      <td>14.805437</td>\n",
       "      <td>15.389362</td>\n",
       "      <td>14.764169</td>\n",
       "      <td>...</td>\n",
       "      <td>1.559278</td>\n",
       "      <td>1.508207</td>\n",
       "      <td>1.488264</td>\n",
       "      <td>1.428252</td>\n",
       "      <td>1.402514</td>\n",
       "      <td>1.364011</td>\n",
       "      <td>1.355402</td>\n",
       "      <td>1.213887</td>\n",
       "      <td>1.198629</td>\n",
       "      <td>1.167121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258552</th>\n",
       "      <td>15.335128</td>\n",
       "      <td>15.180229</td>\n",
       "      <td>15.375848</td>\n",
       "      <td>15.337793</td>\n",
       "      <td>14.939455</td>\n",
       "      <td>14.798201</td>\n",
       "      <td>14.620797</td>\n",
       "      <td>14.818706</td>\n",
       "      <td>15.287337</td>\n",
       "      <td>14.722343</td>\n",
       "      <td>...</td>\n",
       "      <td>1.541614</td>\n",
       "      <td>1.491403</td>\n",
       "      <td>1.476264</td>\n",
       "      <td>1.415345</td>\n",
       "      <td>1.393830</td>\n",
       "      <td>1.354702</td>\n",
       "      <td>1.341295</td>\n",
       "      <td>1.206120</td>\n",
       "      <td>1.192367</td>\n",
       "      <td>1.160904</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258553</th>\n",
       "      <td>15.147944</td>\n",
       "      <td>15.262090</td>\n",
       "      <td>15.259431</td>\n",
       "      <td>14.886320</td>\n",
       "      <td>14.530669</td>\n",
       "      <td>15.114353</td>\n",
       "      <td>14.861911</td>\n",
       "      <td>14.920742</td>\n",
       "      <td>15.084031</td>\n",
       "      <td>14.636443</td>\n",
       "      <td>...</td>\n",
       "      <td>1.562901</td>\n",
       "      <td>1.512844</td>\n",
       "      <td>1.494091</td>\n",
       "      <td>1.419606</td>\n",
       "      <td>1.408625</td>\n",
       "      <td>1.371686</td>\n",
       "      <td>1.351569</td>\n",
       "      <td>1.219612</td>\n",
       "      <td>1.205589</td>\n",
       "      <td>1.374079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258554</th>\n",
       "      <td>15.235262</td>\n",
       "      <td>15.246653</td>\n",
       "      <td>15.250159</td>\n",
       "      <td>14.723359</td>\n",
       "      <td>14.956952</td>\n",
       "      <td>14.861104</td>\n",
       "      <td>14.815595</td>\n",
       "      <td>14.581683</td>\n",
       "      <td>15.245323</td>\n",
       "      <td>14.623252</td>\n",
       "      <td>...</td>\n",
       "      <td>1.564354</td>\n",
       "      <td>1.515226</td>\n",
       "      <td>1.496433</td>\n",
       "      <td>1.429780</td>\n",
       "      <td>1.410923</td>\n",
       "      <td>1.373798</td>\n",
       "      <td>1.358518</td>\n",
       "      <td>1.222785</td>\n",
       "      <td>1.208315</td>\n",
       "      <td>1.175638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258555</th>\n",
       "      <td>15.402517</td>\n",
       "      <td>15.272874</td>\n",
       "      <td>11.996869</td>\n",
       "      <td>14.734361</td>\n",
       "      <td>14.960515</td>\n",
       "      <td>15.183791</td>\n",
       "      <td>14.854306</td>\n",
       "      <td>14.807841</td>\n",
       "      <td>11.158493</td>\n",
       "      <td>14.870209</td>\n",
       "      <td>...</td>\n",
       "      <td>1.552817</td>\n",
       "      <td>1.498749</td>\n",
       "      <td>1.480635</td>\n",
       "      <td>1.414616</td>\n",
       "      <td>1.398829</td>\n",
       "      <td>1.362018</td>\n",
       "      <td>1.340931</td>\n",
       "      <td>1.211382</td>\n",
       "      <td>1.197830</td>\n",
       "      <td>1.165753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258556</th>\n",
       "      <td>15.235435</td>\n",
       "      <td>15.181830</td>\n",
       "      <td>15.053283</td>\n",
       "      <td>14.755676</td>\n",
       "      <td>14.767122</td>\n",
       "      <td>14.844315</td>\n",
       "      <td>14.842160</td>\n",
       "      <td>15.027328</td>\n",
       "      <td>14.795219</td>\n",
       "      <td>14.864671</td>\n",
       "      <td>...</td>\n",
       "      <td>1.547007</td>\n",
       "      <td>1.496887</td>\n",
       "      <td>1.476711</td>\n",
       "      <td>1.411491</td>\n",
       "      <td>1.393893</td>\n",
       "      <td>1.357067</td>\n",
       "      <td>1.340861</td>\n",
       "      <td>1.205536</td>\n",
       "      <td>1.192689</td>\n",
       "      <td>1.155603</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258557</th>\n",
       "      <td>15.266640</td>\n",
       "      <td>15.162696</td>\n",
       "      <td>15.450697</td>\n",
       "      <td>15.090716</td>\n",
       "      <td>14.648069</td>\n",
       "      <td>14.873081</td>\n",
       "      <td>14.714121</td>\n",
       "      <td>14.648810</td>\n",
       "      <td>15.378435</td>\n",
       "      <td>14.732511</td>\n",
       "      <td>...</td>\n",
       "      <td>1.560682</td>\n",
       "      <td>1.510637</td>\n",
       "      <td>1.491109</td>\n",
       "      <td>1.439103</td>\n",
       "      <td>1.410817</td>\n",
       "      <td>1.369805</td>\n",
       "      <td>1.360618</td>\n",
       "      <td>1.224959</td>\n",
       "      <td>1.211072</td>\n",
       "      <td>1.176772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258558</th>\n",
       "      <td>15.236145</td>\n",
       "      <td>15.126815</td>\n",
       "      <td>15.391052</td>\n",
       "      <td>14.747466</td>\n",
       "      <td>14.847039</td>\n",
       "      <td>14.989528</td>\n",
       "      <td>14.802895</td>\n",
       "      <td>14.832079</td>\n",
       "      <td>15.323409</td>\n",
       "      <td>14.704899</td>\n",
       "      <td>...</td>\n",
       "      <td>1.559284</td>\n",
       "      <td>1.508974</td>\n",
       "      <td>1.491793</td>\n",
       "      <td>1.430029</td>\n",
       "      <td>1.408326</td>\n",
       "      <td>1.367987</td>\n",
       "      <td>1.353805</td>\n",
       "      <td>1.217283</td>\n",
       "      <td>1.202751</td>\n",
       "      <td>1.172249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258559</th>\n",
       "      <td>15.530658</td>\n",
       "      <td>15.315813</td>\n",
       "      <td>14.658909</td>\n",
       "      <td>15.357297</td>\n",
       "      <td>14.897657</td>\n",
       "      <td>14.795771</td>\n",
       "      <td>14.918985</td>\n",
       "      <td>14.740109</td>\n",
       "      <td>14.271701</td>\n",
       "      <td>14.745470</td>\n",
       "      <td>...</td>\n",
       "      <td>1.557593</td>\n",
       "      <td>1.508101</td>\n",
       "      <td>1.488935</td>\n",
       "      <td>1.418820</td>\n",
       "      <td>1.408168</td>\n",
       "      <td>1.367468</td>\n",
       "      <td>1.353732</td>\n",
       "      <td>1.219779</td>\n",
       "      <td>1.204510</td>\n",
       "      <td>1.169585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258560</th>\n",
       "      <td>15.323297</td>\n",
       "      <td>15.406522</td>\n",
       "      <td>14.706833</td>\n",
       "      <td>14.563453</td>\n",
       "      <td>15.093474</td>\n",
       "      <td>14.843761</td>\n",
       "      <td>15.142578</td>\n",
       "      <td>14.671068</td>\n",
       "      <td>14.123293</td>\n",
       "      <td>14.866845</td>\n",
       "      <td>...</td>\n",
       "      <td>1.554900</td>\n",
       "      <td>2.500992</td>\n",
       "      <td>1.484531</td>\n",
       "      <td>1.412437</td>\n",
       "      <td>1.401998</td>\n",
       "      <td>1.365011</td>\n",
       "      <td>1.348822</td>\n",
       "      <td>1.212259</td>\n",
       "      <td>1.198294</td>\n",
       "      <td>1.168115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258561</th>\n",
       "      <td>15.440451</td>\n",
       "      <td>15.309941</td>\n",
       "      <td>14.756265</td>\n",
       "      <td>14.946459</td>\n",
       "      <td>15.127632</td>\n",
       "      <td>14.676200</td>\n",
       "      <td>14.767610</td>\n",
       "      <td>14.801915</td>\n",
       "      <td>14.470905</td>\n",
       "      <td>14.759144</td>\n",
       "      <td>...</td>\n",
       "      <td>1.565277</td>\n",
       "      <td>1.515299</td>\n",
       "      <td>1.495985</td>\n",
       "      <td>1.424418</td>\n",
       "      <td>1.403615</td>\n",
       "      <td>1.373038</td>\n",
       "      <td>1.347695</td>\n",
       "      <td>1.219013</td>\n",
       "      <td>1.204189</td>\n",
       "      <td>1.171621</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258562</th>\n",
       "      <td>15.342736</td>\n",
       "      <td>15.268235</td>\n",
       "      <td>14.767487</td>\n",
       "      <td>14.833338</td>\n",
       "      <td>15.182319</td>\n",
       "      <td>15.032301</td>\n",
       "      <td>15.135279</td>\n",
       "      <td>14.986033</td>\n",
       "      <td>14.523176</td>\n",
       "      <td>14.760864</td>\n",
       "      <td>...</td>\n",
       "      <td>1.548098</td>\n",
       "      <td>1.497742</td>\n",
       "      <td>1.476907</td>\n",
       "      <td>1.411053</td>\n",
       "      <td>1.392624</td>\n",
       "      <td>1.357882</td>\n",
       "      <td>1.342165</td>\n",
       "      <td>1.206396</td>\n",
       "      <td>1.192899</td>\n",
       "      <td>1.161873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258563</th>\n",
       "      <td>15.417672</td>\n",
       "      <td>15.209519</td>\n",
       "      <td>15.227086</td>\n",
       "      <td>15.050615</td>\n",
       "      <td>15.014187</td>\n",
       "      <td>14.988183</td>\n",
       "      <td>14.869835</td>\n",
       "      <td>14.833524</td>\n",
       "      <td>15.113196</td>\n",
       "      <td>14.780570</td>\n",
       "      <td>...</td>\n",
       "      <td>1.551618</td>\n",
       "      <td>1.500829</td>\n",
       "      <td>1.481116</td>\n",
       "      <td>1.415062</td>\n",
       "      <td>1.399726</td>\n",
       "      <td>1.361596</td>\n",
       "      <td>1.348633</td>\n",
       "      <td>1.211961</td>\n",
       "      <td>1.196354</td>\n",
       "      <td>1.167099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258564</th>\n",
       "      <td>15.550139</td>\n",
       "      <td>15.165336</td>\n",
       "      <td>15.311102</td>\n",
       "      <td>14.910869</td>\n",
       "      <td>15.164230</td>\n",
       "      <td>14.892066</td>\n",
       "      <td>14.440414</td>\n",
       "      <td>14.737539</td>\n",
       "      <td>15.230703</td>\n",
       "      <td>14.715859</td>\n",
       "      <td>...</td>\n",
       "      <td>1.551810</td>\n",
       "      <td>1.501694</td>\n",
       "      <td>1.482230</td>\n",
       "      <td>1.427955</td>\n",
       "      <td>1.396774</td>\n",
       "      <td>1.363856</td>\n",
       "      <td>1.358862</td>\n",
       "      <td>1.215891</td>\n",
       "      <td>1.203465</td>\n",
       "      <td>1.170352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258565</th>\n",
       "      <td>15.507817</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>14.582937</td>\n",
       "      <td>15.034059</td>\n",
       "      <td>15.065383</td>\n",
       "      <td>15.038354</td>\n",
       "      <td>14.917981</td>\n",
       "      <td>14.720025</td>\n",
       "      <td>14.311971</td>\n",
       "      <td>14.889401</td>\n",
       "      <td>...</td>\n",
       "      <td>1.547706</td>\n",
       "      <td>1.499425</td>\n",
       "      <td>1.487254</td>\n",
       "      <td>1.421248</td>\n",
       "      <td>1.404496</td>\n",
       "      <td>1.364686</td>\n",
       "      <td>1.351653</td>\n",
       "      <td>1.215876</td>\n",
       "      <td>1.202956</td>\n",
       "      <td>1.167359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258566</th>\n",
       "      <td>15.383293</td>\n",
       "      <td>15.198988</td>\n",
       "      <td>14.562686</td>\n",
       "      <td>14.932847</td>\n",
       "      <td>14.683030</td>\n",
       "      <td>15.125163</td>\n",
       "      <td>14.715623</td>\n",
       "      <td>14.864902</td>\n",
       "      <td>14.525089</td>\n",
       "      <td>14.814646</td>\n",
       "      <td>...</td>\n",
       "      <td>1.558149</td>\n",
       "      <td>1.503663</td>\n",
       "      <td>1.484549</td>\n",
       "      <td>1.412808</td>\n",
       "      <td>1.409314</td>\n",
       "      <td>1.368852</td>\n",
       "      <td>1.356838</td>\n",
       "      <td>1.221560</td>\n",
       "      <td>1.208233</td>\n",
       "      <td>1.172770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258567</th>\n",
       "      <td>15.311678</td>\n",
       "      <td>15.160098</td>\n",
       "      <td>15.050910</td>\n",
       "      <td>15.083774</td>\n",
       "      <td>15.079447</td>\n",
       "      <td>14.706424</td>\n",
       "      <td>14.851494</td>\n",
       "      <td>14.892938</td>\n",
       "      <td>14.936247</td>\n",
       "      <td>14.793656</td>\n",
       "      <td>...</td>\n",
       "      <td>1.539600</td>\n",
       "      <td>1.490070</td>\n",
       "      <td>1.469258</td>\n",
       "      <td>1.402975</td>\n",
       "      <td>1.384618</td>\n",
       "      <td>1.349995</td>\n",
       "      <td>1.331127</td>\n",
       "      <td>1.196923</td>\n",
       "      <td>1.183898</td>\n",
       "      <td>1.152347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258568</th>\n",
       "      <td>15.456894</td>\n",
       "      <td>15.324301</td>\n",
       "      <td>13.318613</td>\n",
       "      <td>15.424937</td>\n",
       "      <td>15.097132</td>\n",
       "      <td>14.885293</td>\n",
       "      <td>14.875608</td>\n",
       "      <td>14.869237</td>\n",
       "      <td>12.787630</td>\n",
       "      <td>14.906541</td>\n",
       "      <td>...</td>\n",
       "      <td>1.544716</td>\n",
       "      <td>1.496629</td>\n",
       "      <td>1.476117</td>\n",
       "      <td>1.396135</td>\n",
       "      <td>1.387354</td>\n",
       "      <td>1.353192</td>\n",
       "      <td>1.330413</td>\n",
       "      <td>1.196687</td>\n",
       "      <td>1.182963</td>\n",
       "      <td>1.149842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258569</th>\n",
       "      <td>15.388724</td>\n",
       "      <td>15.190195</td>\n",
       "      <td>15.260268</td>\n",
       "      <td>15.138413</td>\n",
       "      <td>14.928661</td>\n",
       "      <td>14.715119</td>\n",
       "      <td>14.286913</td>\n",
       "      <td>14.888772</td>\n",
       "      <td>15.192167</td>\n",
       "      <td>14.795960</td>\n",
       "      <td>...</td>\n",
       "      <td>1.559484</td>\n",
       "      <td>1.507177</td>\n",
       "      <td>1.489565</td>\n",
       "      <td>1.421882</td>\n",
       "      <td>1.405033</td>\n",
       "      <td>1.365901</td>\n",
       "      <td>1.355275</td>\n",
       "      <td>1.215633</td>\n",
       "      <td>1.201544</td>\n",
       "      <td>1.169977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258570</th>\n",
       "      <td>15.470592</td>\n",
       "      <td>15.194942</td>\n",
       "      <td>15.139555</td>\n",
       "      <td>14.660520</td>\n",
       "      <td>15.265506</td>\n",
       "      <td>15.014727</td>\n",
       "      <td>15.057435</td>\n",
       "      <td>14.671453</td>\n",
       "      <td>14.956932</td>\n",
       "      <td>14.792725</td>\n",
       "      <td>...</td>\n",
       "      <td>1.553501</td>\n",
       "      <td>1.503023</td>\n",
       "      <td>1.482619</td>\n",
       "      <td>1.431348</td>\n",
       "      <td>1.399701</td>\n",
       "      <td>1.363005</td>\n",
       "      <td>1.343547</td>\n",
       "      <td>1.210016</td>\n",
       "      <td>1.190277</td>\n",
       "      <td>1.168791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258571</th>\n",
       "      <td>15.581962</td>\n",
       "      <td>15.317689</td>\n",
       "      <td>14.897017</td>\n",
       "      <td>14.636907</td>\n",
       "      <td>15.174027</td>\n",
       "      <td>14.556255</td>\n",
       "      <td>14.702180</td>\n",
       "      <td>14.835689</td>\n",
       "      <td>14.561365</td>\n",
       "      <td>14.978003</td>\n",
       "      <td>...</td>\n",
       "      <td>1.554288</td>\n",
       "      <td>1.504086</td>\n",
       "      <td>1.484471</td>\n",
       "      <td>1.417357</td>\n",
       "      <td>1.401969</td>\n",
       "      <td>1.364412</td>\n",
       "      <td>1.348826</td>\n",
       "      <td>1.212967</td>\n",
       "      <td>1.199730</td>\n",
       "      <td>1.167925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258572</th>\n",
       "      <td>15.593478</td>\n",
       "      <td>15.084060</td>\n",
       "      <td>15.231037</td>\n",
       "      <td>15.161142</td>\n",
       "      <td>15.385525</td>\n",
       "      <td>14.553924</td>\n",
       "      <td>14.704660</td>\n",
       "      <td>14.585119</td>\n",
       "      <td>15.001003</td>\n",
       "      <td>14.585009</td>\n",
       "      <td>...</td>\n",
       "      <td>1.540897</td>\n",
       "      <td>1.493836</td>\n",
       "      <td>1.467416</td>\n",
       "      <td>1.402674</td>\n",
       "      <td>1.380441</td>\n",
       "      <td>1.350998</td>\n",
       "      <td>1.323967</td>\n",
       "      <td>1.196504</td>\n",
       "      <td>1.178825</td>\n",
       "      <td>1.153953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258573</th>\n",
       "      <td>15.420884</td>\n",
       "      <td>15.382603</td>\n",
       "      <td>15.372107</td>\n",
       "      <td>14.791288</td>\n",
       "      <td>14.976904</td>\n",
       "      <td>14.822740</td>\n",
       "      <td>14.345371</td>\n",
       "      <td>14.873242</td>\n",
       "      <td>15.118467</td>\n",
       "      <td>14.854315</td>\n",
       "      <td>...</td>\n",
       "      <td>1.549430</td>\n",
       "      <td>1.499239</td>\n",
       "      <td>1.479757</td>\n",
       "      <td>1.414266</td>\n",
       "      <td>1.398967</td>\n",
       "      <td>1.360509</td>\n",
       "      <td>1.345789</td>\n",
       "      <td>1.210957</td>\n",
       "      <td>1.197339</td>\n",
       "      <td>1.164143</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258574</th>\n",
       "      <td>15.389529</td>\n",
       "      <td>15.194880</td>\n",
       "      <td>15.364438</td>\n",
       "      <td>15.138276</td>\n",
       "      <td>15.173040</td>\n",
       "      <td>14.700299</td>\n",
       "      <td>14.869185</td>\n",
       "      <td>14.697205</td>\n",
       "      <td>15.402881</td>\n",
       "      <td>14.693478</td>\n",
       "      <td>...</td>\n",
       "      <td>1.545517</td>\n",
       "      <td>1.504592</td>\n",
       "      <td>1.475789</td>\n",
       "      <td>1.415100</td>\n",
       "      <td>1.393654</td>\n",
       "      <td>1.356001</td>\n",
       "      <td>1.345646</td>\n",
       "      <td>1.205727</td>\n",
       "      <td>1.191330</td>\n",
       "      <td>1.329041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258575</th>\n",
       "      <td>15.460724</td>\n",
       "      <td>15.230290</td>\n",
       "      <td>14.880944</td>\n",
       "      <td>15.156775</td>\n",
       "      <td>15.073581</td>\n",
       "      <td>14.848773</td>\n",
       "      <td>14.823035</td>\n",
       "      <td>14.856911</td>\n",
       "      <td>14.654697</td>\n",
       "      <td>14.801658</td>\n",
       "      <td>...</td>\n",
       "      <td>1.541378</td>\n",
       "      <td>1.491040</td>\n",
       "      <td>1.468681</td>\n",
       "      <td>1.401401</td>\n",
       "      <td>1.383824</td>\n",
       "      <td>1.350837</td>\n",
       "      <td>1.334866</td>\n",
       "      <td>1.198366</td>\n",
       "      <td>1.184294</td>\n",
       "      <td>1.153128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258576</th>\n",
       "      <td>15.275249</td>\n",
       "      <td>15.360243</td>\n",
       "      <td>14.992857</td>\n",
       "      <td>15.409637</td>\n",
       "      <td>14.806920</td>\n",
       "      <td>14.954047</td>\n",
       "      <td>14.753681</td>\n",
       "      <td>14.851479</td>\n",
       "      <td>14.644144</td>\n",
       "      <td>14.816859</td>\n",
       "      <td>...</td>\n",
       "      <td>1.557416</td>\n",
       "      <td>1.507411</td>\n",
       "      <td>1.488775</td>\n",
       "      <td>1.412625</td>\n",
       "      <td>1.405375</td>\n",
       "      <td>1.368060</td>\n",
       "      <td>1.352435</td>\n",
       "      <td>1.217116</td>\n",
       "      <td>1.204072</td>\n",
       "      <td>1.170141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258577</th>\n",
       "      <td>15.413322</td>\n",
       "      <td>15.132917</td>\n",
       "      <td>15.436670</td>\n",
       "      <td>15.112096</td>\n",
       "      <td>14.850840</td>\n",
       "      <td>14.773754</td>\n",
       "      <td>14.620209</td>\n",
       "      <td>14.712634</td>\n",
       "      <td>15.414936</td>\n",
       "      <td>14.648556</td>\n",
       "      <td>...</td>\n",
       "      <td>1.553401</td>\n",
       "      <td>1.505624</td>\n",
       "      <td>1.485955</td>\n",
       "      <td>1.480112</td>\n",
       "      <td>1.401029</td>\n",
       "      <td>1.363018</td>\n",
       "      <td>1.349853</td>\n",
       "      <td>1.214708</td>\n",
       "      <td>1.198585</td>\n",
       "      <td>1.175361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258578</th>\n",
       "      <td>15.447568</td>\n",
       "      <td>15.200667</td>\n",
       "      <td>14.974768</td>\n",
       "      <td>14.089715</td>\n",
       "      <td>14.581283</td>\n",
       "      <td>15.554160</td>\n",
       "      <td>15.226023</td>\n",
       "      <td>14.879849</td>\n",
       "      <td>14.430126</td>\n",
       "      <td>14.866126</td>\n",
       "      <td>...</td>\n",
       "      <td>1.561671</td>\n",
       "      <td>1.509003</td>\n",
       "      <td>1.493257</td>\n",
       "      <td>1.426441</td>\n",
       "      <td>1.410848</td>\n",
       "      <td>1.371553</td>\n",
       "      <td>1.359102</td>\n",
       "      <td>1.224112</td>\n",
       "      <td>1.212105</td>\n",
       "      <td>1.175864</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258579</th>\n",
       "      <td>15.405040</td>\n",
       "      <td>15.187845</td>\n",
       "      <td>15.370057</td>\n",
       "      <td>15.078938</td>\n",
       "      <td>15.025821</td>\n",
       "      <td>14.822713</td>\n",
       "      <td>14.588957</td>\n",
       "      <td>14.760382</td>\n",
       "      <td>15.293041</td>\n",
       "      <td>14.714131</td>\n",
       "      <td>...</td>\n",
       "      <td>1.544327</td>\n",
       "      <td>1.495011</td>\n",
       "      <td>1.474709</td>\n",
       "      <td>1.409673</td>\n",
       "      <td>1.392745</td>\n",
       "      <td>1.355444</td>\n",
       "      <td>1.341391</td>\n",
       "      <td>1.206125</td>\n",
       "      <td>1.193002</td>\n",
       "      <td>1.159835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258580</th>\n",
       "      <td>15.305927</td>\n",
       "      <td>15.117646</td>\n",
       "      <td>15.199018</td>\n",
       "      <td>14.973410</td>\n",
       "      <td>14.982769</td>\n",
       "      <td>14.876768</td>\n",
       "      <td>14.908226</td>\n",
       "      <td>14.863563</td>\n",
       "      <td>15.247890</td>\n",
       "      <td>14.755064</td>\n",
       "      <td>...</td>\n",
       "      <td>1.546165</td>\n",
       "      <td>1.497212</td>\n",
       "      <td>1.476967</td>\n",
       "      <td>1.531975</td>\n",
       "      <td>1.395109</td>\n",
       "      <td>1.357701</td>\n",
       "      <td>1.339261</td>\n",
       "      <td>1.206396</td>\n",
       "      <td>1.193075</td>\n",
       "      <td>1.155582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258581</th>\n",
       "      <td>15.308954</td>\n",
       "      <td>15.194398</td>\n",
       "      <td>15.241792</td>\n",
       "      <td>15.055014</td>\n",
       "      <td>14.871706</td>\n",
       "      <td>14.896046</td>\n",
       "      <td>14.735912</td>\n",
       "      <td>14.839741</td>\n",
       "      <td>15.175065</td>\n",
       "      <td>14.887203</td>\n",
       "      <td>...</td>\n",
       "      <td>1.556662</td>\n",
       "      <td>1.507129</td>\n",
       "      <td>1.487751</td>\n",
       "      <td>1.422396</td>\n",
       "      <td>1.404416</td>\n",
       "      <td>1.367108</td>\n",
       "      <td>1.352035</td>\n",
       "      <td>1.216157</td>\n",
       "      <td>1.202958</td>\n",
       "      <td>1.170631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258582</th>\n",
       "      <td>15.558739</td>\n",
       "      <td>15.068993</td>\n",
       "      <td>15.063557</td>\n",
       "      <td>14.575308</td>\n",
       "      <td>15.150017</td>\n",
       "      <td>14.820409</td>\n",
       "      <td>14.857006</td>\n",
       "      <td>14.961782</td>\n",
       "      <td>14.641143</td>\n",
       "      <td>14.779661</td>\n",
       "      <td>...</td>\n",
       "      <td>1.557313</td>\n",
       "      <td>1.507379</td>\n",
       "      <td>1.486345</td>\n",
       "      <td>1.418182</td>\n",
       "      <td>1.404914</td>\n",
       "      <td>1.367108</td>\n",
       "      <td>1.347811</td>\n",
       "      <td>1.215470</td>\n",
       "      <td>1.201845</td>\n",
       "      <td>1.169141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258583</th>\n",
       "      <td>15.352983</td>\n",
       "      <td>15.158652</td>\n",
       "      <td>15.044494</td>\n",
       "      <td>15.212901</td>\n",
       "      <td>14.719315</td>\n",
       "      <td>14.854465</td>\n",
       "      <td>14.869559</td>\n",
       "      <td>14.728704</td>\n",
       "      <td>14.949983</td>\n",
       "      <td>14.755064</td>\n",
       "      <td>...</td>\n",
       "      <td>1.555289</td>\n",
       "      <td>1.504508</td>\n",
       "      <td>1.484138</td>\n",
       "      <td>1.417263</td>\n",
       "      <td>1.405486</td>\n",
       "      <td>1.364485</td>\n",
       "      <td>1.349776</td>\n",
       "      <td>1.212911</td>\n",
       "      <td>1.199868</td>\n",
       "      <td>1.166726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258584</th>\n",
       "      <td>15.338182</td>\n",
       "      <td>15.365560</td>\n",
       "      <td>14.361351</td>\n",
       "      <td>15.393332</td>\n",
       "      <td>14.798405</td>\n",
       "      <td>14.746990</td>\n",
       "      <td>14.988743</td>\n",
       "      <td>14.693599</td>\n",
       "      <td>13.552999</td>\n",
       "      <td>14.693478</td>\n",
       "      <td>...</td>\n",
       "      <td>1.582721</td>\n",
       "      <td>1.503691</td>\n",
       "      <td>1.485357</td>\n",
       "      <td>1.414018</td>\n",
       "      <td>1.402461</td>\n",
       "      <td>1.361736</td>\n",
       "      <td>1.343262</td>\n",
       "      <td>1.210269</td>\n",
       "      <td>1.196264</td>\n",
       "      <td>1.166726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258585</th>\n",
       "      <td>15.287563</td>\n",
       "      <td>15.169083</td>\n",
       "      <td>15.276290</td>\n",
       "      <td>15.225818</td>\n",
       "      <td>14.594334</td>\n",
       "      <td>14.889330</td>\n",
       "      <td>14.837293</td>\n",
       "      <td>14.716005</td>\n",
       "      <td>15.205129</td>\n",
       "      <td>14.789558</td>\n",
       "      <td>...</td>\n",
       "      <td>1.555162</td>\n",
       "      <td>1.506120</td>\n",
       "      <td>1.485558</td>\n",
       "      <td>1.421532</td>\n",
       "      <td>1.403427</td>\n",
       "      <td>1.367408</td>\n",
       "      <td>1.357999</td>\n",
       "      <td>1.218854</td>\n",
       "      <td>1.205389</td>\n",
       "      <td>1.173320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258586</th>\n",
       "      <td>15.456894</td>\n",
       "      <td>15.054371</td>\n",
       "      <td>15.300131</td>\n",
       "      <td>15.010965</td>\n",
       "      <td>15.181326</td>\n",
       "      <td>14.747956</td>\n",
       "      <td>14.939821</td>\n",
       "      <td>14.837503</td>\n",
       "      <td>15.115634</td>\n",
       "      <td>14.411098</td>\n",
       "      <td>...</td>\n",
       "      <td>1.551932</td>\n",
       "      <td>1.501634</td>\n",
       "      <td>1.481830</td>\n",
       "      <td>1.414875</td>\n",
       "      <td>1.405448</td>\n",
       "      <td>1.362316</td>\n",
       "      <td>1.340563</td>\n",
       "      <td>1.208829</td>\n",
       "      <td>1.192051</td>\n",
       "      <td>1.166276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258587</th>\n",
       "      <td>15.544019</td>\n",
       "      <td>15.268235</td>\n",
       "      <td>15.091359</td>\n",
       "      <td>15.218384</td>\n",
       "      <td>15.211854</td>\n",
       "      <td>14.571803</td>\n",
       "      <td>15.050639</td>\n",
       "      <td>14.885250</td>\n",
       "      <td>14.946872</td>\n",
       "      <td>14.679872</td>\n",
       "      <td>...</td>\n",
       "      <td>1.558749</td>\n",
       "      <td>1.508000</td>\n",
       "      <td>1.487986</td>\n",
       "      <td>1.422632</td>\n",
       "      <td>1.408730</td>\n",
       "      <td>1.368608</td>\n",
       "      <td>1.354762</td>\n",
       "      <td>1.218915</td>\n",
       "      <td>1.204563</td>\n",
       "      <td>1.174166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258588</th>\n",
       "      <td>15.287721</td>\n",
       "      <td>15.219766</td>\n",
       "      <td>15.061977</td>\n",
       "      <td>15.126815</td>\n",
       "      <td>14.968223</td>\n",
       "      <td>14.822531</td>\n",
       "      <td>14.883539</td>\n",
       "      <td>14.875955</td>\n",
       "      <td>14.935229</td>\n",
       "      <td>14.791578</td>\n",
       "      <td>...</td>\n",
       "      <td>1.547447</td>\n",
       "      <td>1.498066</td>\n",
       "      <td>1.478301</td>\n",
       "      <td>1.411119</td>\n",
       "      <td>1.394569</td>\n",
       "      <td>1.358972</td>\n",
       "      <td>1.338936</td>\n",
       "      <td>1.206359</td>\n",
       "      <td>1.195155</td>\n",
       "      <td>1.155259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258589</th>\n",
       "      <td>15.624072</td>\n",
       "      <td>15.180497</td>\n",
       "      <td>15.104633</td>\n",
       "      <td>15.057051</td>\n",
       "      <td>15.362707</td>\n",
       "      <td>14.610286</td>\n",
       "      <td>15.022866</td>\n",
       "      <td>14.820564</td>\n",
       "      <td>15.071229</td>\n",
       "      <td>14.684655</td>\n",
       "      <td>...</td>\n",
       "      <td>1.547478</td>\n",
       "      <td>1.496635</td>\n",
       "      <td>1.476645</td>\n",
       "      <td>1.413020</td>\n",
       "      <td>1.393982</td>\n",
       "      <td>1.357734</td>\n",
       "      <td>1.341086</td>\n",
       "      <td>1.205877</td>\n",
       "      <td>1.192624</td>\n",
       "      <td>1.158108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258590</th>\n",
       "      <td>15.381699</td>\n",
       "      <td>15.313609</td>\n",
       "      <td>15.269821</td>\n",
       "      <td>14.993920</td>\n",
       "      <td>14.811308</td>\n",
       "      <td>14.916525</td>\n",
       "      <td>14.649242</td>\n",
       "      <td>14.766787</td>\n",
       "      <td>15.046244</td>\n",
       "      <td>14.686403</td>\n",
       "      <td>...</td>\n",
       "      <td>1.559913</td>\n",
       "      <td>1.508386</td>\n",
       "      <td>1.488468</td>\n",
       "      <td>1.418456</td>\n",
       "      <td>1.400592</td>\n",
       "      <td>1.365247</td>\n",
       "      <td>1.345349</td>\n",
       "      <td>1.212023</td>\n",
       "      <td>1.192982</td>\n",
       "      <td>1.166579</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258591</th>\n",
       "      <td>15.448223</td>\n",
       "      <td>15.117904</td>\n",
       "      <td>15.024713</td>\n",
       "      <td>14.897309</td>\n",
       "      <td>15.192571</td>\n",
       "      <td>14.939831</td>\n",
       "      <td>14.908583</td>\n",
       "      <td>14.842694</td>\n",
       "      <td>14.688859</td>\n",
       "      <td>14.766377</td>\n",
       "      <td>...</td>\n",
       "      <td>1.557596</td>\n",
       "      <td>1.507311</td>\n",
       "      <td>1.487927</td>\n",
       "      <td>1.419848</td>\n",
       "      <td>1.402768</td>\n",
       "      <td>1.367694</td>\n",
       "      <td>1.349350</td>\n",
       "      <td>1.214023</td>\n",
       "      <td>1.207107</td>\n",
       "      <td>1.177073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258592</th>\n",
       "      <td>15.363464</td>\n",
       "      <td>15.129538</td>\n",
       "      <td>14.261464</td>\n",
       "      <td>14.989365</td>\n",
       "      <td>14.977250</td>\n",
       "      <td>14.616286</td>\n",
       "      <td>14.737873</td>\n",
       "      <td>14.762414</td>\n",
       "      <td>13.364838</td>\n",
       "      <td>14.776681</td>\n",
       "      <td>...</td>\n",
       "      <td>1.537591</td>\n",
       "      <td>1.489485</td>\n",
       "      <td>1.470248</td>\n",
       "      <td>1.398058</td>\n",
       "      <td>1.388236</td>\n",
       "      <td>1.352045</td>\n",
       "      <td>1.332857</td>\n",
       "      <td>1.199192</td>\n",
       "      <td>1.186807</td>\n",
       "      <td>1.147629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258593</th>\n",
       "      <td>15.450450</td>\n",
       "      <td>15.123037</td>\n",
       "      <td>15.391052</td>\n",
       "      <td>14.861351</td>\n",
       "      <td>14.967892</td>\n",
       "      <td>14.863656</td>\n",
       "      <td>15.005543</td>\n",
       "      <td>14.784143</td>\n",
       "      <td>15.318700</td>\n",
       "      <td>14.691450</td>\n",
       "      <td>...</td>\n",
       "      <td>1.546356</td>\n",
       "      <td>1.497000</td>\n",
       "      <td>1.478557</td>\n",
       "      <td>1.414392</td>\n",
       "      <td>1.394363</td>\n",
       "      <td>1.358473</td>\n",
       "      <td>1.343702</td>\n",
       "      <td>1.208340</td>\n",
       "      <td>1.193726</td>\n",
       "      <td>1.159873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258594</th>\n",
       "      <td>15.372743</td>\n",
       "      <td>15.235543</td>\n",
       "      <td>15.024330</td>\n",
       "      <td>15.250732</td>\n",
       "      <td>14.874520</td>\n",
       "      <td>14.997891</td>\n",
       "      <td>14.753845</td>\n",
       "      <td>14.687381</td>\n",
       "      <td>14.911613</td>\n",
       "      <td>14.900293</td>\n",
       "      <td>...</td>\n",
       "      <td>1.550985</td>\n",
       "      <td>1.503463</td>\n",
       "      <td>1.483897</td>\n",
       "      <td>1.418075</td>\n",
       "      <td>1.406680</td>\n",
       "      <td>1.367017</td>\n",
       "      <td>1.355043</td>\n",
       "      <td>1.219500</td>\n",
       "      <td>1.204245</td>\n",
       "      <td>1.170354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258595</th>\n",
       "      <td>15.432787</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>15.064958</td>\n",
       "      <td>15.131238</td>\n",
       "      <td>15.137947</td>\n",
       "      <td>14.967332</td>\n",
       "      <td>14.947018</td>\n",
       "      <td>14.903766</td>\n",
       "      <td>15.005450</td>\n",
       "      <td>14.851190</td>\n",
       "      <td>...</td>\n",
       "      <td>1.545633</td>\n",
       "      <td>1.495302</td>\n",
       "      <td>1.474515</td>\n",
       "      <td>1.399458</td>\n",
       "      <td>1.391393</td>\n",
       "      <td>1.355333</td>\n",
       "      <td>1.336569</td>\n",
       "      <td>1.201654</td>\n",
       "      <td>1.188556</td>\n",
       "      <td>1.156444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258596</th>\n",
       "      <td>15.335618</td>\n",
       "      <td>15.145347</td>\n",
       "      <td>15.257909</td>\n",
       "      <td>15.088672</td>\n",
       "      <td>14.841105</td>\n",
       "      <td>15.308472</td>\n",
       "      <td>15.009477</td>\n",
       "      <td>15.017152</td>\n",
       "      <td>15.059889</td>\n",
       "      <td>14.839604</td>\n",
       "      <td>...</td>\n",
       "      <td>1.560372</td>\n",
       "      <td>1.505848</td>\n",
       "      <td>1.503134</td>\n",
       "      <td>1.424420</td>\n",
       "      <td>1.405820</td>\n",
       "      <td>1.369455</td>\n",
       "      <td>1.364683</td>\n",
       "      <td>1.224723</td>\n",
       "      <td>1.211418</td>\n",
       "      <td>1.174233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258597</th>\n",
       "      <td>15.473202</td>\n",
       "      <td>15.075278</td>\n",
       "      <td>15.180714</td>\n",
       "      <td>14.597938</td>\n",
       "      <td>15.101030</td>\n",
       "      <td>14.994434</td>\n",
       "      <td>14.457027</td>\n",
       "      <td>14.831579</td>\n",
       "      <td>15.194942</td>\n",
       "      <td>14.880279</td>\n",
       "      <td>...</td>\n",
       "      <td>1.552953</td>\n",
       "      <td>1.502408</td>\n",
       "      <td>1.483354</td>\n",
       "      <td>1.412244</td>\n",
       "      <td>1.397181</td>\n",
       "      <td>1.359167</td>\n",
       "      <td>1.341703</td>\n",
       "      <td>1.206501</td>\n",
       "      <td>1.192692</td>\n",
       "      <td>1.162230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258598</th>\n",
       "      <td>15.381699</td>\n",
       "      <td>15.109011</td>\n",
       "      <td>14.867799</td>\n",
       "      <td>15.323954</td>\n",
       "      <td>15.290286</td>\n",
       "      <td>14.950929</td>\n",
       "      <td>14.854496</td>\n",
       "      <td>14.938597</td>\n",
       "      <td>14.770643</td>\n",
       "      <td>14.776681</td>\n",
       "      <td>...</td>\n",
       "      <td>1.543583</td>\n",
       "      <td>1.493576</td>\n",
       "      <td>1.473757</td>\n",
       "      <td>1.403246</td>\n",
       "      <td>1.388468</td>\n",
       "      <td>1.353216</td>\n",
       "      <td>1.333909</td>\n",
       "      <td>1.199511</td>\n",
       "      <td>1.186807</td>\n",
       "      <td>1.156781</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258599</th>\n",
       "      <td>15.354914</td>\n",
       "      <td>15.165107</td>\n",
       "      <td>15.177085</td>\n",
       "      <td>14.861333</td>\n",
       "      <td>14.963645</td>\n",
       "      <td>14.910319</td>\n",
       "      <td>15.027229</td>\n",
       "      <td>14.729553</td>\n",
       "      <td>15.119695</td>\n",
       "      <td>14.750155</td>\n",
       "      <td>...</td>\n",
       "      <td>1.560536</td>\n",
       "      <td>1.509679</td>\n",
       "      <td>1.490041</td>\n",
       "      <td>1.427455</td>\n",
       "      <td>1.406989</td>\n",
       "      <td>1.369809</td>\n",
       "      <td>1.353969</td>\n",
       "      <td>1.217651</td>\n",
       "      <td>1.199005</td>\n",
       "      <td>1.178356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258600</th>\n",
       "      <td>15.495646</td>\n",
       "      <td>15.184177</td>\n",
       "      <td>15.370751</td>\n",
       "      <td>15.184035</td>\n",
       "      <td>15.086318</td>\n",
       "      <td>14.704673</td>\n",
       "      <td>14.816816</td>\n",
       "      <td>14.625736</td>\n",
       "      <td>15.372068</td>\n",
       "      <td>14.709706</td>\n",
       "      <td>...</td>\n",
       "      <td>1.549749</td>\n",
       "      <td>1.499536</td>\n",
       "      <td>1.479015</td>\n",
       "      <td>1.427200</td>\n",
       "      <td>1.400323</td>\n",
       "      <td>1.360947</td>\n",
       "      <td>1.355172</td>\n",
       "      <td>1.212179</td>\n",
       "      <td>1.198235</td>\n",
       "      <td>1.164804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258601</th>\n",
       "      <td>15.392720</td>\n",
       "      <td>15.130715</td>\n",
       "      <td>14.985706</td>\n",
       "      <td>14.716383</td>\n",
       "      <td>14.994576</td>\n",
       "      <td>14.693560</td>\n",
       "      <td>15.212597</td>\n",
       "      <td>14.979656</td>\n",
       "      <td>14.858457</td>\n",
       "      <td>14.708731</td>\n",
       "      <td>...</td>\n",
       "      <td>1.540612</td>\n",
       "      <td>1.489235</td>\n",
       "      <td>1.469753</td>\n",
       "      <td>1.403362</td>\n",
       "      <td>1.383288</td>\n",
       "      <td>1.350041</td>\n",
       "      <td>1.331026</td>\n",
       "      <td>1.196986</td>\n",
       "      <td>1.183507</td>\n",
       "      <td>1.154469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258602</th>\n",
       "      <td>15.480795</td>\n",
       "      <td>15.191393</td>\n",
       "      <td>14.771899</td>\n",
       "      <td>14.885592</td>\n",
       "      <td>15.079311</td>\n",
       "      <td>14.941034</td>\n",
       "      <td>15.329712</td>\n",
       "      <td>15.035706</td>\n",
       "      <td>14.632084</td>\n",
       "      <td>15.017137</td>\n",
       "      <td>...</td>\n",
       "      <td>1.555494</td>\n",
       "      <td>1.503482</td>\n",
       "      <td>1.485640</td>\n",
       "      <td>1.416238</td>\n",
       "      <td>1.402319</td>\n",
       "      <td>1.365309</td>\n",
       "      <td>1.351185</td>\n",
       "      <td>1.215158</td>\n",
       "      <td>1.202357</td>\n",
       "      <td>1.162814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258603</th>\n",
       "      <td>15.372107</td>\n",
       "      <td>15.174592</td>\n",
       "      <td>15.048471</td>\n",
       "      <td>14.950120</td>\n",
       "      <td>14.952733</td>\n",
       "      <td>14.866206</td>\n",
       "      <td>14.797710</td>\n",
       "      <td>14.781063</td>\n",
       "      <td>15.086269</td>\n",
       "      <td>14.796931</td>\n",
       "      <td>...</td>\n",
       "      <td>1.554691</td>\n",
       "      <td>1.503830</td>\n",
       "      <td>1.483157</td>\n",
       "      <td>1.420468</td>\n",
       "      <td>1.399915</td>\n",
       "      <td>1.362940</td>\n",
       "      <td>1.346951</td>\n",
       "      <td>1.211097</td>\n",
       "      <td>1.194424</td>\n",
       "      <td>1.166227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258604</th>\n",
       "      <td>15.476776</td>\n",
       "      <td>15.192399</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.881242</td>\n",
       "      <td>15.047349</td>\n",
       "      <td>14.711054</td>\n",
       "      <td>15.245537</td>\n",
       "      <td>14.916401</td>\n",
       "      <td>15.016711</td>\n",
       "      <td>14.852756</td>\n",
       "      <td>...</td>\n",
       "      <td>1.537926</td>\n",
       "      <td>1.487827</td>\n",
       "      <td>1.467043</td>\n",
       "      <td>1.392884</td>\n",
       "      <td>1.381319</td>\n",
       "      <td>1.348269</td>\n",
       "      <td>1.326824</td>\n",
       "      <td>1.194717</td>\n",
       "      <td>1.181320</td>\n",
       "      <td>1.148943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258605</th>\n",
       "      <td>15.125095</td>\n",
       "      <td>15.167122</td>\n",
       "      <td>14.678163</td>\n",
       "      <td>15.236298</td>\n",
       "      <td>14.124653</td>\n",
       "      <td>15.027319</td>\n",
       "      <td>15.149306</td>\n",
       "      <td>14.981514</td>\n",
       "      <td>14.454290</td>\n",
       "      <td>15.031401</td>\n",
       "      <td>...</td>\n",
       "      <td>1.563489</td>\n",
       "      <td>3.320512</td>\n",
       "      <td>1.492524</td>\n",
       "      <td>1.427368</td>\n",
       "      <td>1.411563</td>\n",
       "      <td>1.372869</td>\n",
       "      <td>1.364472</td>\n",
       "      <td>1.226010</td>\n",
       "      <td>1.212394</td>\n",
       "      <td>1.177165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258606</th>\n",
       "      <td>14.820450</td>\n",
       "      <td>15.298337</td>\n",
       "      <td>15.120340</td>\n",
       "      <td>14.600380</td>\n",
       "      <td>14.317633</td>\n",
       "      <td>14.839234</td>\n",
       "      <td>14.970261</td>\n",
       "      <td>14.934777</td>\n",
       "      <td>15.084482</td>\n",
       "      <td>14.832232</td>\n",
       "      <td>...</td>\n",
       "      <td>1.551376</td>\n",
       "      <td>1.500645</td>\n",
       "      <td>1.480938</td>\n",
       "      <td>1.408299</td>\n",
       "      <td>1.396570</td>\n",
       "      <td>1.360560</td>\n",
       "      <td>1.342032</td>\n",
       "      <td>1.206894</td>\n",
       "      <td>1.193261</td>\n",
       "      <td>1.161136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258607</th>\n",
       "      <td>15.409687</td>\n",
       "      <td>15.110260</td>\n",
       "      <td>15.349818</td>\n",
       "      <td>15.158579</td>\n",
       "      <td>14.954273</td>\n",
       "      <td>15.256287</td>\n",
       "      <td>14.581131</td>\n",
       "      <td>14.753274</td>\n",
       "      <td>15.218134</td>\n",
       "      <td>14.886753</td>\n",
       "      <td>...</td>\n",
       "      <td>1.555668</td>\n",
       "      <td>1.508720</td>\n",
       "      <td>1.486536</td>\n",
       "      <td>1.419724</td>\n",
       "      <td>1.411752</td>\n",
       "      <td>1.367447</td>\n",
       "      <td>1.356828</td>\n",
       "      <td>1.221090</td>\n",
       "      <td>1.207828</td>\n",
       "      <td>1.174995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM258608</th>\n",
       "      <td>15.145438</td>\n",
       "      <td>15.088462</td>\n",
       "      <td>15.372265</td>\n",
       "      <td>15.059676</td>\n",
       "      <td>14.737609</td>\n",
       "      <td>14.826989</td>\n",
       "      <td>14.902070</td>\n",
       "      <td>14.843601</td>\n",
       "      <td>15.276786</td>\n",
       "      <td>14.770150</td>\n",
       "      <td>...</td>\n",
       "      <td>1.551896</td>\n",
       "      <td>1.501483</td>\n",
       "      <td>1.482247</td>\n",
       "      <td>1.722942</td>\n",
       "      <td>1.396606</td>\n",
       "      <td>1.359985</td>\n",
       "      <td>1.343060</td>\n",
       "      <td>1.207954</td>\n",
       "      <td>1.193456</td>\n",
       "      <td>1.167486</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58 rows × 21989 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Gene Symbol       COX1     EEF1A1  IGK /// IGKC     TMSB4X       ATP6  \\\n",
       "GSM258551    15.312190  15.180497     15.394483  15.164475  14.765942   \n",
       "GSM258552    15.335128  15.180229     15.375848  15.337793  14.939455   \n",
       "GSM258553    15.147944  15.262090     15.259431  14.886320  14.530669   \n",
       "GSM258554    15.235262  15.246653     15.250159  14.723359  14.956952   \n",
       "GSM258555    15.402517  15.272874     11.996869  14.734361  14.960515   \n",
       "GSM258556    15.235435  15.181830     15.053283  14.755676  14.767122   \n",
       "GSM258557    15.266640  15.162696     15.450697  15.090716  14.648069   \n",
       "GSM258558    15.236145  15.126815     15.391052  14.747466  14.847039   \n",
       "GSM258559    15.530658  15.315813     14.658909  15.357297  14.897657   \n",
       "GSM258560    15.323297  15.406522     14.706833  14.563453  15.093474   \n",
       "GSM258561    15.440451  15.309941     14.756265  14.946459  15.127632   \n",
       "GSM258562    15.342736  15.268235     14.767487  14.833338  15.182319   \n",
       "GSM258563    15.417672  15.209519     15.227086  15.050615  15.014187   \n",
       "GSM258564    15.550139  15.165336     15.311102  14.910869  15.164230   \n",
       "GSM258565    15.507817  15.191393     14.582937  15.034059  15.065383   \n",
       "GSM258566    15.383293  15.198988     14.562686  14.932847  14.683030   \n",
       "GSM258567    15.311678  15.160098     15.050910  15.083774  15.079447   \n",
       "GSM258568    15.456894  15.324301     13.318613  15.424937  15.097132   \n",
       "GSM258569    15.388724  15.190195     15.260268  15.138413  14.928661   \n",
       "GSM258570    15.470592  15.194942     15.139555  14.660520  15.265506   \n",
       "GSM258571    15.581962  15.317689     14.897017  14.636907  15.174027   \n",
       "GSM258572    15.593478  15.084060     15.231037  15.161142  15.385525   \n",
       "GSM258573    15.420884  15.382603     15.372107  14.791288  14.976904   \n",
       "GSM258574    15.389529  15.194880     15.364438  15.138276  15.173040   \n",
       "GSM258575    15.460724  15.230290     14.880944  15.156775  15.073581   \n",
       "GSM258576    15.275249  15.360243     14.992857  15.409637  14.806920   \n",
       "GSM258577    15.413322  15.132917     15.436670  15.112096  14.850840   \n",
       "GSM258578    15.447568  15.200667     14.974768  14.089715  14.581283   \n",
       "GSM258579    15.405040  15.187845     15.370057  15.078938  15.025821   \n",
       "GSM258580    15.305927  15.117646     15.199018  14.973410  14.982769   \n",
       "GSM258581    15.308954  15.194398     15.241792  15.055014  14.871706   \n",
       "GSM258582    15.558739  15.068993     15.063557  14.575308  15.150017   \n",
       "GSM258583    15.352983  15.158652     15.044494  15.212901  14.719315   \n",
       "GSM258584    15.338182  15.365560     14.361351  15.393332  14.798405   \n",
       "GSM258585    15.287563  15.169083     15.276290  15.225818  14.594334   \n",
       "GSM258586    15.456894  15.054371     15.300131  15.010965  15.181326   \n",
       "GSM258587    15.544019  15.268235     15.091359  15.218384  15.211854   \n",
       "GSM258588    15.287721  15.219766     15.061977  15.126815  14.968223   \n",
       "GSM258589    15.624072  15.180497     15.104633  15.057051  15.362707   \n",
       "GSM258590    15.381699  15.313609     15.269821  14.993920  14.811308   \n",
       "GSM258591    15.448223  15.117904     15.024713  14.897309  15.192571   \n",
       "GSM258592    15.363464  15.129538     14.261464  14.989365  14.977250   \n",
       "GSM258593    15.450450  15.123037     15.391052  14.861351  14.967892   \n",
       "GSM258594    15.372743  15.235543     15.024330  15.250732  14.874520   \n",
       "GSM258595    15.432787  15.191393     15.064958  15.131238  15.137947   \n",
       "GSM258596    15.335618  15.145347     15.257909  15.088672  14.841105   \n",
       "GSM258597    15.473202  15.075278     15.180714  14.597938  15.101030   \n",
       "GSM258598    15.381699  15.109011     14.867799  15.323954  15.290286   \n",
       "GSM258599    15.354914  15.165107     15.177085  14.861333  14.963645   \n",
       "GSM258600    15.495646  15.184177     15.370751  15.184035  15.086318   \n",
       "GSM258601    15.392720  15.130715     14.985706  14.716383  14.994576   \n",
       "GSM258602    15.480795  15.191393     14.771899  14.885592  15.079311   \n",
       "GSM258603    15.372107  15.174592     15.048471  14.950120  14.952733   \n",
       "GSM258604    15.476776  15.192399     15.236298  14.881242  15.047349   \n",
       "GSM258605    15.125095  15.167122     14.678163  15.236298  14.124653   \n",
       "GSM258606    14.820450  15.298337     15.120340  14.600380  14.317633   \n",
       "GSM258607    15.409687  15.110260     15.349818  15.158579  14.954273   \n",
       "GSM258608    15.145438  15.088462     15.372265  15.059676  14.737609   \n",
       "\n",
       "Gene Symbol      RPLP1      RPL37     RPL37A      IGLC1      RPL41  ...  \\\n",
       "GSM258551    14.899661  14.607402  14.805437  15.389362  14.764169  ...   \n",
       "GSM258552    14.798201  14.620797  14.818706  15.287337  14.722343  ...   \n",
       "GSM258553    15.114353  14.861911  14.920742  15.084031  14.636443  ...   \n",
       "GSM258554    14.861104  14.815595  14.581683  15.245323  14.623252  ...   \n",
       "GSM258555    15.183791  14.854306  14.807841  11.158493  14.870209  ...   \n",
       "GSM258556    14.844315  14.842160  15.027328  14.795219  14.864671  ...   \n",
       "GSM258557    14.873081  14.714121  14.648810  15.378435  14.732511  ...   \n",
       "GSM258558    14.989528  14.802895  14.832079  15.323409  14.704899  ...   \n",
       "GSM258559    14.795771  14.918985  14.740109  14.271701  14.745470  ...   \n",
       "GSM258560    14.843761  15.142578  14.671068  14.123293  14.866845  ...   \n",
       "GSM258561    14.676200  14.767610  14.801915  14.470905  14.759144  ...   \n",
       "GSM258562    15.032301  15.135279  14.986033  14.523176  14.760864  ...   \n",
       "GSM258563    14.988183  14.869835  14.833524  15.113196  14.780570  ...   \n",
       "GSM258564    14.892066  14.440414  14.737539  15.230703  14.715859  ...   \n",
       "GSM258565    15.038354  14.917981  14.720025  14.311971  14.889401  ...   \n",
       "GSM258566    15.125163  14.715623  14.864902  14.525089  14.814646  ...   \n",
       "GSM258567    14.706424  14.851494  14.892938  14.936247  14.793656  ...   \n",
       "GSM258568    14.885293  14.875608  14.869237  12.787630  14.906541  ...   \n",
       "GSM258569    14.715119  14.286913  14.888772  15.192167  14.795960  ...   \n",
       "GSM258570    15.014727  15.057435  14.671453  14.956932  14.792725  ...   \n",
       "GSM258571    14.556255  14.702180  14.835689  14.561365  14.978003  ...   \n",
       "GSM258572    14.553924  14.704660  14.585119  15.001003  14.585009  ...   \n",
       "GSM258573    14.822740  14.345371  14.873242  15.118467  14.854315  ...   \n",
       "GSM258574    14.700299  14.869185  14.697205  15.402881  14.693478  ...   \n",
       "GSM258575    14.848773  14.823035  14.856911  14.654697  14.801658  ...   \n",
       "GSM258576    14.954047  14.753681  14.851479  14.644144  14.816859  ...   \n",
       "GSM258577    14.773754  14.620209  14.712634  15.414936  14.648556  ...   \n",
       "GSM258578    15.554160  15.226023  14.879849  14.430126  14.866126  ...   \n",
       "GSM258579    14.822713  14.588957  14.760382  15.293041  14.714131  ...   \n",
       "GSM258580    14.876768  14.908226  14.863563  15.247890  14.755064  ...   \n",
       "GSM258581    14.896046  14.735912  14.839741  15.175065  14.887203  ...   \n",
       "GSM258582    14.820409  14.857006  14.961782  14.641143  14.779661  ...   \n",
       "GSM258583    14.854465  14.869559  14.728704  14.949983  14.755064  ...   \n",
       "GSM258584    14.746990  14.988743  14.693599  13.552999  14.693478  ...   \n",
       "GSM258585    14.889330  14.837293  14.716005  15.205129  14.789558  ...   \n",
       "GSM258586    14.747956  14.939821  14.837503  15.115634  14.411098  ...   \n",
       "GSM258587    14.571803  15.050639  14.885250  14.946872  14.679872  ...   \n",
       "GSM258588    14.822531  14.883539  14.875955  14.935229  14.791578  ...   \n",
       "GSM258589    14.610286  15.022866  14.820564  15.071229  14.684655  ...   \n",
       "GSM258590    14.916525  14.649242  14.766787  15.046244  14.686403  ...   \n",
       "GSM258591    14.939831  14.908583  14.842694  14.688859  14.766377  ...   \n",
       "GSM258592    14.616286  14.737873  14.762414  13.364838  14.776681  ...   \n",
       "GSM258593    14.863656  15.005543  14.784143  15.318700  14.691450  ...   \n",
       "GSM258594    14.997891  14.753845  14.687381  14.911613  14.900293  ...   \n",
       "GSM258595    14.967332  14.947018  14.903766  15.005450  14.851190  ...   \n",
       "GSM258596    15.308472  15.009477  15.017152  15.059889  14.839604  ...   \n",
       "GSM258597    14.994434  14.457027  14.831579  15.194942  14.880279  ...   \n",
       "GSM258598    14.950929  14.854496  14.938597  14.770643  14.776681  ...   \n",
       "GSM258599    14.910319  15.027229  14.729553  15.119695  14.750155  ...   \n",
       "GSM258600    14.704673  14.816816  14.625736  15.372068  14.709706  ...   \n",
       "GSM258601    14.693560  15.212597  14.979656  14.858457  14.708731  ...   \n",
       "GSM258602    14.941034  15.329712  15.035706  14.632084  15.017137  ...   \n",
       "GSM258603    14.866206  14.797710  14.781063  15.086269  14.796931  ...   \n",
       "GSM258604    14.711054  15.245537  14.916401  15.016711  14.852756  ...   \n",
       "GSM258605    15.027319  15.149306  14.981514  14.454290  15.031401  ...   \n",
       "GSM258606    14.839234  14.970261  14.934777  15.084482  14.832232  ...   \n",
       "GSM258607    15.256287  14.581131  14.753274  15.218134  14.886753  ...   \n",
       "GSM258608    14.826989  14.902070  14.843601  15.276786  14.770150  ...   \n",
       "\n",
       "Gene Symbol  LOC100130078  BAGE2 /// BAGE4  VPS9D1-AS1  LOC100128343  \\\n",
       "GSM258551        1.559278         1.508207    1.488264      1.428252   \n",
       "GSM258552        1.541614         1.491403    1.476264      1.415345   \n",
       "GSM258553        1.562901         1.512844    1.494091      1.419606   \n",
       "GSM258554        1.564354         1.515226    1.496433      1.429780   \n",
       "GSM258555        1.552817         1.498749    1.480635      1.414616   \n",
       "GSM258556        1.547007         1.496887    1.476711      1.411491   \n",
       "GSM258557        1.560682         1.510637    1.491109      1.439103   \n",
       "GSM258558        1.559284         1.508974    1.491793      1.430029   \n",
       "GSM258559        1.557593         1.508101    1.488935      1.418820   \n",
       "GSM258560        1.554900         2.500992    1.484531      1.412437   \n",
       "GSM258561        1.565277         1.515299    1.495985      1.424418   \n",
       "GSM258562        1.548098         1.497742    1.476907      1.411053   \n",
       "GSM258563        1.551618         1.500829    1.481116      1.415062   \n",
       "GSM258564        1.551810         1.501694    1.482230      1.427955   \n",
       "GSM258565        1.547706         1.499425    1.487254      1.421248   \n",
       "GSM258566        1.558149         1.503663    1.484549      1.412808   \n",
       "GSM258567        1.539600         1.490070    1.469258      1.402975   \n",
       "GSM258568        1.544716         1.496629    1.476117      1.396135   \n",
       "GSM258569        1.559484         1.507177    1.489565      1.421882   \n",
       "GSM258570        1.553501         1.503023    1.482619      1.431348   \n",
       "GSM258571        1.554288         1.504086    1.484471      1.417357   \n",
       "GSM258572        1.540897         1.493836    1.467416      1.402674   \n",
       "GSM258573        1.549430         1.499239    1.479757      1.414266   \n",
       "GSM258574        1.545517         1.504592    1.475789      1.415100   \n",
       "GSM258575        1.541378         1.491040    1.468681      1.401401   \n",
       "GSM258576        1.557416         1.507411    1.488775      1.412625   \n",
       "GSM258577        1.553401         1.505624    1.485955      1.480112   \n",
       "GSM258578        1.561671         1.509003    1.493257      1.426441   \n",
       "GSM258579        1.544327         1.495011    1.474709      1.409673   \n",
       "GSM258580        1.546165         1.497212    1.476967      1.531975   \n",
       "GSM258581        1.556662         1.507129    1.487751      1.422396   \n",
       "GSM258582        1.557313         1.507379    1.486345      1.418182   \n",
       "GSM258583        1.555289         1.504508    1.484138      1.417263   \n",
       "GSM258584        1.582721         1.503691    1.485357      1.414018   \n",
       "GSM258585        1.555162         1.506120    1.485558      1.421532   \n",
       "GSM258586        1.551932         1.501634    1.481830      1.414875   \n",
       "GSM258587        1.558749         1.508000    1.487986      1.422632   \n",
       "GSM258588        1.547447         1.498066    1.478301      1.411119   \n",
       "GSM258589        1.547478         1.496635    1.476645      1.413020   \n",
       "GSM258590        1.559913         1.508386    1.488468      1.418456   \n",
       "GSM258591        1.557596         1.507311    1.487927      1.419848   \n",
       "GSM258592        1.537591         1.489485    1.470248      1.398058   \n",
       "GSM258593        1.546356         1.497000    1.478557      1.414392   \n",
       "GSM258594        1.550985         1.503463    1.483897      1.418075   \n",
       "GSM258595        1.545633         1.495302    1.474515      1.399458   \n",
       "GSM258596        1.560372         1.505848    1.503134      1.424420   \n",
       "GSM258597        1.552953         1.502408    1.483354      1.412244   \n",
       "GSM258598        1.543583         1.493576    1.473757      1.403246   \n",
       "GSM258599        1.560536         1.509679    1.490041      1.427455   \n",
       "GSM258600        1.549749         1.499536    1.479015      1.427200   \n",
       "GSM258601        1.540612         1.489235    1.469753      1.403362   \n",
       "GSM258602        1.555494         1.503482    1.485640      1.416238   \n",
       "GSM258603        1.554691         1.503830    1.483157      1.420468   \n",
       "GSM258604        1.537926         1.487827    1.467043      1.392884   \n",
       "GSM258605        1.563489         3.320512    1.492524      1.427368   \n",
       "GSM258606        1.551376         1.500645    1.480938      1.408299   \n",
       "GSM258607        1.555668         1.508720    1.486536      1.419724   \n",
       "GSM258608        1.551896         1.501483    1.482247      1.722942   \n",
       "\n",
       "Gene Symbol   SNORA78  EHMT1-IT1     GALR3    NKX1-1     UTS2R  OR7E104P  \n",
       "GSM258551    1.402514   1.364011  1.355402  1.213887  1.198629  1.167121  \n",
       "GSM258552    1.393830   1.354702  1.341295  1.206120  1.192367  1.160904  \n",
       "GSM258553    1.408625   1.371686  1.351569  1.219612  1.205589  1.374079  \n",
       "GSM258554    1.410923   1.373798  1.358518  1.222785  1.208315  1.175638  \n",
       "GSM258555    1.398829   1.362018  1.340931  1.211382  1.197830  1.165753  \n",
       "GSM258556    1.393893   1.357067  1.340861  1.205536  1.192689  1.155603  \n",
       "GSM258557    1.410817   1.369805  1.360618  1.224959  1.211072  1.176772  \n",
       "GSM258558    1.408326   1.367987  1.353805  1.217283  1.202751  1.172249  \n",
       "GSM258559    1.408168   1.367468  1.353732  1.219779  1.204510  1.169585  \n",
       "GSM258560    1.401998   1.365011  1.348822  1.212259  1.198294  1.168115  \n",
       "GSM258561    1.403615   1.373038  1.347695  1.219013  1.204189  1.171621  \n",
       "GSM258562    1.392624   1.357882  1.342165  1.206396  1.192899  1.161873  \n",
       "GSM258563    1.399726   1.361596  1.348633  1.211961  1.196354  1.167099  \n",
       "GSM258564    1.396774   1.363856  1.358862  1.215891  1.203465  1.170352  \n",
       "GSM258565    1.404496   1.364686  1.351653  1.215876  1.202956  1.167359  \n",
       "GSM258566    1.409314   1.368852  1.356838  1.221560  1.208233  1.172770  \n",
       "GSM258567    1.384618   1.349995  1.331127  1.196923  1.183898  1.152347  \n",
       "GSM258568    1.387354   1.353192  1.330413  1.196687  1.182963  1.149842  \n",
       "GSM258569    1.405033   1.365901  1.355275  1.215633  1.201544  1.169977  \n",
       "GSM258570    1.399701   1.363005  1.343547  1.210016  1.190277  1.168791  \n",
       "GSM258571    1.401969   1.364412  1.348826  1.212967  1.199730  1.167925  \n",
       "GSM258572    1.380441   1.350998  1.323967  1.196504  1.178825  1.153953  \n",
       "GSM258573    1.398967   1.360509  1.345789  1.210957  1.197339  1.164143  \n",
       "GSM258574    1.393654   1.356001  1.345646  1.205727  1.191330  1.329041  \n",
       "GSM258575    1.383824   1.350837  1.334866  1.198366  1.184294  1.153128  \n",
       "GSM258576    1.405375   1.368060  1.352435  1.217116  1.204072  1.170141  \n",
       "GSM258577    1.401029   1.363018  1.349853  1.214708  1.198585  1.175361  \n",
       "GSM258578    1.410848   1.371553  1.359102  1.224112  1.212105  1.175864  \n",
       "GSM258579    1.392745   1.355444  1.341391  1.206125  1.193002  1.159835  \n",
       "GSM258580    1.395109   1.357701  1.339261  1.206396  1.193075  1.155582  \n",
       "GSM258581    1.404416   1.367108  1.352035  1.216157  1.202958  1.170631  \n",
       "GSM258582    1.404914   1.367108  1.347811  1.215470  1.201845  1.169141  \n",
       "GSM258583    1.405486   1.364485  1.349776  1.212911  1.199868  1.166726  \n",
       "GSM258584    1.402461   1.361736  1.343262  1.210269  1.196264  1.166726  \n",
       "GSM258585    1.403427   1.367408  1.357999  1.218854  1.205389  1.173320  \n",
       "GSM258586    1.405448   1.362316  1.340563  1.208829  1.192051  1.166276  \n",
       "GSM258587    1.408730   1.368608  1.354762  1.218915  1.204563  1.174166  \n",
       "GSM258588    1.394569   1.358972  1.338936  1.206359  1.195155  1.155259  \n",
       "GSM258589    1.393982   1.357734  1.341086  1.205877  1.192624  1.158108  \n",
       "GSM258590    1.400592   1.365247  1.345349  1.212023  1.192982  1.166579  \n",
       "GSM258591    1.402768   1.367694  1.349350  1.214023  1.207107  1.177073  \n",
       "GSM258592    1.388236   1.352045  1.332857  1.199192  1.186807  1.147629  \n",
       "GSM258593    1.394363   1.358473  1.343702  1.208340  1.193726  1.159873  \n",
       "GSM258594    1.406680   1.367017  1.355043  1.219500  1.204245  1.170354  \n",
       "GSM258595    1.391393   1.355333  1.336569  1.201654  1.188556  1.156444  \n",
       "GSM258596    1.405820   1.369455  1.364683  1.224723  1.211418  1.174233  \n",
       "GSM258597    1.397181   1.359167  1.341703  1.206501  1.192692  1.162230  \n",
       "GSM258598    1.388468   1.353216  1.333909  1.199511  1.186807  1.156781  \n",
       "GSM258599    1.406989   1.369809  1.353969  1.217651  1.199005  1.178356  \n",
       "GSM258600    1.400323   1.360947  1.355172  1.212179  1.198235  1.164804  \n",
       "GSM258601    1.383288   1.350041  1.331026  1.196986  1.183507  1.154469  \n",
       "GSM258602    1.402319   1.365309  1.351185  1.215158  1.202357  1.162814  \n",
       "GSM258603    1.399915   1.362940  1.346951  1.211097  1.194424  1.166227  \n",
       "GSM258604    1.381319   1.348269  1.326824  1.194717  1.181320  1.148943  \n",
       "GSM258605    1.411563   1.372869  1.364472  1.226010  1.212394  1.177165  \n",
       "GSM258606    1.396570   1.360560  1.342032  1.206894  1.193261  1.161136  \n",
       "GSM258607    1.411752   1.367447  1.356828  1.221090  1.207828  1.174995  \n",
       "GSM258608    1.396606   1.359985  1.343060  1.207954  1.193456  1.167486  \n",
       "\n",
       "[58 rows x 21989 columns]"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# columns should be feautures (genes) and rows should be data points (gene expression per sample)\n",
    "# the model will sepearte the samples (rows) into two clusters based on their features (expression values per gene)\n",
    "final_df = reset_index_df.T\n",
    "final_df\n",
    "\n",
    "## number of rows and columns in our dataframe\n",
    "#final_df.shape\n",
    "\n",
    "# .values gives an array of all the values for each feature (for GSM258551 [15.312190, 15.180497, 15.394483...], for GSM258552 [15.33512815, 15.18022902, 15.37584761...], for GSM258553 [15.14794428, 15.26208991, 15.25943145...], ... )\n",
    "# final_df.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1, 1, 0, 1, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1,\n",
       "       1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0,\n",
       "       0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1], dtype=int32)"
      ]
     },
     "execution_count": 274,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create a K-Means model with the specified number of clusters'\n",
    "# n_init is the number of times the K-Means algorithm will be run with different centroid seeds\n",
    "kmeans = KMeans(n_clusters = 2, n_init=10, random_state = 0)\n",
    "\n",
    "# fit the model to our data (array for the gene expression values per feature)\n",
    "kmeans.fit(final_df)\n",
    "\n",
    "# the number of clusters (2) and the number of features (genes) in our data\n",
    "kmeans.labels_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample Label_x</th>\n",
       "      <th>Cluster Numbers_x</th>\n",
       "      <th>Sample Label_y</th>\n",
       "      <th>Cluster Numbers_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GSM258551</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258551</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GSM258552</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258552</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GSM258553</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258553</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GSM258554</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258554</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GSM258555</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258555</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GSM258556</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258556</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GSM258557</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258557</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GSM258558</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258558</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GSM258559</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258559</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GSM258560</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258560</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GSM258561</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258561</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GSM258562</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258562</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GSM258563</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258563</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GSM258564</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258564</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GSM258565</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258565</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GSM258566</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258566</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GSM258567</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258567</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GSM258568</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258568</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GSM258569</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258569</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GSM258570</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258570</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GSM258571</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258571</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GSM258572</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258572</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GSM258573</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258573</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GSM258574</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258574</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GSM258575</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258575</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GSM258576</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258576</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GSM258577</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258577</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GSM258578</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258578</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GSM258579</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258579</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GSM258580</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258580</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GSM258581</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258581</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GSM258582</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258582</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GSM258583</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258583</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>GSM258584</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258584</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>GSM258585</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258585</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>GSM258586</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258586</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>GSM258587</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258587</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GSM258588</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258588</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>GSM258589</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258589</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>GSM258590</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258590</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>GSM258591</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258591</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GSM258592</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258592</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>GSM258593</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258593</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>GSM258594</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258594</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>GSM258595</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258595</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>GSM258596</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258596</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>GSM258597</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258597</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>GSM258598</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258598</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>GSM258599</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258599</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>GSM258600</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258600</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>GSM258601</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258601</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>GSM258602</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258602</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>GSM258603</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258603</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>GSM258604</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258604</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>GSM258605</td>\n",
       "      <td>0</td>\n",
       "      <td>GSM258605</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>GSM258606</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258606</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>GSM258607</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258607</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>GSM258608</td>\n",
       "      <td>1</td>\n",
       "      <td>GSM258608</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Sample Label_x  Cluster Numbers_x Sample Label_y  Cluster Numbers_y\n",
       "0       GSM258551                  1      GSM258551                  1\n",
       "1       GSM258552                  1      GSM258552                  1\n",
       "2       GSM258553                  0      GSM258553                  0\n",
       "3       GSM258554                  1      GSM258554                  1\n",
       "4       GSM258555                  0      GSM258555                  0\n",
       "5       GSM258556                  1      GSM258556                  1\n",
       "6       GSM258557                  0      GSM258557                  0\n",
       "7       GSM258558                  1      GSM258558                  1\n",
       "8       GSM258559                  1      GSM258559                  1\n",
       "9       GSM258560                  1      GSM258560                  1\n",
       "10      GSM258561                  1      GSM258561                  1\n",
       "11      GSM258562                  0      GSM258562                  0\n",
       "12      GSM258563                  0      GSM258563                  0\n",
       "13      GSM258564                  1      GSM258564                  1\n",
       "14      GSM258565                  0      GSM258565                  0\n",
       "15      GSM258566                  0      GSM258566                  0\n",
       "16      GSM258567                  1      GSM258567                  1\n",
       "17      GSM258568                  1      GSM258568                  1\n",
       "18      GSM258569                  1      GSM258569                  1\n",
       "19      GSM258570                  0      GSM258570                  0\n",
       "20      GSM258571                  0      GSM258571                  0\n",
       "21      GSM258572                  1      GSM258572                  1\n",
       "22      GSM258573                  1      GSM258573                  1\n",
       "23      GSM258574                  1      GSM258574                  1\n",
       "24      GSM258575                  1      GSM258575                  1\n",
       "25      GSM258576                  1      GSM258576                  1\n",
       "26      GSM258577                  1      GSM258577                  1\n",
       "27      GSM258578                  0      GSM258578                  0\n",
       "28      GSM258579                  1      GSM258579                  1\n",
       "29      GSM258580                  1      GSM258580                  1\n",
       "30      GSM258581                  1      GSM258581                  1\n",
       "31      GSM258582                  0      GSM258582                  0\n",
       "32      GSM258583                  0      GSM258583                  0\n",
       "33      GSM258584                  1      GSM258584                  1\n",
       "34      GSM258585                  0      GSM258585                  0\n",
       "35      GSM258586                  1      GSM258586                  1\n",
       "36      GSM258587                  1      GSM258587                  1\n",
       "37      GSM258588                  1      GSM258588                  1\n",
       "38      GSM258589                  1      GSM258589                  1\n",
       "39      GSM258590                  0      GSM258590                  0\n",
       "40      GSM258591                  1      GSM258591                  1\n",
       "41      GSM258592                  1      GSM258592                  1\n",
       "42      GSM258593                  0      GSM258593                  0\n",
       "43      GSM258594                  0      GSM258594                  0\n",
       "44      GSM258595                  0      GSM258595                  0\n",
       "45      GSM258596                  0      GSM258596                  0\n",
       "46      GSM258597                  1      GSM258597                  1\n",
       "47      GSM258598                  1      GSM258598                  1\n",
       "48      GSM258599                  0      GSM258599                  0\n",
       "49      GSM258600                  1      GSM258600                  1\n",
       "50      GSM258601                  1      GSM258601                  1\n",
       "51      GSM258602                  1      GSM258602                  1\n",
       "52      GSM258603                  0      GSM258603                  0\n",
       "53      GSM258604                  0      GSM258604                  0\n",
       "54      GSM258605                  0      GSM258605                  0\n",
       "55      GSM258606                  1      GSM258606                  1\n",
       "56      GSM258607                  1      GSM258607                  1\n",
       "57      GSM258608                  1      GSM258608                  1"
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframe with 58 smaple lables as a column, cluster number and cancer type as another column \n",
    "\n",
    "# sample label\n",
    "index_data = final_df.index\n",
    "cluster_type_df = pd.DataFrame(index_data, columns=['Sample Label'])\n",
    "cluster_type_df.set_index('Sample Label')\n",
    "\n",
    "# cluster number\n",
    "cluster_num = kmeans.labels_\n",
    "cluster_type_df['Cluster Numbers'] = cluster_num\n",
    "\n",
    "# ac or scc\n",
    "ac_scc_series = gse.phenotype_data[\"title\"].str.split('_').str[1]\n",
    "ac_scc_df = ac_scc_series.to_frame('Subtype')\n",
    "\n",
    "\n",
    "cluster_type_df = cluster_type_df.merge(cluster_type_df, left_index=True, right_index=True)\n",
    "cluster_type_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Q8. maybe see and plot which genes have highest expression in AC vs SC cells using plotting we did in session 12"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
