{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lung Cancer Subtyping\n",
    "a mathematical model that can discriminate between two lung cancer subtypes to build the model \n",
    "we use an unsupervised k-means clustering algorithm (Euclidean distance) of 58 NSCLC tumors using k=2\n",
    "\n",
    "data from: \n",
    "\n",
    "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE10245\n",
    "\n",
    "[GSE10245_family.soft.gz](https://ftp.ncbi.nlm.nih.gov/geo/series/GSE10nnn/GSE10245/soft/)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "19-Oct-2023 14:18:03 DEBUG utils - Directory ./ already exists. Skipping.\n",
      "19-Oct-2023 14:18:03 INFO GEOparse - Downloading ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE10nnn/GSE10245/soft/GSE10245_family.soft.gz to ./GSE10245_family.soft.gz\n",
      "100%|██████████| 37.1M/37.1M [00:13<00:00, 2.88MB/s]\n",
      "19-Oct-2023 14:18:16 DEBUG downloader - Size validation passed\n",
      "19-Oct-2023 14:18:16 DEBUG downloader - Moving /var/folders/gf/79pl6dp1569b76cgzy_9yv7c0000gs/T/tmpimol3pq0 to /Users/ymanasa/Library/CloudStorage/OneDrive-MichiganMedicine/1PhD/bioinf575/lung_cancer_subtyping/GSE10245_family.soft.gz\n",
      "19-Oct-2023 14:18:16 DEBUG downloader - Successfully downloaded ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE10nnn/GSE10245/soft/GSE10245_family.soft.gz\n",
      "19-Oct-2023 14:18:16 INFO GEOparse - Parsing ./GSE10245_family.soft.gz: \n",
      "19-Oct-2023 14:18:16 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "19-Oct-2023 14:18:16 DEBUG GEOparse - SERIES: GSE10245\n",
      "19-Oct-2023 14:18:16 DEBUG GEOparse - PLATFORM: GPL570\n",
      "/Users/ymanasa/anaconda3/lib/python3.11/site-packages/GEOparse/GEOparse.py:401: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "19-Oct-2023 14:18:17 DEBUG GEOparse - SAMPLE: GSM258551\n",
      "19-Oct-2023 14:18:17 DEBUG GEOparse - SAMPLE: GSM258552\n",
      "19-Oct-2023 14:18:17 DEBUG GEOparse - SAMPLE: GSM258553\n",
      "19-Oct-2023 14:18:17 DEBUG GEOparse - SAMPLE: GSM258554\n",
      "19-Oct-2023 14:18:17 DEBUG GEOparse - SAMPLE: GSM258555\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258556\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258557\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258558\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258559\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258560\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258561\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258562\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258563\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258564\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258565\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258566\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258567\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258568\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258569\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258570\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258571\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258572\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258573\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258574\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258575\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258576\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258577\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258578\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258579\n",
      "19-Oct-2023 14:18:18 DEBUG GEOparse - SAMPLE: GSM258580\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258581\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258582\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258583\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258584\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258585\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258586\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258587\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258588\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258589\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258590\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258591\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258592\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258593\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258594\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258595\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258596\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258597\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258598\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258599\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258600\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258601\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258602\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258603\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258604\n",
      "19-Oct-2023 14:18:19 DEBUG GEOparse - SAMPLE: GSM258605\n",
      "19-Oct-2023 14:18:20 DEBUG GEOparse - SAMPLE: GSM258606\n",
      "19-Oct-2023 14:18:20 DEBUG GEOparse - SAMPLE: GSM258607\n",
      "19-Oct-2023 14:18:20 DEBUG GEOparse - SAMPLE: GSM258608\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GSM example:\n",
      "Name:  GSM258551\n",
      "Metadata:\n",
      " - title : NSCLC_AC_10\n",
      " - geo_accession : GSM258551\n",
      " - status : Public on Oct 01 2009\n",
      " - submission_date : Jan 23 2008\n",
      " - last_update_date : Aug 28 2018\n",
      " - type : RNA\n",
      " - channel_count : 1\n",
      " - source_name_ch1 : human non-small cell lung cancer tumor tissue\n",
      " - organism_ch1 : Homo sapiens\n",
      " - taxid_ch1 : 9606\n",
      " - characteristics_ch1 : disease state: adenocarcinoma\n",
      " - treatment_protocol_ch1 : tumor tissue samples from surgically treated patients were collected intraoperatively, snap-frozen and stored at -80 C. All samples underwent a careful analysis of the histopathology, estimation of tumor and stromal content and subsequent macro-dissection of the tumor regions before enrollment into the microarray experiments.\n",
      " - molecule_ch1 : total RNA\n",
      " - extract_protocol_ch1 : total RNA was extracted using the RNeasy Kit (Qiagen, Hilden, Germany) according to the manufacturersÆ instructions.\n",
      " - label_ch1 : biotin\n",
      " - label_protocol_ch1 : Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 2 ug total RNA (Affymetrix ROC technical manual 701021 Rev.5).\n",
      " - hyb_protocol : Following fragmentation, 10 ug of cRNA were hybridized for 16 hr at 45C on Affymetrix GeneChip« U133 Plus 2.0 in the Affymetrix GeneChip« Hybridization Oven 640. GeneChips were washed and stained in the Affymetrix GeneChip« Fluidics Station 450.\n",
      " - scan_protocol : Hybridized microarrays were scanned with the Affymetrix GeneChip« Scanner 3000.\n",
      " - description : Gene expression data from human non-small cell lung cancer tumor tissue 10.\n",
      " - data_processing : Bioconductor [Gentleman et al. Genome Biology 2004] and R language [Ihaka and Gentleman. Journal of Computational and Graphical Statistics 1996] were used for pre-processing of the 58 hybridizations. The microarray data was normalized by the gcRMA method [Wu et al. Journal of the American Statistical Association 2004].\n",
      " - platform_id : GPL570\n",
      " - contact_name : Ruprecht,,Kuner\n",
      " - contact_laboratory : Unit Cancer Genome Research\n",
      " - contact_department : Molecular Genetics\n",
      " - contact_institute : German Cancer Research Center and National Center of Tumor Diseases\n",
      " - contact_address : Im Neuenheimer Feld 460\n",
      " - contact_city : Heidelberg\n",
      " - contact_zip/postal_code : 69120\n",
      " - contact_country : Germany\n",
      " - supplementary_file : ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM258nnn/GSM258551/suppl/GSM258551.CEL.gz\n",
      " - relation : Reanalyzed by: GSE60486, Reanalyzed by: GSE119087\n",
      " - series_id : GSE10245\n",
      " - data_row_count : 54675\n",
      "Table data:\n",
      "      ID_REF     VALUE\n",
      "0  1007_s_at  9.129905\n",
      "1    1053_at  8.034022\n",
      "2     117_at  3.564520\n",
      "3     121_at  4.746490\n",
      "4  1255_g_at  2.320698\n",
      "\n",
      "GPL example:\n",
      "Name:  GPL570\n",
      "Metadata:\n",
      " - title : [HG-U133_Plus_2] Affymetrix Human Genome U133 Plus 2.0 Array\n",
      " - geo_accession : GPL570\n",
      " - status : Public on Nov 07 2003\n",
      " - submission_date : Nov 07 2003\n",
      " - last_update_date : Dec 14 2020\n",
      " - technology : in situ oligonucleotide\n",
      " - distribution : commercial\n",
      " - organism : Homo sapiens\n",
      " - taxid : 9606\n",
      " - manufacturer : Affymetrix\n",
      " - manufacture_protocol : see manufacturer's web site, , Complete coverage of the Human Genome U133 Set plus 6,500 additional genes for analysis of over 47,000 transcripts, All probe sets represented on the GeneChip Human Genome U133 Set are identically replicated on the GeneChip Human Genome U133 Plus 2.0 Array. The sequences from which these probe sets were derived were selected from GenBank®, dbEST, and RefSeq. The sequence clusters were created from the UniGene database (Build 133, April 20, 2001) and then refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the University of California, Santa Cruz Golden-Path human genome database (April 2001 release)., In addition, there are 9,921 new probe sets representing approximately 6,500 new genes. These gene sequences were selected from GenBank, dbEST, and RefSeq. Sequence clusters were created from the UniGene database (Build 159, January 25, 2003) and refined by analysis and comparison with a number of other publicly available databases, including the Washington University EST trace repository and the NCBI human genome assembly (Build 31)., \n",
      " - description : Affymetrix submissions are typically submitted to GEO using the GEOarchive method described at http://www.ncbi.nlm.nih.gov/projects/geo/info/geo_affy.html, , June 03, 2009: annotation table updated with netaffx build 28, June 06, 2012: annotation table updated with netaffx build 32, June 23, 2016: annotation table updated with netaffx build 35\n",
      " - web_link : http://www.affymetrix.com/support/technical/byproduct.affx?product=hg-u133-plus, http://www.affymetrix.com/analysis/index.affx\n",
      " - contact_name : ,,Affymetrix, Inc.\n",
      " - contact_email : geo@ncbi.nlm.nih.gov, support@affymetrix.com\n",
      " - contact_phone : 888-362-2447\n",
      " - contact_institute : Affymetrix, Inc.\n",
      " - contact_address : \n",
      " - contact_city : Santa Clara\n",
      " - contact_state : CA\n",
      " - contact_zip/postal_code : 95051\n",
      " - contact_country : USA\n",
      " - contact_web_link : http://www.affymetrix.com/index.affx\n",
      " - relation : Alternative to: GPL4454 (Alternative CDF), Alternative to: GPL4866 (Alternative CDF), Alternative to: GPL5760 (Alternative CDF), Alternative to: GPL6671 (Alternative CDF), Alternative to: GPL6791 (Alternative CDF), Alternative to: GPL6879 (Alternative CDF), Alternative to: GPL7567 (Alternative CDF), Alternative to: GPL8019 (Alternative CDF), Alternative to: GPL8542 (Alternative CDF), Alternative to: GPL8715 (Alternative CDF), Alternative to: GPL8712 (Alternative CDF), Alternative to: GPL9102 (Probe Level Version), Alternative to: GPL9099 (Alternative CDF), Alternative to: GPL9101 (Alternative CDF), Alternative to: GPL9324 (Alternative CDF), Alternative to: GPL9486 (Alternative CDF), Alternative to: GPL9987 (Alternative CDF), Alternative to: GPL10175 (Alternative CDF), Alternative to: GPL10335 (Alternative CDF), Alternative to: GPL10371 (Alternative CDF), Alternative to: GPL10526 (Alternative CDF), Alternative to: GPL10881 (Alternative CDF), Alternative to: GPL10925 (Alternative CDF), Alternative to: GPL11084 (Alternative CDF), Alternative to: GPL11433 (Alternative CDF), Alternative to: GPL11670 (Alternative CDF), Alternative to: GPL13232 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 13.0.0]), Alternative to: GPL13303 (Alternative CDF.), Alternative to: GPL13916 ([Brainarray ENSG Version 14.1]), Alternative to: GPL14837 (Alternative CDF), Alternative to: GPL14877 ([Alternative CDF]), Alternative to: GPL15394 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 14.0]), Alternative to: GPL6732 (Alternative CDF [HGU133Plus2_Hs_REFSEQ, Brainarray version 7]), Alternative to: GPL15445 (Alternative CDF), Alternative to: GPL16066 (Alternative CDF), Alternative to: GPL16100 (Alternative CDF [GATExplorer_Ensembl v57]), Alternative to: GPL16268 (Alternative CDF [Brainarray ENSG Version 16]), Alternative to: GPL16273 (Alternative CDF [Brainarray ENSG Version 15]), Alternative to: GPL16311 (Alternative CDF [HGU133Plus2_Hs_ENTREZG Brainarray version 14.0.0]), Alternative to: GPL16356 (Alternative CDF [Brainarray HGU133Plus2_Hs_ENTREZG 15.0.0]), Alternative to: GPL16372 (Alternative CDF [Brainarray HGU133Plus2_Hs_UG v.13]), Alternative to: GPL17392 (Alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray 17.1]), Alternative to: GPL17394 (Alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray 13.0]), Alternative to: GPL17810 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 16]), Alternative to: GPL17811 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 17]), Alternative to: GPL17996 (Alternative CDF:HGU133Plus2_Hs_ENTREZG.cdf version 15.1.0), Alternative to: GPL18121 (custom CDF), Alternative to: GPL18756 (Alternative CDF [CDF: HGU133Plus2_Hs_ENTREZG_14.1.0]), Alternative to: GPL18850 (Alternative CDF), Alternative to: GPL18478 (Alternative CDF [Hs_ENTREZG_V13]), Alternative to: GPL19109 (Alternative CDF [Brainarray HGU133Plus2_Hs_ENTREZG_v18]), Alternative to: GPL19883 (Alternative CDF [Broad GenePattern]), Alternative to: GPL19918 (Gene symbol version, 10K), Alternative to: GPL20182 (Gene Symbol Version), Alternative to: GPL22321 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 20]), Alternative to: GPL22945 (alternative), Alternative to: GPL23270 ((Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 21]), Alternative to: GPL23432 (alternative CDF [HGU133Plus2_Hs_ENSG, Brainarray version 18.0.0]), Alternative to: GPL25740 (alternative), Alternative to: GPL28718 (Alternative CDF [HGU133Plus2_Hs_ENTREZG, Brainarray 23]), Alternative to: GPL29499 (Alternative CDF)\n",
      " - data_row_count : 54675\n",
      "Table data:\n",
      "          ID  GB_ACC SPOT_ID Species Scientific Name Annotation Date  \\\n",
      "0  1007_s_at  U48705     NaN            Homo sapiens     Oct 6, 2014   \n",
      "1    1053_at  M87338     NaN            Homo sapiens     Oct 6, 2014   \n",
      "2     117_at  X51757     NaN            Homo sapiens     Oct 6, 2014   \n",
      "3     121_at  X69699     NaN            Homo sapiens     Oct 6, 2014   \n",
      "4  1255_g_at  L36861     NaN            Homo sapiens     Oct 6, 2014   \n",
      "\n",
      "       Sequence Type                  Sequence Source  \\\n",
      "0  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "1  Exemplar sequence                          GenBank   \n",
      "2  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "3  Exemplar sequence                          GenBank   \n",
      "4  Exemplar sequence  Affymetrix Proprietary Database   \n",
      "\n",
      "                                  Target Description Representative Public ID  \\\n",
      "0  U48705 /FEATURE=mRNA /DEFINITION=HSU48705 Huma...                   U48705   \n",
      "1  M87338 /FEATURE= /DEFINITION=HUMA1SBU Human re...                   M87338   \n",
      "2  X51757 /FEATURE=cds /DEFINITION=HSP70B Human h...                   X51757   \n",
      "3  X69699 /FEATURE= /DEFINITION=HSPAX8A H.sapiens...                   X69699   \n",
      "4  L36861 /FEATURE=expanded_cds /DEFINITION=HUMGC...                   L36861   \n",
      "\n",
      "                                          Gene Title       Gene Symbol  \\\n",
      "0  discoidin domain receptor tyrosine kinase 1 //...  DDR1 /// MIR4640   \n",
      "1        replication factor C (activator 1) 2, 40kDa              RFC2   \n",
      "2               heat shock 70kDa protein 6 (HSP70B')             HSPA6   \n",
      "3                                       paired box 8              PAX8   \n",
      "4            guanylate cyclase activator 1A (retina)            GUCA1A   \n",
      "\n",
      "      ENTREZ_GENE_ID                               RefSeq Transcript ID  \\\n",
      "0  780 /// 100616237  NM_001202521 /// NM_001202522 /// NM_001202523...   \n",
      "1               5982  NM_001278791 /// NM_001278792 /// NM_001278793...   \n",
      "2               3310                                          NM_002155   \n",
      "3               7849  NM_003466 /// NM_013951 /// NM_013952 /// NM_0...   \n",
      "4               2978                         NM_000409 /// XM_006715073   \n",
      "\n",
      "                    Gene Ontology Biological Process  \\\n",
      "0  0001558 // regulation of cell growth // inferr...   \n",
      "1  0000278 // mitotic cell cycle // traceable aut...   \n",
      "2  0000902 // cell morphogenesis // inferred from...   \n",
      "3  0001655 // urogenital system development // in...   \n",
      "4  0007165 // signal transduction // non-traceabl...   \n",
      "\n",
      "                    Gene Ontology Cellular Component  \\\n",
      "0  0005576 // extracellular region // inferred fr...   \n",
      "1  0005634 // nucleus // inferred from electronic...   \n",
      "2  0005737 // cytoplasm // inferred from direct a...   \n",
      "3  0005634 // nucleus // inferred from direct ass...   \n",
      "4  0001750 // photoreceptor outer segment // infe...   \n",
      "\n",
      "                    Gene Ontology Molecular Function  \n",
      "0  0000166 // nucleotide binding // inferred from...  \n",
      "1  0000166 // nucleotide binding // inferred from...  \n",
      "2  0000166 // nucleotide binding // inferred from...  \n",
      "3  0000979 // RNA polymerase II core promoter seq...  \n",
      "4  0005509 // calcium ion binding // inferred fro...  \n"
     ]
    }
   ],
   "source": [
    "# load the data using the GEOparse module\n",
    "# code adopted from: https://geoparse.readthedocs.io/en/latest/usage.html\n",
    "\n",
    "import GEOparse\n",
    "gse = GEOparse.get_GEO(geo=\"GSE10245\", destdir=\"./\")\n",
    "\n",
    "print()\n",
    "print(\"GSM example:\")\n",
    "for gsm_name, gsm in gse.gsms.items():\n",
    "    print(\"Name: \", gsm_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gsm.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print (\"Table data:\",)\n",
    "    print (gsm.table.head())\n",
    "    break\n",
    "\n",
    "print()\n",
    "print(\"GPL example:\")\n",
    "for gpl_name, gpl in gse.gpls.items():\n",
    "    print(\"Name: \", gpl_name)\n",
    "    print(\"Metadata:\",)\n",
    "    for key, value in gpl.metadata.items():\n",
    "        print(\" - %s : %s\" % (key, \", \".join(value)))\n",
    "    print(\"Table data:\",)\n",
    "    print(gpl.table.head())\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
